{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import json\n",
    "from langchain_core.runnables import (\n",
    "    RunnableBranch,\n",
    "    RunnableLambda,\n",
    "    RunnableParallel,\n",
    "    RunnablePassthrough,\n",
    ")\n",
    "from langchain_core.prompts import ChatPromptTemplate\n",
    "from langchain_core.prompts.prompt import PromptTemplate\n",
    "from langchain_core.pydantic_v1 import BaseModel, Field\n",
    "from typing import Tuple, List, Optional\n",
    "from langchain_core.messages import AIMessage, HumanMessage\n",
    "from langchain_core.output_parsers import StrOutputParser\n",
    "from langchain_community.graphs import Neo4jGraph\n",
    "# from langchain.document_loaders import WikipediaLoader\n",
    "from langchain.text_splitter import TokenTextSplitter\n",
    "from langchain_openai import ChatOpenAI\n",
    "from langchain_experimental.graph_transformers import LLMGraphTransformer\n",
    "from neo4j import GraphDatabase\n",
    "from yfiles_jupyter_graphs import GraphWidget\n",
    "from langchain_community.vectorstores import Neo4jVector\n",
    "from langchain_openai import OpenAIEmbeddings\n",
    "from langchain_community.vectorstores.neo4j_vector import remove_lucene_chars\n",
    "from langchain_core.runnables import ConfigurableField, RunnableParallel, RunnablePassthrough\n",
    "# from llama_index.readers.json import JSONReader\n",
    "from langchain.schema import Document\n",
    "from pymongo import MongoClient\n",
    "from concurrent.futures import ThreadPoolExecutor, as_completed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "graph = Neo4jGraph()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "mongo_uri = \"uri\"  \n",
    "\n",
    "client = MongoClient(mongo_uri)\n",
    "\n",
    "db = client['research_articles']\n",
    "collection = db['gut_microbiome']\n",
    "\n",
    "# Retrieve all documents from the collection\n",
    "papers = collection.find()\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<pymongo.synchronous.cursor.Cursor object at 0x00000231822E5750>\n",
      "{'_id': ObjectId('66f3fd777027ab88318f65d6'), 'pmc_id': 'PMC11417245', 'title': 'MBCN: A novel reference database for Effcient Metagenomic analysis of human gut microbiome', 'abstract': 'No Abstract Available', 'authors': ['Zheng B', 'Xu J', 'Zhang Y', 'Qin J', 'Yuan D', 'Fan T', 'Wu W', 'Chen Y', 'Jiang Y'], 'publication_date': '2024 Sep 6', 'journal_name': 'Heliyon', 'doi': '10.1016/j.heliyon.2024.e37422', 'keywords': [], 'score': 10, 'filled_sections_count': 0.8333333333333334, 'Introduction': 'IntroductionThe human gut microbiome,  which is now recognized as a complex ecosystem that plays a critical role in human health and disease,  has been the subject of extensive research in recent years.  Prokaryotic microorganisms perform a variety of functions,  such as breaking down complex carbohydrates,  producing vitamins,  and modulating the immune system,  so much so that most of the functions are unclear.  However,  it is difficult to study these microorganisms because most cannot be cultured by traditional methods.  Recent advances in DNA sequencing technology have enabled the sequencing of the entire genomes of these prokaryotic microorganisms,  which has led to a wealth of information on the genetic diversity and functional potential of the gut microbiome. Metagenomics approaches have been driven by exponential increases in sequencing throughput and associated declines in costs,  leading to its widespread adoption in clinical research.  However,  accurate classification of the composition of microbial community from metagenomic data remains a challenge due to their complexity,  the short-read lengths,  and incomplete genome reference database.  The reference database is crucial in metagenomic analysis as the quality of the data analysis using metagenomic classifiers relies heavily on the quality of the reference database.  Despite their importance,  most \\x80\\x9cstandard\\x80\\x9d reference databases of metagenomic classifiers may not be tailored to specific research questions,  such as the gut microbiome.  To address this,  initiatives like Metagenomics of the Human Intestinal Tract (MetaHIT),  the integrated gene catalog (IGC) have obtained prokaryotic microorganisms genomes from gut,  enabling their use in metagenomic studies.  Recent approaches that recover high-quality metagenomic assembled genomes (MAGs) from metagenomic datasets are addressing this limitation,  providing draft genomes of uncultured taxa.  Genome collections of human gut microbes have been built based on MAGs,  such as Unified Human Gut Genome (UHGG) and Humgut,  but these collections often use only one representative genome for a species,  leading to a large number of gene deletions.  Furthermore,  even for the same microorganism,  genomes from different environments usually have significant differences,  highlighting the need for a comprehensive and tailored reference database for accurate metagenomic analysis. The ability to combine samples from multiple studies using new bioinformatics tools has transformed our understanding of microbial diversity and its relationship with human health,  offering insights into the physiological states involved.  While incremental data can provide a more comprehensive understanding of the functional composition of the microbial world,  the inconsistent methodologies used in different studies make it difficult for non-specialist researchers to apply meta-analysis to metagenomics research.  To address this,  various online analysis platforms,  including Qiita,  KBase,  Galaxy,  have been developed to provide easy,  systematic,  and comprehensive statistical analysis,  interactive visualization,  and meta-analysis of microbiome data for researchers without specialized programming skills.  However,  some of these platforms require complex registration and data uploading processes,  which can be inconvenient for researchers who only need to perform simple analyses on small amounts of data.  Other platforms,  such as MicrobiomeAnalyst and MG-RAST,  allow direct analysis without registration,  but have limitations in terms of the types of analysis they can perform,  or only collect amplicon sequencing data for meta-analysis.  Overall,  the availability of user-friendly platforms for meta-analysis has facilitated access to valuable tools and insights for non-specialist researchers in the field of metagenomics. We develop a method for analyzing human gut metagenomic data that involves assembling a high-quality microbial reference database from recent human gut metagenomic datasets,  as well as genomes from UHGG and Refseq.  This reference database,  called the Microbiome Collection Navigator (MBCN),  uses the Genome Taxonomy Database (GTDB) for taxonomy classification,  which provides a quantitative and convenient operational species definition compared to the NCBI taxonomy.  To demonstrate the effectiveness of MBCN,  we compare it to several widely used human gut microbial genome collections and find that it had superior recall and precision,  as well as a higher proportion of assigned reads.  Additionally,  we have developed a web-based program called to meet the requirements of current metagenomic shotgun data analysis.  This web allows researchers to easily obtain generated profiles and metadata of human gut shotgun metagenomic data,  and integrates common statistical and visualization tools for exploratory analysis of metagenomic species abundance profiles and taxonomic features. ', 'Methods': \"Star methodsData collectionA text search using the keywords \\x80\\x9chuman gut microbiome metagenomics\\x80\\x9d is performed within BioProject database of NCBI to retrieve all relevant BioProjects.  Following this initial retrieval,  a manual screening process is meticulously performed to exclude unrelated BioProjects.  To further refine our selection,  the identified BioProjects are compared against those utilized in the construction by UHGG.  Through this rigorous comparison and elimination process,  14 BioProjects containing 2379 samples that haven't previously been assembled or utilized for such purposes are identified and selected.  All samples are downloaded as compressed express files using the Aspera download system (https://www. ibm. com/products/aspera). The main source comes from the recently published Unified Human Gut Genome (UHGG) collection (http://ftp. ebi. ac. uk/pub/databases/metagenomics/mgnify_genomes/human-gut/v2. 0/all_genomes/),  which contains genomes of 4744 species related to human gut.  Another source is RefSeq,  which can be obtained from ftp://ftp. ncbi. nlm. nih. gov/genomes/refseq/bacteria/and ftp://ftp. ncbi. nlm. nih. gov/genomes/refseq/archaea/.  We select some high-quality genomes from RefSeq according to a series of criteria to construct a reference database: 1) Completeness greater than 90 % and contamination less than 5 %.  2) N50 more than 20 kb.  3) Number of contigs less than 500.  4) Number of undetermined bases less than 10, 000.  5) Has a species name in GTDB R07-RS207 classification.  A total of 8761 high-quality genomes are obtained from RefSeq. Metagenomic assembly,  binning,  and quality assessmentThe quality control of sequencing reads belonging to 14 different BioProjects is performed using fastp (https://github. com/OpenGene/fastp) with the following options: \\x80\\x98-q 20 -u 30 -n 5 -y -Y 30 -l 90 --trim_poly_g\\x80\\x99.  For metagenomic assembly,  Megahit (https://github. com/voutcn/megahit) with the following options: \\x80\\x98--min-contig-len 200 --k-min 21 --k-max 121 --k-step 20\\x80\\x99 is used.  The reads are mapped back to each component using BWA-MEM (https://github. com/lh3/bwa) with default parameters.  The continuous depth is calculated by using jgi_summarize_bam_contig_depths.  Subsequently,  MetaWRAP (https://github. com/bxlab/metaWRAP) is used with default parameters and three different binning algorithms (metaBAT2,  MaxBin2,  CONCOCT) for metagenomic binning and refinement of each sample individually to obtain MAGs.  After quality assessment of MAGs using the CheckM (https://github. com/Ecogenomics/CheckM) lineage_wf workflow,  16, 785 high-quality MAGs with completeness >90 % and contamination <5 % are retained. We have made the code repository for the complete assembly workflow publicly available on the website (https://gitee. com/zjhzxjm/mbcn-flow).  This workflow enables users to assemble contigs and optionally further binning contigs into bins from metagenomic data and leverage the database for taxonomic profiling and abundance calculation.  The reproducible workflow can be utilized on new datasets,  customized to suit specific needs,  or built upon for novel approaches to microbiome characterization. Genome clustering and taxonomic assignmentGenome clustering is carried out using dRep (https://github. com/MrOlm/drep) with a 95 % genome distance threshold,  which is considered a rough estimate of species delineation.  Only genomes with completeness >90 % and contamination <5 % are retained and assigned a genome quality score (completeness - 5 Ã\\x97 contamination + 0. 5 Ã\\x97 log (N50)).  The genome with the highest quality score in each cluster is designated as the representative genome for taxonomic assignment. Taxonomic assignment of genomes is performed using the GTDB toolkit (GTDB-Tk,  version 07-RS207,  https://github. com/Ecogenomics/GTDBTk) with default parameters.  The GTDB classification system differs from that of the NCBI taxonomic database.  The NCBI taxonomic names corresponding to the GTDB taxonomic names of all genomes are obtained using the script. Pan-genome construction and build reference databaseAlthough representative genomes provide species-level resolution and have a small size,  they do not capture the intraspecific diversity present in microbial communities,  which contains a significant portion of genetic information.  To address this,  we combine the genes predicted by Prokka (https://github. com/tseemann/prokka) with default parameters of each cluster and use the pan-genome pipeline Roary (https://sanger-pathogens. github. io/Roary) to generate a non-redundant pan-genome based on clusters,  which integrate sequence information from all strains of the same species.  Roary is configured with the following options: \\x80\\x98-i 90 -cd 90 -s\\x80\\x99 (a minimum amino acid identity of 90 % for a positive match,  a core gene defined at 90 % presence,  and no paralog splitting). Microbial identification and abundance estimation from complex organisms or environmental samples have been a challenge in microbiology for a long time.  Various profiling methods have been proposed for this purpose,  based on sequence similarity to a reference database of previously characterized sequence data.  These methods can be divided into four groups based on how they establish sequence similarity: (i) nucleotide or protein alignment methods such as Centrifuge,  Kaiju and DIAMOND,  (ii) marker gene methods such as MetaPhlAn and mOTUs,  and (iii) k-mer based methods such as Kraken2 and Bracken.  With the improvement of computing devices,  k-mer based methods are gradually becoming the mainstream method due to their high computational efficiency,  despite their high memory requirements. As all profiling methods can report false positives,  manual filtering by researchers is required.  We build Kraken2 and its derivatives Bracken database with MBCN to profile metagenomic data,  providing good performance on a server with a large amount of memory.  Using the taxonomic name and tree information from GTDB,  we build the classification tree files and assign the GTDB taxonomy ID to each sequence of all pan-genomes.  All pan-genomes are combined to build the Kraken2 and Bracken database. We perform statistics on all assembled bins and select frequently occurring species to generate a condensed database MBCN_bac120 containing only 120 representative core gut microbial genomes.  A more compact database will be preferable in certain scenarios,  for example: For quick preliminary analysis where comprehensive characterization is not needed like validation and benchmarking of new analytical methodologies,  or conducting experiments on low-memory machines or portable devices where storage space and computing capacity is limited.  The condensed database retains genomes reflecting the core functional diversity of the gut microbiome while dramatically reducing the overall size.  By judiciously selecting representative genomes,  the key capabilities of the full database can be preserved in a streamlined package suitable for particular applications or accessibility constraints. Development of web programThe web interface is implemented in Shiny (version 1. 6. 0) in R (version 4. 0. 5) and designed in a TabLayout.  The \\x80\\x9cHome\\x80\\x9d tab provides a general introduction to the web program and guides the user through the web.  The \\x80\\x9cQuick Search\\x80\\x9d tab and \\x80\\x9cAdvanced Search\\x80\\x9d tab contain the relative abundance profile of species using Kraken2 database built by MBCN with corresponding metadata,  including simple or advanced filter options of metadata for users to select samples of interest.  The \\x80\\x9cTool\\x80\\x9d tab contains the download of the reference database built in this study.  Users can download the database built in this study to profile metagenomic sequencing data from the web to obtain the relative abundance profile of species.  The \\x80\\x9cAnalyze\\x80\\x9d tab provides users with online analysis capabilities.  The \\x80\\x9cHelp\\x80\\x9d tab provides assistance.  The application can be accessed at http://www. microbiota. cn. A set of data analysis and visualization tools is provided through a web interface to perform data analysis in a simple,  code-free manner.  Users can upload the abundance profile obtained using our database or any other tool and the corresponding sample metadata for visual online analysis,  including abundance display,  Venn diagram analysis,  alpha diversity,  beta diversity,  difference analysis,  environmental factor analysis,  network analysis.  Most analysis modules are developed based on the R package microeco (version 0. 3. 3). \", 'Results': \"ResultsHuman gut microbiome genome collectionIn this study,  human gut metagenomic sequencing data from 2379 samples collected worldwide from individuals of various ages are obtained from 14 different bio-projects (Supplementary Table S1).  We develop a workflow to construct a collection of human gut microbiome genomes and facilitate downstream analysis (Fig.  1): The latest human gut metagenomic sequencing data from NCBI is collected and a workflow is constructed for metagenomic assembly,  binning,  and quality assessment,  which is then used to assemble high-quality microbial reference MAGs.  Fastp is used for quality control.  Megahit assembled contigs,  and BWA-MEM mapped reads back.  MetaWRAP with three binning algorithms (metaBAT2,  MaxBin2,  CONCOCT) refined samples into MAGs.  CheckM assessed MAGs,  retaining 16, 785 high-quality MAGs,  contain 14, 166 clusters,  with 8673 representative genomes from Refseq,  4374 from UHGG,  and 1119 from our own assembled MAGs.  Subsequently,  we perform clustering by dRep and taxonomic assignment by GTDB-Tk on the obtained MAGs.  The majority of clusters (9995 out of 14, 166) comprising a single MAG.  The most diverse MBCN cluster is Escherichia coli,  which consist of 8314 different MAGs,  followed by 7511 MAGs related to Agathobacter rectalis.  Pan-genomes are constructed for each species using Prokka and Roary,  and utilized for building Kraken2 and Bracken databases.  These MAGs are integrated with publicly available human gut microbial genomes to create the most up-to-date human gut microbial genome collection,  named MBCN. Both GTDB and NCBI naming systems have limitations in accurately identifying species-level clusters,  leading to ambiguous taxonomic annotations.  This results in a disparity between the number of species-level clusters (14, 166) and the total cluster names (12, 450 in GTDB and 6318 in NCBI).  Notably,  many clusters share names,  including 2838 clusters share GTDB names with other clusters,  and 8587 clusters share NCBI names.  This problem is particularly pronounced in the case of Collinsella clusters,  with 160 clusters named after GTDB Collinsella species and 471 clusters named after NCBI Collinsella species. Fig.  2 depicts the distribution of 13, 853 bacterial and 313 archaeal species across the phylogenetic tree.  Additionally,  21 % of Firmicutes A's diversity comprises unassigned species-level GTDB clusters,  while other main gut phyla contribute less than 10 %.  Our assembly identified 191 representative genomes of unassigned GTDB clusters,  primarily from Firmicutes A (88 clusters) and Actinobacteriota (42 clusters).  Some phyla have few genomes,  but they are closely associated with health in our assembled MAGs.  Akkermansia is the only genus of Verrucomicrobiota in the human gut,  which has received extensive attention for its role in health and disease.  Only one unassigned representative genome of a Pyramidobacter cluster assembled by our workflow belonging to the Synergistota phylum is reported to be related to disease. Evaluating Kraken2 database performance by simulated readsThe performance of Kraken2 and Bracken databases built by MBCN,  UHGG and Humgut is evaluated using data containing simulated Illumina reads from 100 genomes.  These genomes come from 84 representative species that are previously used to compare metagenomic assembly algorithms and are available at http://www. bork. embl. de/\\x88¼mende/simulated_data. Kraken2 is initially used to assign reads,  and the unassigned reads accounted for 9. 03 % using the MBCN database,  61. 85 % using the UHGG database,  and 76. 81 % using the Humgut database.  All metagenomic classifiers report false positives for species that are not present,  resulting in thousands of low-abundance predictions that researchers need to filter.  Therefore,  Kraken2 profiling results are adjusted using Bracken at the species and genus levels and are compared with different minimum species abundance thresholds.  The performance of Kraken2 and Bracken databases built by MBCN,  UHGG,  and Humgut with different minimum species abundance thresholds (Fig.  3) show that MBCN has significantly fewer false positives than UHGG and Humgut at both species and genus levels.  Moreover,  the false discovery rate (FDR) of MBCN is much smaller than that of UHGG and Humgut.  By setting the detection limit at 0. 025 %,  most false positives are filtered out. To evaluate the performance of databases further quantitatively in predicting relative abundance of species the database,  we utilize real reads from a mock microbial community (https://github. com/LomanLab/mockcommunity) to assess the database.  This microbial community contain 8 bacteria and 2 yeasts,  and here we focus only on the relative abundances of the bacteria.  The performance of the MBCN,  UHGG,  and Humgut databases in predicting relative abundance of species is evaluated and compared with the results of MetaPhlAn,  as shown in Table 1.  MBCN,  UHGG,  and Humgut performs well in avoiding false positive species,  while MetaPhlAn erroneously identifies more false positives.  In terms of recall,  MBCN,  UHGG,  and MetaPhlAn all identified the 8 bacterial species,  while Humgut only identify 5 species.  For predicting relative abundances,  MBCN performs the best in terms of both L1 and L2 distances.  Overall,  MBCN demonstrate superior performance in quantitative evaluation using real reads from the mock microbial community,  excelling at avoiding false positives,  achieving full recall of species,  and most accurately predicting relative abundances based on L1 and L2 distance metrics.  UHGG performs well at avoiding false positives and recall respectively,  but are surpassed by MBCN for overall quantitative assessment.  In comparison,  Humgut has difficulty achieving full species recall,  and MetaPhlAn has worse performance avoiding false positives,  which lead in worse results in predicting abundances. Comparison of the different databases on human gut metagenomic samplesTo further investigate the performance of different databases in practical applications,  we utilize data from PRJNA453965,  a study that employs shotgun metagenomic analysis to compare gut microbial communities between breast cancer patients and healthy controls.  The study obtains 133 stool samples from premenopausal breast cancer patients (n = 18),  healthy premenopausal controls (n = 25),  postmenopausal breast cancer patients (n = 44),  and healthy postmenopausal controls (n = 46),  which are previously aligned with a reference catalog of the human gut microbiome (IGC) to obtain a relative abundance profile.  We compare the community profiles generated using Kraken2 databases built based on MBCN with those obtained using the IGC,  UHGG,  and Humgut databases.  To mitigate potential confounding factors,  all breast cancer patients include in the study are diagnosed by pathological examination at the Affiliated Tumor Hospital.  The healthy control group is recruited from individuals visiting the Medical Examination Center of the First Affiliated Hospital of Guangxi Medical University who are confirmed to be free of breast cancer upon examination.  Exclusionary criteria for all study participants are: presence of diarrhea,  diabetes,  ulcerative colitis,  Crohn's disease,  or other infectious diseases;  and use of antibiotics,  steroid hormones,  Chinese herbal medicines (oral,  intramuscular,  or intravenous),  or probiotics such as yogurt during the 3 months prior to fecal sample collection.  Additionally,  breast cancer patients do not receive chemotherapy,  radiation therapy,  or surgery prior to the collection of fecal samples for analysis. Since the community composition of the fecal samples is unknown,  to assess the community profiles produced by each metagenomic classifier,  other measurable aspects are evaluated.  The percentage of reads assigned to a species by MBCN account for 89. 77 ± 2. 21 %,  which is significantly higher than that of IGC (50. 59 ± 13. 45 %),  UHGG (72. 39 ± 4. 29 %) and Humgut (79. 03 ± 4. 68 %).  Even though Humgut has introduced genomes from Refseq,  it shows little improvement compared to UHGG.  MBCN's ability to identify more sequences than other databases can be attributed to the addition of many genomes from RefSeq and MAGs assembled that are not present in other reference databases.  Furthermore,  we assess the difference in gut microbes (with a mean relative abundance \\x89¥0. 025 %) between breast cancer patients and healthy controls compare with the results of mOTUs and MetaPhlAn.  Base on the results of the simulated communities,  species with an estimated mean relative abundance of <0. 025 % are removed.  The results of IGC,  UHGG,  Humgut,  mOTUs and MetaPhlAn do not demonstrate any significant differential species between premenopausal breast cancer patients and healthy premenopausal controls (Wilcoxon rank-sum test,  FDR-adjusted p-value (q-value) < 0. 05).  However,  the results obtained using MBCN show that six species are significantly different in premenopausal samples,  all of which are enriched in healthy controls (Table 2).  Notably,  Sutterella parvirubra and Parabacteroides gordonii are reported to be enriched in healthy individuals. MBCN has the potential to identify more differential species in postmenopausal breast cancer patients and healthy postmenopausal controls compared to IGC,  UHGG,  Humgut,  mOTUs and MetaPhlAn databases (Fig.  4).  Supplementary Table S2 shows the comparison of the differential species reported by these databases.  Supplementary Table S3-S8 include the details of differential species reported by each database.  Specifically,  MBCN identifies 35 differential species,  which is substantially higher than Humgut (16 species),  UHGG (15 species),  IGC (7 species),  mOTUs (7 species),  and MetaPhlAn (7 species).  It is important to acknowledge that due to inherent differences in the underlying algorithms and methodologies employed by these databases and tools,  direct comparisons of absolute numbers should be interpreted with caution.  Nevertheless,  MBCN's ability to detect a higher number of differential species suggests a potential increase in sensitivity or a broader taxonomic coverage.  All databases consistently reported a higher abundance of Escherichia coli in postmenopausal breast cancer patients,  while Lachnospira eligens was reported to be lower.  MBCN stands out by containing a larger number of genes to identify species not present in other databases,  with 3449 species for IGC,  4744 species for UHGG,  and 5170 species for Humgut.  This expanded gene content enables MBCN to report several novel differential species in postmenopausal breast cancer patients and healthy controls,  such as Enterobacter hormaechei,  Anaerostipes hadrus,  Phocaeicola species,  and Alistipes are associated with breast cancer or human health.  The higher number of reads assigned to most species in MBCN compared to other databases results in the identification of more differential species whose average relative abundance in other databases is below 0. 025 %,  such as F23-B02 sp000431075,  CAG-349 sp003539515,  Prevotella sp000436915.  After applying FDR correction,  some species,  including Megamonas funiformis,  CAG-41 sp900066215,  and CAG-882 sp003486385,  show significant differences in the Wilcoxon test (p-value <0. 05),  whereas others,  such as Eubacterium_R sp000434995,  Lachnospira sp003537285,  and Odoribacte splanchnicus,  do not reach statistical significance.  It is worth noting that not all species exhibit an increase in the number of assigned reads in MBCN results.  Increased reference sequences and variations in taxonomic assignments across species and genera can lead to the reassignment of reads.  For instance,  some reads initially assigned to Escherichia fergusonii in UHGG were reassigned to Escherichia coli in MBCN,  resulting in a reduction of Escherichia fergusonii's relative abundance below 0. 025 % in MBCN's output.  This underscores the importance of considering both the completeness and the specific taxonomic resolution of each database when interpreting results. Web programTo ensure consistency in the metagenomic abundance profiles provided on our web program,  we collect metadata from 1082 human gut metagenomic sequencing samples and generate profiles using the Kraken2 reference database built with our proprietary MBCN reference database.  We organize the metadata and profiles to create a profile database on our web program.  Meanwhile,  using MBCN as a reference database,  we also develop a unified,  standardized,  and systematic metagenomic analysis pipeline and platform,  named MicrobiotaCN (http://www. microbiota. cn) and common statistical and visualization tools for microbiome research are integrated into the web program.  Within the interface of our web program,  users can filter samples based on metadata and obtain a table of profiling results that can be downloaded or imported into subsequent statistical and visualization analyses. As an illustration of the online analysis capabilities of web program,  we utilize the analysis tool of web program to analyze the data from the aforementioned breast cancer study (PRJNA453965).  The previously obtained profile and metadata are uploaded to the analysis module of web program to showcase the utility of commonly used statistical and visualization tools for microbiome research on our web program. Data importIn the data input module,  users can submit four tables to perform online visual analysis.  The taxon or OTU abundance profile table and the metadata file containing the grouping information are necessary.  If the row names of the submitted abundance profile table meet the requirement for the seven-level taxonomy (e. g.,  \\x80\\x9ck__Bacteria;  p__Actinobacteria;  c__Actinobacteria;  o Bifidobacteriales;  f__Bifidobacteriaceae;  g__Bifidobacterium;  s__Bifidobacterium_longum\\x80\\x9d),  there is no need to submit the corresponding table (OUT to tax table) containing correspondence between the row names of the submitted abundance profile table and the seven-level taxonomy.  Metadata must be submitted containing the sample number in the first column consistent with the column names of the abundance profile table and the corresponding grouping table.  Factor metadata containing categorical metadata information is necessary,  while numeric metadata information is required only for environmental factor analysis.  Files can be uploaded as tab-delimited text (. txt) or comma-separated values (. csv).  Users can refer to the related FAQs and tutorials for more details or try our test examples. Species composition displayAfter data filtering,  this module can use histograms (Fig.  5-A),  box plots (Fig.  5-B),  pie charts (Fig.  5-C),  and heat maps (Fig.  5-D) to display the species abundance of each classification level,  and groups are displayed separately.  These four kinds of plots are used to display the phylum-level relative abundance of the PRJNA453965 project profiled using the Kraken2 database built by MBCN for postmenopausal breast cancer patients (Post + Case),  healthy postmenopausal controls (Post + Control),  premenopausal breast cancer patients (Pre + Case),  and healthy premenopausal controls (Pre + Control) separately (Fig.  5A\\x80\\x93D),  which indicate Bacteroidota dominates gut microbiota in all groups,  followed by Firmicutes_A. Community diversity analysisCommunity diversity analysis is mainly implemented based on the R vegan packages and is performed at different classification levels depending on the available annotations (Fig.  6-A).  The alpha-diversity analysis function currently supports five common diversity measures and can automatically estimate the corresponding statistical significance.  Beta-diversity analysis supports three common distance measures,  and results are presented based on principal component analysis () (Fig.  6-B),  principal coordinate analysis (PCoA),  or non-metric multidimensional scaling (NMDS).  We demonstrate the analysis of alpha-diversity and beta-diversity for the PRJNA453965 project (Fig.  6A and B),  which indicate no significant difference of alpha-diversity or beta-diversity between each two groups. Differential analysis and biomarker identificationThe module offers four methods for differential analysis to distinguish important features between different groups,  including Metastats,  Wilcoxon rank-sum test,  LEfSe,  and random forest software.  Metastats is the first statistical method that specifically targets questions arising in clinical research.  It analyzes metagenomic samples and identifies features that differentiate the two populations statistically (Fig.  7-A),  which indicates relative abundance of 10 species with the most significant differences between postmenopausal breast cancer patients (Post + Case) and postmenopausal healthy controls (Post + Control) from Metastats analysis.  Wilcoxon rank-sum test uses a volcano plot to show the results,  where statistical significance is combined with the magnitude of change to identify data with large and statistically significant changes point (Fig.  7-B),  which indicates fold difference and p-value of species enriched in two groups,  such as Sutterella parvirubra is enriched in postmenopausal breast cancer patients (Post + Case),  and Prevottela rara is enriched in postmenopausal healthy controls (Post + Control).  LEfSe uses the Kruskal-Wallis rank-sum test to detect features with significantly different abundances in different groups,  and then it performs linear discriminant analysis to determine the most likely characteristics that explain the differences between classes (Fig.  7-C),  which indicates LDA scores of significantly different clades between two group.  We analyze the relative abundance of significantly different clades in different populations and plotted taxonomic cladograms to demonstrate the taxonomic hierarchy distribution of significantly enriched marker species in each group of community samples (Fig.  7D and E).  For example,  Bacteroides fragilis is enriched in postmenopausal healthy controls (Post + Control),  while Bacteroidota,  Bacteroidia,  Bacteroidales are enriched in postmenopausal breast cancer patients (Post + Case).  Random forest is a non-parametric machine learning algorithm that predicts class based on the majority vote of the ensemble.  It weighs the importance of each feature and provides unbiased estimates of classification error.  MeanDecreaseAccuracy of random forest analysis indicates the importance of features on the left side of Fig.  7F,  corresponding relative abundance of species on the right side,  which indicates importance species in two groups,  such as Sutterella parvirubra in postmenopausal breast cancer patients (Post + Case),  and Prevottela rara in postmenopausal healthy controls (Post + Control). Other analysisAdditionally,  the web program offers other analysis modules to cater to various analysis requirements such as network analysis,  cluster analysis,  and environmental analysis.  The network analysis module enables users to calculate the correlation of different species and generate network diagrams that can be saved in HTML format and interacted with using a mouse.  This module provides two different graph types: chord graph and NetworkD3 dynamic network graph.  Users can display the taxonomic classification level and set the Spearman or Pearson correlation coefficient threshold to draw a network relationship diagram of the eligible taxonomic classification. The cluster analysis module allows users to choose the clustering method and distance for performing unsupervised clustering of samples.  The cluster index tab offers a function for selecting the number of clusters to use in the next cluster tab. The environmental analysis module performs Redundancy analysis (RDA) and Distance based redundancy analysis (db-RDA) on continuous environmental factors to analyze their impact on taxonomic abundance.  Permission must be obtained for use of copyrighted material from other sources (including the web).  Please note that it is compulsory to follow figure instructions. Comparison with other web programsSeveral web-based tools have been developed for microbiome data analysis.  Here we compare MicrobiotaCN with MicrobiomeAnalyst,  MicrobioSee and ImageGP.  Table 3 summarizes the main features of each tool. In terms of specific analysis functions,  all four platforms support species composition,  diversity,  and biomarker analysis.  Additionally,  MicrobiotaCN and MicrobiomeAnalyst enables clustering and correlation analysis.  For statistical comparisons and discovery,  MicrobiotaCN,  MicrobiomeAnalyst and MicrobioSee provide robust capabilities,  while ImageGP does not offer these functionalities.  Another strength of MicrobiotaCN is its integration with public data resources to enable meta-analysis,  a feature only matched by MicrobioSee among the other tools.  Overall,  MicrobiotaCN demonstrates the most comprehensive feature set.  A key advantage of MicrobiotaCN is its one-time data upload requirement for all downstream visualizations,  which greatly improves analysis efficiency compared to other tools like MicrobiomeAnalyst,  MicrobioSee and ImageGP that mandate repetitive data uploads. In summary,  MicrobiotaCN provides a comprehensive set of analysis and visualization capabilities for microbial community studies.  Its one-time data upload design significantly improves analysis efficiency.  These advantages make MicrobiotaCN a uniquely powerful platform for microbiome research. \", 'Discussion': 'DiscussionMBCN is a genome collection used for profiling reference database construction and can serve as a global reference for bacteria in the intestinal tract of healthy individuals.  The profiling reference database built by MBCN is available for download in web program and is more accurate and comprehensive than other intestinal microbial genome collections.  It utilizes an improved standardized assembly process to obtain high-quality MAGs,  and selects high-quality genomes from public databases to build a reference database,  significantly reducing genome contamination.  In addition,  MBCN innovatively uses pan-genomes to build a reference database instead of a single representative genome,  covering as much genetic information as possible for the species.  By introducing high-quality genomes from Refseq,  the variety of species included is significantly increased,  helping to reduce unclassified sequences.  In comparison to other studies,  our study has constructed a reference database and downstream metagenomic analysis,  and innovatively developed a web program for the entire metagenomic analysis pipeline based on the constructed genome collection,  providing a directly available reference database for profiling.  The web program provides online interactive analysis using the profiles obtained from a unified process for meta-analysis,  allowing for the comparison and analysis of profiles from different studies.  Moreover,  it provides a more comprehensive and innovative analysis module. The wealth of genetic information also poses some challenges,  with the size and resources required for the database being significantly increased.  A marker-based mOTUs database is built alongside the Kraken2 database for users with low computing resources.  However,  the results of different databases cannot be combined for analysis.  One challenge that remains is the naming of species in our genome collection,  as there is a profound inconsistency between the total number of species-level clusters and the number of annotated names.  To address this issue,  a higher-order taxonomy is used for clusters without precise taxonomy,  and the abundance of these clusters is not analyzed at the species level.  Files are prepared to build a custom Kraken2 database,  in which all MBCN clusters received artificial TAXIDs,  classified as clusters instead of classifications. MBCN database will be updated 1\\x80\\x932 times per year to maintain currency and incorporate new data.  Each update will include newly available gut metagenomic datasets that meet our criteria for quality,  completeness,  and relevance.  Priority will be given to large-scale datasets from diverse human populations.  New reference genome sequences relevant to the human gut microbiome will be added based on publication in public databases.  Newly developed bioinformatic pipelines and algorithms validated by the research community will be implemented to enhance the analytic capabilities of MBCN.  This regular update schedule will ensure MBCN remains a comprehensive,  cutting-edge resource for human gut metagenomic research and discovery.  User feedback will help guide enhancements incorporated through each update. ', 'Conclusion': 'ConclusionIn summary,  we create a novel collection of human gut prokaryotic genomes named Microbiome Collection Navigator (MBCN) and build reference databases for human gut metagenome research.  Using MBCN as a reference database,  we also develop a unified,  standardized,  and systematic metagenomic analysis pipeline and platform,  named MicrobiotaCN (http://www. microbiota. cn) and common statistical and visualization tools for microbiome research are integrated into the web program.  MBCN and MicrobiotaCN can be a valuable resource and a powerful tool that allows researchers to perform metagenomic analysis by a unified pipeline efficiently.  Our database primarily focuses on human gut microbiota to enhance the accuracy of metagenomic sequencing results and facilitate the exploration of the relationship between gut microbiota and diseases.  In the future,  we plan to expand our database to include other microorganisms.  Specifically,  as research reveals the colonization of oral microorganisms in the intestine,  we intend to incorporate oral microbiota genomes into our database to enable more comprehensive analysis of human microbiome data. ', 'References': []}\n",
      "<pymongo.synchronous.cursor.Cursor object at 0x00000231822E5750>\n",
      "{'_id': ObjectId('66f3fd777027ab88318f65d8'), 'pmc_id': 'PMC11414306', 'title': 'Relationship between the rumen microbiome and liver transcriptome in beef cattle divergent for feed efficiency', 'abstract': 'No Abstract Available', 'authors': ['Keogh K', 'Kenny DA', 'Alexandre PA', 'Waters SM', 'McGovern E', 'McGee M', 'Reverter A'], 'publication_date': '2024 Sep 20', 'journal_name': 'Anim Microbiome', 'doi': '10.1186/s42523-024-00337-0', 'keywords': [], 'score': 10, 'filled_sections_count': 0.8333333333333334, 'Introduction': 'BackgroundWithin beef production systems,  feed costs alone account for up to 75% of the total variable cost of production,  thus impacting overall profitability.  Therefore,  selecting and breeding cattle that are more feed efficient,  for example through residual feed intake (RFI) phenotype,  may increase production profitability by reducing feed input costs,  whilst also contributing to the environmental sustainability of beef production.  Moreover,  feed intake (measured as average daily dry matter intake,  DMI) and growth rate (measured as average daily gain,  ADG) are important traits within the beef industry due to their direct and indirect effects on overall productivity and consequently profitability and sustainability.  Thus,  it is of interest for beef producers to rear faster growing animals with reduced dietary intake to optimise productivity of production systems.  Indeed,  the aforementioned traits (DMI,  ADG and RFI) have been shown to be moderately heritable,  thus offering an opportunity to reduce feed costs through breeding cattle that are more feed efficient through genomic selection breeding programs.  However,  despite the clear benefit of breeding cattle that are feed efficient,  evaluations of the molecular control regulating feed efficiency in beef cattle are not conclusive,  such that key genes or genomic regions contributing to the trait are yet to be identified.  The contrasting results from molecular and genomic based studies across the literature are undoubtedly due to the multifaceted nature of the feed efficiency trait as well as the various confounding experimental parameters employed across studies such as breed,  dietary management system and stage of development during which feed efficiency was evaluated.  If reliable genomic selection processes are to be implemented for feed efficiency in beef cattle it is essential that genomic regions contributing to the trait are reliable across these various confounding factors. The availability of nutrients for both growth and maintenance purposes in ruminants is dependent on the functionality of the rumen microbiome.  Ruminants rely on the complement of bacteria,  archaea and protozoa,  amongst others,  in the rumen microbiota for the degradation of feed and the production of microbial protein,  vitamins and volatile fatty acids (VFAs),  the latter of which provide up to 70% of the host\\x80\\x99s energy requirements.  Research from our own group as well as others has identified links between the rumen microbiome and feed efficiency phenotype.  Moreover,  Shabat et al.  reported that the rumen microbiome could predict variation in an animal\\x80\\x99s feed efficiency phenotype and concluded that reduced rumen microbial community diversity may support a more feed efficient animal.  However,  similar to the identification of genomic regions associated with feed efficiency in beef cattle,  results related to the contribution of the rumen microbiome to the feed efficiency phenotype are inconsistent,  again,  most likely due to the aforementioned confounding experimental parameters employed.  Thus,  it is important that such factors are considered and examined together.  Determination of the relationship between the rumen microbiome and host feed efficiency phenotype has the potential to not only facilitate the selection of cattle with enhanced nutrient utilisation,  but also to enable the manipulation of the rumen microbiome to enhance its energy harvesting capacity. The VFAs produced within the rumen following microbial degradation of feed are responsible for a large proportion of the host\\x80\\x99s energy requirements.  Furthermore,  the liver,  a highly metabolically active organ,  has been shown to be responsive to dietary intake as well as being affected by feed efficiency phenotype.  Thus,  the objective of this study was to evaluate the interactions between the rumen microbiome,  VFA concentrations and hepatic gene expression profiles with three production and efficiency traits of interest,  namely DMI,  RFI and ADG,  through a network based systems biology analysis.  In order to account for the confounding effects of both breed and dietary source,  data utilised in this study (rumen microbiome,  liver transcriptome,  VFA and phenotype data) were derived from two contrasting breed types (Charolais and Holstein-Friesian) divergent for RFI across contrasting dietary phases (high concentrate and zero-grazed grass). ', 'Methods': 'MethodsAnimal management and phenotype collectionThe animal model utilised in this study was conducted as part of a larger research programme designed to investigate the within-animal repeatability of feed intake,  growth and feed efficiency in two contrasting breeds (Charolais and Holstein-Friesian) of beef steers,  which were offered contrasting diets over separate dietary intake test periods.  Details related to the original animal model are described previously in Higgins et al.  and are only briefly outlined here.  Charolais (n\\x80\\x89=\\x80\\x8990) and Holstein-Friesian (n\\x80\\x89=\\x80\\x8977) steers were offered contrasting diets over different stages of development as follows: (i) a high-concentrate diet during the growing phase;  (ii) zero-grazed grass diet during the growing phase and;  (iii) high-concentrate diet during the finishing phase.  Charolais steers were on average 373 (±\\x80\\x8918) days of age and weighed 485 (±\\x80\\x8938) kg,  whilst Holstein-Friesian steers were 399 (±\\x80\\x897. 6) days of age and weighed 401 (±\\x80\\x8943. 3) kg at the start of the trial.  Upon completion of a dietary adaptation period,  lasting 14 days,  individual animal intakes were recorded (using an electronic Calan gate system;  American Calan Inc.,  Northwood,  NH,  USA) over the three feeding phases,  which each lasted for 70 days.  Steers were weighed at the beginning and end of each dietary phase as well as on a fortnightly basis throughout.  All steers were offered the same concentrate diet ad libitum during each of the two high-concentrate phases,  with a restricted allowance of grass silage also provided.  For the interim zero-grazed grass phase,  steers were individually offered fresh herbage,  harvested twice daily from Lolium perenne dominant swards,  ad libitum.  All steers had unrestricted access to fresh,  clean drinking water.  Upon completion of each dietary phase,  individual RFI values were determined within breed for all steers as previously described in Higgins et al.,  and animals were ranked as either High-RFI or Low-RFI,  selecting the highest High-RFI and lowest Low-RFI for subsequent analyses. Rumen digesta sampling and sequencingFull details related to rumen digesta sampling and subsequent sequencing analysis are described in full in McGovern et al..  At the midpoint of each dietary phase,  following a dietary adaptation period,  a single rumen fluid sample was collected from all steers via stomach intubation (Flora Rumen Scoop,  Profs-Products,  Guelph,  Canada).  All samples were harvested approximately 2\\x80\\x934 h post-feeding,  and were immediately snap frozen in liquid nitrogen and subsequently stored at -80 °C,  pending further analysis.  Ten rumen digesta samples per High- and Low-RFI groups from each breed and diet were used for subsequent microbial DNA isolation,  with the exception of the Low-RFI Charolais and High-RFI Holstein-Friesian steers during the zero-grazed grass diet and the Low-RFI Holstein Friesian steers during the second high-concentrate diet,  where only 9 samples were available for each.  Frozen rumen liquid samples (20 g) was homogenised to a fine powder under liquid nitrogen using a pestle and mortar and stored at -80 °C.  Approximately 250 mg of the homogenised frozen powder was then used for DNA isolation,  which was undertaken using the repeated bead beating and column purification method.  The quality of the resultant DNA samples was assessed on an agarose gel,  with DNA yield and purity also assessed on a Nanodrop 1000 spectrophotometer.  Amplicon libraries were prepared through PCR amplification,  targeting the V4 region of the 16 S rRNA gene in both bacteria and archaea.  Full details of library preparation are previously outlined in McGovern et al..  Amplicon generation was validated through visualisation on an agarose gel.  Amplicons were pooled in equal concentrations and gel purified to remove unwanted products using the Qiagen Gel Extraction Kit (Qiagen,  Manchester,  UK).  The pooled purified libraries were measured for purity and quantity on the Nanodrop 1000 spectrophotometer and further quantified using the KAPA SYBR FAST universal kit with Illumina Primer Premix (Roche Diagnostics,  West Sussex,  UK).  The library pool was then diluted and denatured according to the Illumina MiSeq library preparation guide.  The sequencing was conducted using 500 cycle MiSeq reagent kits (Illumins,  San Diego,  CA,  USA).  The sequencing reads generated were imported into Qiime2,  where the DADA2 pipeline was used for the detection of operational taxonomic units.  Taxonomy was assigned using a naÃ¯ve Bayes classifier trained on the RefSeq database.  Sequence read files associated with this analysis are available through the NCBI Sequence Read Archive (Accession no.  PRJNA483745). Rumen fermentation profilingThe concentration of ruminal VFA composition was measured using a gas chromatograph (model 3800 Varian gas chromatograph) as per McGovern et al..  The concentration of the following acids was determined: acetic,  propionic,  isobutyric,  butyric,  isovaleric,  valeric,  total short chain fatty acids,  acetic:propionic acid ratio;  D-lactic acid;  L-lactic acid;  DL-lactic acid;  g-lactic acid and ammonia (NH3). Liver tissue sample collection and RNA-sequencingLiver tissue sample collection and RNA-sequencing methodology is described in full in Higgins et al..  Briefly,  at the end of each dietary phase and within breed,  steers were ranked as either High-RFI (feed-inefficient;  n\\x80\\x89=\\x80\\x8912) or Low-RFI (feed-efficient;  n\\x80\\x89=\\x80\\x8912) and were subsequently used for the collection of liver tissue biopsies.  All steers selected for biopsy collection were administered a local anaesthetic (5 ml;  Adrenacaine,  Norbrook Laboratories,  Ireland Ltd. ) to the biopsy site location.  Following anaesthetisation,  liver tissue was harvested through percutaneous punch between the 11th and 12th ribs as previously described by McCarthy et al..  Care was taken to ensure that all samples were consistently harvested from the same location from each animal.  All instruments used for biopsy collection were sterilised,  washed with 70% ethanol and treated with RNaseZap (Ambion,  Applera Ireland,  Dublin,  Ireland),  prior to use.  Following collection,  all tissue samples were washed with sterile DPBS and immediately snap frozen in liquid nitrogen before subsequent storage at -80 °C pending further processing. Full details related to RNA isolation and subsequent RNA sequencing and bioinformatic analysis are provided in Higgins et al.  and are only summarised here.  Briefly,  50 mg of liver tissue from each biopsy sample was used for the isolation of total RNA.  RNA was isolated from tissue samples in 3 ml of QIAzol reagent using a rotor-stator tissue lyser (Qiagen,  UK).  RNA was subsequently precipitated and purified using the RNeasy plus Universal kit (Qiagen,  UK) according to the manufacturer\\x80\\x99s instructions.  Quality and quantity of RNA isolated were determined using the RNA 6000 Nano Lab Chip kit (Aglient Technologies Ireland Ltd.,  Dublin,  Ireland) on an Aglient Bioanalyser 2100 and using a Nanodrop spectrophotometer (Nanodrop Technologies,  Wilmington,  DE,  USA),  respectively.  All RNA samples displayed RNA integrity numbers (RIN) greater than 8 and thus were deemed to be of suitable quality for subsequent RNA-sequencing.  Individual cDNA libraries were prepared from each separate liver RNA sample for cattle divergent for RFI across each breed and dietary phase,  using the Illumina TruSeq stranded mRNA sample prep kit (Illumina,  San Diego,  CA,  USA) according to the manufacturer\\x80\\x99s instructions.  Resultant cDNA libraries were validated using the DNA 1000 Nano Lab Chip kit on the Aglient Bioanalyser 2100.  Sequencing was subsequently undertaken on an Illumina HiSeq 2500 sequencer.  All sequencing data used in this study are publicly available in NCBI\\x80\\x99s Gene Expression Omnibus and can be accessed through GEO ID GSE111464. Quality control of sequencing reads was undertaken using FastQC (v 0. 11. 5),  followed by removal of sequencing adapters and any low quality reads using Cutadapt (v 1. 13).  Trimmed sequencing reads were mapped to the bovine reference genome (ARS-UCD1. 2) and also quantified using STAR (v. 2. 5. 1).  Differential expression was undertaken using the edgeR package within the R environment.  Within edgeR,  gene expression reads were estimated as Counts Per Million (CPM) and genes which presented with at least 1 CPM in at least half of the samples were retained for differential expression analysis.  Differentially expressed genes were identified for each of the main contrasts of RFI phenotype (Low-RFI versus High-RFI),  breed (Charolais versus Holstein-Friesian) and dietary source (high concentrate versus zero-grazed grass).  The model for differential expression due to a given contrast (e. g.,  RFI) contained the other two contrasts as main effects (e. g.,  breed and diet). Co-expression network analysisThe following datasets were utilised for the gene co-expression network analysis: liver transcriptomics;  16 S rumen microbial abundance taxonomy;  VFA concentrations;  and phenotype data (DMI,  RFI and ADG;  Fig.  7).  From the differential expression analysis conducted in edgeR,  the top 5% differentially expressed genes,  for each of the three contrasts (RFI,  breed,  and diet) based on corrected p-value were selected for subsequent inclusion in the co-expression network analysis.  Thus,  nodes selected for subsequent co-expression analysis included: (i) genes differentially expressed based on the RFI contrast;  (ii) genes differentially expressed based on the breed contrast;  (iii) genes differentially expressed based on the diet contrast;  (iv) ruminal microbial abundance;  (v) VFA concentrations and;  (vi) three phenotypes of interest: RFI,  DMI and ADG.  Significant connections (edges) between nodes were identified using the Partial Correlation and Information Theory (PCIT) algorithm.  The PCIT algorithm determines the significance of the correlation between a pair of nodes after accounting for all other nodes within the network.  The resultant network of co-expressed genes was imported into Cytoscape software for visualisation.  In order to assign biological annotation of the generated network,  co-expression network analysis results were further evaluated for functional enrichment using Gprofiler and David gene ontology. Stepwise regression and correlation analysesTo determine the independent rumen microbial predictors of RFI,  DMI and ADG,  a stepwise regression analysis was conducted.  This analysis was undertaken in SAS (version 9. 4) using the REG procedure,  incorporating the three phenotypes examined in this study as well as any microbe directly connected to the phenotypes,  based on the results from the co-expression network analysis within the model and utilising R2 as the selection criteria.  A correlation analysis was undertaken on the aforementioned phenotype and microbe variables using the CORR procedure of SAS.  Additionally,  a hierarchical clustering analysis was performed between the phenotypes and first neighbour microbial connections using PermutMatrix (version 1. 9. 4;  http://www. atgc-montpellier. fr/permutmatrix/, ). ', 'Results': 'ResultsAnimal performanceDescriptive results pertaining to growth,  dietary intake and RFI values are outlined in full in Higgins et al..  Groups selected as divergent for RFI were significantly different from one another (P\\x80\\x89<\\x80\\x890. 05).  Across all dietary phases for each breed,  High-RFI steers consumed more feed on average than their Low-RFI counterparts (P\\x80\\x89<\\x80\\x890. 001),  whilst there was no difference (P\\x80\\x89>\\x80\\x890. 05) in ADG across groups for each breed and each dietary phase. Microbial sequencing analysisComplete results related to microbial sequencing are presented in full in McGovern et al..  Briefly an average of 272, 460 (±\\x80\\x8969, 596) reads were generated from the rumen fluid samples.  Following merging of sequences and quality filtering,  an average of 217, 817 (±\\x80\\x8955, 519) reads were retained.  The average number of counts per sample that were assigned to an open taxonomic unit (post filtering) was 175, 304\\x80\\x89±\\x80\\x8974, 272.  Microbial taxa identified within the rumen fluid samples and utilised within the current study are presented in Additional Table 1. Liver gene expressionFor the RFI,  breed and diet contrasts,  12, 161,  12, 114 and 12, 581 genes,  respectively,  were identified as expressed of which 608,  605 and 629 were classified as differentially expressed and used for subsequent co-expression network analysis.  Of these differentially expressed genes,  only 2 were common across the three contrasts,  namely SPP1,  which encodes a cytokine and ABHD2 which encodes an acylglycerol lipase protein.  Genes identified as differentially expressed and included in the co-expression network analysis are presented in Additional Table 2. Co-expression network analysisOf the 2, 017 nodes (1, 842 differentially expressed genes,  159 microbial taxa,  13 VFA and 3 phenotypes) used for network analysis,  1, 534 displayed significant correlations,  resulting in a total of 35, 549 significant (P\\x80\\x89<\\x80\\x890. 05) connections between nodes (Fig.  1).  Of the three phenotypes examined,  DMI had the highest number of first neighbour connections,  followed by RFI,  with ADG displaying the fewest number of first neighbour connections (Table 1).  Connections between hepatic genes accounted for the highest proportion of first neighbour connections for each of the three phenotypes,  followed by microbial interactions,  and VFA connections (Table 2).  The co-expression network pertaining specifically to direct connections of the DMI,  RFI and ADG phenotypes is presented in Fig.  2 and a comparison of nodes directly connected to DMI,  RFI and ADG is provided in Fig.  3 and Additional Table 3.  Direct connections between the phenotypes and microbial taxa revealed a lack of commonality between DMI and RFI,  as well as DMI and ADG,  with only the Succiniclasticum genus negatively connected to both RFI and ADG (Fig.  4).  Details related to interactions of microbial taxa directly connected to DMI,  RFI and ADG are presented in Table 3,  along with enriched (P\\x80\\x89<\\x80\\x890. 05) gene ontology and pathway analysis results.  Full details related to interactions of microbial taxa connected to DMI,  RFI and ADG are presented in full in Additional Table 4. Stepwise regression and correlation analysesWhen each phenotype and their associated microbial connections were analysed separately through the regression analysis,  83% of the variation was accounted for through the Pyramidobacter and Succiniclasticum genera for the RFI phenotype.  Similarly,  the Succiniclasticum genus,  together with the CW040 order explained 73% within the ADG trait,  whilst the Butyrivibrio and Prevotella genera,  together with S. copri sp.  and R. faecis sp.  explained 89. 9% of the variation within the DMI trait.  When all microbes directly connected to the phenotypes examined were analysed together,  90% of the variation for RFI,  DMI and ADG was explained by the following microbial taxa: the Planctomycetes phylum;  CW040 order;  the Prevotella,  Roseburia,  Succiniclasticum,  Methanobrevibacter,  Pyramidobacter,  Butyrivibrio genera and;  S. copri sp.,  R. faecis sp.,  L. ruminis sp.,  C. aminophilum sp.  and D. D168 sp.  (Table 4).  Of the significantly correlated microbes,  only the Succiniclasticum genus and R. faecis sp.  were commonly contributing to variation across the three traits examined. Results of the correlation analysis undertaken in SAS are presented in Fig.  5.  Correlation results followed the same pattern as per the co-expression network results for RFI (Fig.  4),  with negative correlations (P\\x80\\x89<\\x80\\x890. 05) apparent between RFI and the Succiniclasticum and Pyramidobacter genera,  the Synergistetes and Planctomycetes phyla,  Planctomycetia and Synergistia classes and Synergistales and Pirellulales orders.  In addition to the positive connections evident through the co-expression network analysis,  through the CORR procedure in SAS,  negative associations (P\\x80\\x89<\\x80\\x890. 05) were also apparent between DMI and the Succiniclasticum and Methanobrevibacter genera and the CW040 order.  Similar to both RFI and DMI,  a negative association (P\\x80\\x89<\\x80\\x890. 05) was also apparent between the Succiniclasticum genus and ADG,  representing the only significant association between ADG and the microbial taxa from the correlation analysis. Hierarchical clustering of the phenotypes and first neighbour microbial connections is presented in Fig.  6.  From Fig.  6,  it is evident that the RFI and DMI phenotypes were clustered with the Roseburia and Butyrivibrio genera,  as well as with L. Salivarius sp.  and R. Faecis sp.  The ADG phenotype was separately clustered together with the Lachnobacterium and Prevotella genera,  whilst also clustered with S. Copri sp.,  C. Aminophilium sp.  and L. Ruminis sp.  Hierarchical clustering also revealed a relationship between members of the same clade,  for example the Pyramidobacter genus,  Synergistetes phylum,  Synergistia class and Synergistales order were all clustered together,  with the Planctomycetia class,  Planctomycetes phylum and Pirellulales order also clustered together separately. ', 'Discussion': 'DiscussionWithin the published literature,  biological processes related to immune function and lipid metabolism have been attributed to the underlying biology governing the RFI phenotype.  Indeed,  results from the current study further corroborate that finding,  where lipid metabolism related genes pertaining to the RFI differential expression contrast were connected to the RFI phenotype,  whilst also connected to both DMI and ADG too.  For example,  RFI was positively connected to CYP26B1,  whilst also negatively related to both PDK4 and OSBPL11.  Indeed,  all three of these differentially expressed genes were pertaining to the RFI contrast,  whilst CYP26B1 was related to both RFI and diet contrasts,  indicating a role for this gene in mediating the intersection of RFI with diet.  CYP26B1 encodes a member of the cytochrome P450 superfamily which are responsible for catalysing reactions involved in the synthesis of cholesterol,  steroids and other lipids.  Additionally,  CYP26A1,  which functions similarly to CYP26B1 was also positively connected to the DMI phenotype.  OSBPL11 encodes an intracellular lipid receptor,  which plays a role in regulating ADIPOQ and FABP4 levels in differentiating adipocytes and is also involved in regulating adipocyte triglyceride storage.  Whilst PDK4 encodes a mitochondrial protein which functions in the regulation of both glucose and fatty acid metabolism.  The relevance of the PDK4 gene towards the RFI phenotype is further apparent through previous reports of this gene in relation to RFI,  specifically PDK4 was down-regulated in skeletal muscle tissue of efficient Holstein-Friesian bulls following a high-concentrate finishing diet,  as well as in the liver tissue of efficient Charolais steers.  Of the lipid related genes connected to the DMI phenotype,  all were pertaining specifically to the RFI differential expression contrast,  whilst also being directly connected to the DMI phenotype,  suggesting a role for these genes towards both RFI and DMI phenotypes.  Moreover,  a number of these genes have previously been implicated with variation in RFI,  including ASCL1;  APOA1,  ELOVL2 and FADS1.  The FADS1 gene,  which encodes a desaturase enzyme and regulates the unsaturation of fatty acids was also connected to ADG.  Interestingly of the lipid metabolism genes connected to ADG (HMGCS1,  FDFT1,  SQLE and FADS1),  all were pertaining to the diet differential expression contrast,  with the exception of FADS1,  highlighting the influence of contrasting diets with growth rates in beef cattle. A role for altered immune response towards the RFI phenotype across various tissues throughout the body including the liver,  has already been established.  Interestingly within the current study we did not observe such a relationship between the RFI phenotype and immune response genes,  with the exception of,  C1QTNF7 which was negatively connected to RFI.  However,  although not strongly apparent within the RFI phenotype,  a more pronounced relationship of immune response genes was evident with the DMI phenotype.  This was apparent through connections between DMI and the following genes: A2M,  C1QBP,  C7,  CCL21,  CD19,  CD22,  CRP,  FCER2,  FCRL5,  FNDC4,  HMCN1,  IFNLR1 and IRF2BP2.  Indeed,  A2M,  CCL21 and CRP were previously reported as differentially expressed between cattle divergent for RFI,  suggesting the importance of these genes to both the RFI and DMI traits.  The relationship between the DMI phenotype and immune response is further established through the connection between the Prevotella genus and S. copri sp.  (Segatella copri sp.  formerly Prevotella copri sp. ) of bacteria within the rumen microbiome,  both of which have been implicated in immune function.  For example,  high abundance of P. copri sp.  was correlated with increased concentrations of serum metabolites related to chronic inflammation in the gut of pigs in the data of Chen et al..  Whilst the Prevotella genus was reported to be associated with gut mucosal inflammation in mice.  Moreover,  a positive relationship was apparent in the current study between the Prevotella genus and DMI,  whilst Jiang et al.  and Carberry et al.  both conversely reported negative correlations between relative abundance of ruminal Prevotella genus.  Moreover,  both Prevotella genus and S. copri sp.  have been related to RFI within various species too.  For example,  Jiang et al.  reported that P. copri sp.  was one of the most abundant microbes in low-feed efficiency pigs and also noted the importance of that particular species to feed efficiency during daily-phase feeding strategies in pigs.  Furthermore,  in cattle,  Brooke et al.  suggested that P. copri sp.  may be a potential microbial marker for the identification of cattle with improved feed efficiency in their life-span and in the production cycle.  Additionally,  Carberry et al.  reported an effect of RFI phenotype on ruminal abundance of Prevotella genus,  irrespective of the varied diets offered in that study,  but did note greater abundance of Prevotella genus when cattle were offered a low forage diet compared to a high forage diet.  Conversely though,  Lopes et al.  reported that the correlation between the Prevotella genus and feed efficiency was dependent on the diet offered as well as the specific Prevotella spp.  Indeed,  different Prevotella spp.  have been associated with both higher and lower feed efficiency in cattle and sheep.  Furthermore,  Zhou et al.  reported that the Prevotella genus was the most abundant genus in both rumen content-associated and epithelial tissue-attached bacterial communities suggesting a role for this genus in serving as a marker for host RFI classification.  Additionally,  Jewell et al.  reported that the abundance of specific Prevotella spp.  is host specific,  with Yang et al.  suggested that Prevotella may be a key microbe increasing host feed intake,  suggesting that Prevotella spp.  could promote the host\\x80\\x99s appetite and decrease feed efficiency.  Overall,  results indicate that the role of varied immune response in relation to RFI may also be attributed to variation in DMI. Despite the known relationship between RFI and DMI,  namely RFI being computed after accounting for DMI,  results from the current study highlight differential connections between these two phenotypes and ruminal microbiome taxa,  whereby microbial taxa directly connected to RFI were different to those directly connected to DMI.  For the DMI phenotype all direct connections between DMI and microbial taxa were positive,  with interactions amongst these specific taxa also positive.  Conversely though,  for the RFI trait,  all connections between RFI and microbial taxa were negative,  whilst connections between the RFI taxa were positive.  Overall suggesting differential microbial responses,  despite the relationship between the two traits. The Butyrivibrio genus,  which was positively related to DMI in the current study,  is involved in a number of ruminal functions in addition to butyrate production including fibre degradation,  protein breakdown,  biohydrogenation of lipids and the production of microbial inhibitors.  Of particular importance to ruminant digestion,  and therefore productivity is the contribution of this bacteria to the degradation of plant structural carbohydrates,  principally hemicellulose.  Together with the Prevotella genus,  both Butyrivibrio and Prevotella genera are among the most abundant bacteria found within the rumen and have important functions in the metabolism of proteins and peptides.  Both of these genera breakdown protein and carbohydrates in feed,  undertake de novo peptide synthesis and use products of cellulose degradation from other cellulotyic bacteria as an energy source.  In addition to being related to DMI,  hierarchical clustering analysis revealed a relationship between the Butyrivibrio genus with RFI.  Indeed,  within the context of feed efficiency,  Myer et al.  and McGovern et al.  reported greater abundance of the Butyrivibrio genus in efficient steers and a negative correlation with RFI,  respectively.  Conversely,  Jewell et al.  reported greater abundance in the rumen of High-RFI dairy cows,  indicating a positive relationship between RFI and rumen Butyrivibrio genus.  The relative importance of the Butyrivibrio genus towards feed efficiency is also apparent in monogastrics,  specifically the Butyrivibrio genus was strongly correlated with feed efficiency in pigs during the weaning phase,  which the authors attributed to potentially be due to an enhanced ability to ferment complex carbohydrates.  Similarly,  Kubasova et al.  also reported greater abundance of Butyrivibrio genus in fecal samples of pigs.  Although the Butyrivibrio genus is a primarily butyrate producing bacteria,  results from the current interaction study reported direct connections with only the lactic acid VFAs,  which in turn were connected to hepatic genes involved in functions related to gene expression,  lipid metabolism,  growth,  as well as protease activity,  which given the function of the Butyrivibrio genus towards protein breakdown is of interest.  Specifically,  genes involved in the regulation of peptidase activity (PI16) as well as those with an associated protease function (CAPN6,  SERPINA1,  SERPINA3-1) were connected to lactic acid VFAs within the rumen.  Moreover,  of these,  PI16,  SERPINA1 and SERPINA3 were previously reported as differentially expressed in the RFI based datasets of Keogh et al.,  Weber et al.,  and Alexandre et al..  The lactic acid VFAs were also connected to the leptin receptor gene (LEPR) highlighting the role of leptin towards mediating satiety status within the body.  Moreover,  the LEPR gene was differentially expressed through the breed contrast,  highlighting differential hepatic expression of this gene between Charolais and Holstein-Friesian steers.  Additionally,  genes involved in TGF-beta growth signaling (BMP2,  CRM1A,  FNDC4 and HMCN1) were also connected to ruminal lactic acid concentrations,  which was of interest as Alexandre et al.  previously reported TGFB1 as a key regulator for feed efficiency in skeletal muscle of Nellore cattle. Similar to the Butyrivibrio genus,  the R. faecis sp.  is also a primarily butyrate producing bacteria.  Indeed,  although correlated with DMI in the current study,  as well as clustered with both DMI and RFI through the hierarchical clustering analysis,  R. faecis sp.  was previously associated with growth rate in pigs,  with a greater abundance of this microbe in pigs supplemented with a carbohydrate complex diet,  which the authors of that study suggested could have accounted for the improved feed efficiency observed in those animals.  However,  through the regression analysis in the current study,  R. faecis sp.  was identified,  together with the Succiniclasticum genus,  as playing a role in determining variation in the RFI,  DMI and ADG traits examined in this study.  Volatile fatty acids connected to R. faecis sp.  (lactic acid,  ammonia,  valeric acid,  acetic:propionic,  isobutyric acid and total-SCFA),  were also connected to hepatic genes involved in processes related to gene expression,  growth,  immune and lipid metabolism.  In the current study,  the Roseburia genus was positively associated with DMI,  this microbe utilises carbohydrates for growth and its abundance is known to increase with greater proportions of concentrates within the diet.  Similarly,  Li et al.  and Ellison et al.  reported greater abundance of the Roseburia genus in cattle fed a high-energy diet and lambs fed a concentrate diet,  respectively,  indicating a role for this bacterium depending on diet composition.  Additionally through its connections with isobutyric acid,  the R. faecis sp.  was connected to CMKLR2 and ND6 which encode proteins involved in adipokinetic hormone activity and glucose homeostasis as well as a mitochondrial gene,  respectively,  potentially suggesting a role for R. faecis sp.  in mediating hepatic glucose homeostasis and mitochondrial function as a consequence of dietary intake.  Moreover,  CMKLR2 was specifically pertaining to the diet contrast,  whilst ND6 was differentially expressed in both diet and RFI contrasts,  overall highlighting the effect of diet on these genes related to R. faecis sp.  abundance.  Whilst a role for these genes related to RFI is already established through their differential expression within the published literature. The Synergistetes phylum was negatively connected with RFI in the current study.  Similarly,  McLoughlin et al.  also reported a negative correlation between this phylum and feed efficiency in the solid rumen fraction of sheep.  Additionally,  McCormack et al.  also reported a role for this microbe towards variation in RFI in pigs.  In addition to the negative relationship between the Synergistetes phylum and RFI;  class,  order and genus members of this clade were also negatively connected to RFI.  Moreover,  the members of this specific phylum were all positively connected to each other.  Specifically other members of this clade negatively connected to RFI included the Synergistia class,  Synergistales order and Pyramidobacter genus.  Indeed,  the relationship between the members of this specific phylum was apparent through the PCIT;  correlation and;  hierarchical clustering analyses undertaken.  Of these microbial taxa,  the Pyramidobacter genus has previously been implicated towards variation in feed efficiency.  For example,  McLoughlin et al.  identified positive associations between the relative abundance of the Pyramidobacter genus and feed conversion ratio in sheep,  whilst also reporting a negative association with ADG.  Similarly in Simmental bulls,  McGovern et al.  identified an association between RFI and abundance of the Pyramidobacter genus.  Whilst in pigs McCormack et al.  and Kubasova et al.  identified an association between the Pyramidobacter genus and RFI in the cecal digesta of pigs and reported greater abundance of the Pyramidobacter genus in the fecal microbiota of Low-RFI pigs compared to High-RFI pigs,  respectively.  Similar to the Synergistetes phylum,  the Planctomycetes phylum as well as class sub-member (Planctomycetia),  and order member (Pirellulales) were also negatively correlated with RFI,  whilst positively correlated with one another.  Indeed,  the Planctomycetes phylum was shown to be altered in both pigs and hens divergent in feed efficiency potential.  In cattle,  Freetly et al.  reported an effect of ADG on abundance of the Planctomyetes phylum,  Planctomycetia class and Pirellulales order in the rumen of beef cattle,  with results from this current study highlighting a role for these microbes towards RFI in beef cattle. Of all the microbes directly connected to the phenotypes examined in this study only one was commonly significantly correlated across more than one phenotype.  Specifically,  the Succiniclasticum genus was negatively correlated with both RFI and ADG.  Moreover,  through the regression analysis,  the Succiniclasticum genus was observed to contribute to variation in all three traits examined in this study.  A role for the Succiniclasticum genus towards variation in feed efficiency has been established previously within the published literature,  for example,  in a study by Myer et al.  more efficient steers were observed to have greater abundance of succinate producing bacteria including the Succiniclasticum genus.  Auffret et al.  also reported significantly greater abundance of the Succiniclasticum genus in high feed efficiency beef cattle.  Conversely though,  Manzanares-Miranda et al.  reported lower abundance of the Succiniclasticum genus in Low-RFI bulls.  Interestingly,  an effect of diet on the abundance of Succiniclasticum genus is evident within the literature.  For example,  McCann reported greater proportions of the Succiniclasticum genus in steers consuming low quality forage,  however the same authors observed abundance to be mostly undetected in forage diets but more abundant in a high grain diet.  Additionally,  McCabe et al.  reported alterations to the abundance of the Succiniclasticum genus dependent on the amount of feed consumed,  whilst Luo et al.  showed that a high concentrate diet increased the abundance of the Succiniclasticum genus.  Moreover,  there is also evidence for an effect of breed on ruminal abundance of Succiniclasticum genus.  For example,  the Succiniclasticum genus was higher in the High-RFI beef cattle in Li et al.,  however this difference was only apparent in Charolais and not in the other breeds examined in that study.  Similarly,  in sheep,  McLoughlin et al.  identified higher abundance of the Succiniclasticum genus in the Connemara breed of sheep compared to other breeds.  Together these results indicate towards a role for the Succiniclasticum genus towards the phenotypes examined in this study,  however the contribution may be dependent on both individual genotype as well as the dietary management system in place. ', 'Conclusion': 'ConclusionsThe rumen microbiome influences the availability of nutrients for subsequent growth purposes through the degradation of ingested feed.  Equally the liver is a highly metabolically active organ and both the rumen microbiome and liver have been shown to be affected by RFI phenotype as well as dietary intake.  Results from this study highlight the interaction amongst the rumen microbiome,  VFA concentrations and hepatic gene expression profiles with three production and efficiency traits of interest,  namely DMI,  ADG and RFI.  Results from this interactive study show a clear relationship between hepatic genes related to lipid metabolism towards RFI,  whilst genes with an associated immune function were reported as primarily related to DMI.  This study also indicated towards differential microbiome interactions between RFI and DMI despite the known correlation between these two traits,  whilst the Succiniclasticum genus was identified as the only microbe connected to more than one trait,  namely ADG and RFI.  However,  results highlight a potential role for both the Succiniclasticum genus and R. faecis sp.  towards RFI,  DMI and ADG phenotypes in beef cattle.  Taken together,  this study provides insights into the interaction amongst rumen microbiome and hepatic gene expression,  which may be contributing to the underlying biology of DMI,  ADG and RFI in beef cattle. ', 'References': []}\n",
      "<pymongo.synchronous.cursor.Cursor object at 0x00000231822E5750>\n",
      "{'_id': ObjectId('66f3fd777027ab88318f65da'), 'pmc_id': 'PMC11416653', 'title': 'Impact of postsurgical vaginal microbiome on high-risk HPV infection and recurrence risk in patients with cervical cancer and intraepithelial neoplasia: A retrospective study', 'abstract': 'No Abstract Available', 'authors': ['Ma Y', 'Wan L', 'Li R', 'Chen X', 'Wang H'], 'publication_date': '2024 Sep 11', 'journal_name': 'Gynecol Oncol Rep', 'doi': '10.1016/j.gore.2024.101506', 'keywords': [], 'score': 10, 'filled_sections_count': 0.6666666666666666, 'Introduction': 'IntroductionCervical cancer (CC) is one of the most common malignancies affecting women globally.  In 2022,  CC accounted for approximately 660, 000 new cases and 350, 000 mortalities worldwide.  Persistent infection with high-risk human papillomavirus (hrHPV) has been established as the primary etiological factor for cervical intraepithelial neoplasia (CIN) and the subsequent development of CC.  Alterations in the vaginal microbiome (VM) are known to play a role in the onset and progression of CIN and CC.  Furthermore,  a decrease in lactobacilli populations and increased bacterial diversity have been associated with hrHPV infection and the development of CIN. Surgical resection is the cornerstone of CIN and CC management,  as it significantly diminishes the likelihood of disease progression and recurrence.  However,  compared with the general population,  women who have undergone surgical treatment have a two- to fivefold greater risk of recurrence or progression to invasive carcinoma.  Persistent and new-onset hrHPV infections are major contributors to postsurgical recurrence in patients with CIN or CC.  Although VM is closely associated with hrHPV infection and the progression of CIN in individuals who have not undergone surgery,  limited research has been conducted on postsurgical VM in patients treated for CIN or CC.  Consequently,  it remains unclear whether the postsurgical VM influences hrHPV infection and disease recurrence in these patients.  This study aimed to elucidate the influence of postsurgical VM on hrHPV infection and disease recurrence in patients who were treated surgically for CIN or CC.  The findings are anticipated to enhance understanding and guide both monitoring and intervention strategies. ', 'Methods': 'MethodsStudy populationClinical data were retrospectively collected from the electronic medical records of 3182 women who underwent VM examinations at the Department of Gynecologic Oncology of the First Affiliated Hospital of University of Science and Technology of China between November 2016 and October 2023.  The eligibility criteria were that the participants were female patients who had been diagnosed with CIN or CC and subsequently underwent surgical intervention,  including loop electrosurgical excision procedure (LEEP),  cold knife conization (CKC),  laparoscopic total hysterectomy (TLH),  and radical hysterectomy (RH),  in the past 6\\x80\\x9312 months.  Furthermore,  eligible participants were required to have no prior history of cancer,  not be pregnant or breastfeeding,  and test negative for human immunodeficiency virus (HIV).  The exclusion criteria included a negative hrHPV test result at the time of surgery or a positive surgical margin postintervention. Sample collectionCervical specimens were collected for hrHPV detection and the ThinPrep cytologic test (TCT).  Vaginal swabs were also collected alongside cervical specimens to assess the VM indices.  The Tongren Medical Digital Electronic Colposcope System (Tongren,  China) was used to perform colposcopic examinations and to obtain biopsy samples. hrHPV examinationThe hrHPV examination was performed via a Zhongsheng Fangzheng Human Papillomavirus Genotyping Kit (Zhongsheng Fangzheng,  China).  A Light Cycler LC480 PCR instrument (Roche,  Germany) was used to detect fifteen hrHPV subtypes: 16,  18,  31,  33,  35,  39,  45,  51,  52,  53,  56,  58,  59,  66,  and 68. Cervical cytology and histologyThe cervical cytological and histological analyses were completed by the pathology department of the hospital.  Following the preparation of liquid-based thin-layer slides,  the cervical specimens were diagnosed by cytologists via the TBS (the Bethesda system) classification system.  The pathological specimens were evaluated by at least two expert pathologists.  Disease recurrence was defined as the reappearance of cervical dysplasia or cancerous tissue at the cervix or at the cervical stump. VM assessmentVaginal secretions were collected using two sterile cotton swabs for VM analyses.  For direct examination,  one swab was smeared onto a glass slide,  subjected to Gram staining,  and examined under an oil immersion microscope (Olympus,  Japan).  The second swab was placed into a tube,  to which 8\\x80\\x9310 drops of saline were added.  After sufficient squeezing,  the secretions were washed,  and the swabs were removed.  A 50-µL aliquot of the eluate was dropped onto a glass slide,  covered with a cover slip,  and left to stand for 3\\x80\\x935 min before microscopy was performed.  The morphological microscopy examination included leukocytes,  epithelial cells,  dominant bacteria,  trichomonas,  fungal hyphae,  and spores.  On the basis of the microscopy results,  analyses of bacterial density,  flora diversity,  Nugent score,  and Donders score were conducted.  Furthermore,  an additional 50 µL of eluate was subjected to a functional assay using the Shidasi Vaginal Microbiome Test Kit (Shidasi,  China) to assess parameters such as pH,  hydrogen peroxide (H2O2) production,  leukocyte esterase activity,  and sialidase activity. The bacterial density refers to the degree of distribution of bacteria in a sample,  which can reflect the total biomass of the microbiome.  The classification of bacterial density is based on the average number of bacteria per field of view under each oil immersion lens: 1\\x80\\x939 bacteria per field are defined as Grade I,  10\\x80\\x9399 bacteria as Grade II,  100\\x80\\x93999 bacteria as Grade III,  and >1000 bacteria as Grade IV.  Flora diversity is graded on the basis of the average number of bacterial species per field of view under the oil immersion lens,  reflecting the complexity of bacterial species distribution in the sample.  The identification of 1\\x80\\x933 bacterial species per field was defined as Grade I,  4\\x80\\x936 species as Grade II,  7\\x80\\x939 species as Grade III,  and >10 species as Grade IV.  The presence of bacterial vaginosis (BV) was diagnosed with a Nugent score of 7 or higher,  and the presence of aerobic vaginitis (AV) was indicated by a Donders score of 3 or higher (Donders et al.,  2001).  The balance in VM was defined as Grade II\\x80\\x93III bacterial density and flora diversity,  a vaginal pH < 4. 5,  a leukocyte count of <10 per high-power microscopic field,  and lactobacilli as the dominant bacteria identified.  If any one of the bacterial density,  flora diversity,  leukocyte count,  dominant bacteria,  or pH value was abnormal,  this status was defined as VM imbalance. Statistical analysisStatistical analysis was performed via SPSS version 21. 0 (IBM Corp.,  Armonk,  NY,  USA).  The skewed distribution quantitative data were represented by median and inter-quartile range,  and differences between two groups were detected by non-parametric test.  The differences in qualitative data between groups were determined via the chi-square test,  and ranked data from different groups were analyzed via the Mann-Whitney U test.  The odds ratios (ORs) and 95 % confidence intervals (CIs) were analyzed via the chi-square test.  All the statistical tests were two-sided and considered to be significant at P < 0. 05. ', 'Results': \"ResultsAfter the clinical data of the 3182 women who underwent VM examinations at the First Affiliated Hospital of USTC between November 2016 and October 2023 were screened,  231 women met the eligibility criteria.  Fourteen women who had negative hrHPV results at the time of surgery,  along with 10 women with positive surgical margins postintervention,  were excluded.  Consequently,  a cohort of 207 women was included in the analysis (Fig.  1).  The median age (interquartile range,  IQR) of the cohort was 46. 0 (38. 0\\x80\\x9351. 0) years,  and 55 of the 207 patients (26. 6 %) were postmenopausal.  Among the 207 patients included in this study,  59 underwent hysterectomy (22 TLH and 37 RH) with complete excision of the cervix,  whereas the other 148 underwent LEEP or CKC,  by which the cervix was not excised completely. Histological analysis of surgical specimens revealed CIN1 in 11 patients,  CIN2 in 40 patients,  CIN3 in 112 patients,  and cervical carcinoma in 44 patients (39 squamous-cell carcinomas,  4 adenocarcinomas,  and 1 adenosquamous cell carcinoma). At 6\\x80\\x9312 months after surgery,  hrHPV was detected in 64 of the 207 patients,  whereas 143 tested negative for hrHPV.  Cervical cytology yielded normal results in 181 patients but revealed abnormalities in 26 patients.  Upon review of all colposcopy results for patients with positive hrHPV or abnormal TCT results,  histopathological examination of cervical biopsy tissues revealed pathological abnormalities suggestive of recurrence in 25 patients.  Specifically,  there were 21 cases of CIN1,  2 cases of CIN2,  and 2 cases of CIN3.  The detailed patient information is presented in Table 1 and Supplementary Table 1. The 207 women included were grouped according to their postsurgical hrHPV,  cervical cytology,  and recurrence status.  The patients\\x80\\x99 characteristics were comparatively evaluated.  As indicated in Table 2,  variables such as age,  surgical modality,  and diagnosis at time of index surgery did not significantly differ between patients who tested positive and those who tested negative for hrHPV.  However,  the hrHPV-positive group presented a greater incidence of VM imbalance,  abnormal cervical cytology,  and disease recurrence than did the hrHPV-negative group (P < 0. 05).  Similarly,  there were no significant differences in variables,  such as age,  surgical modality,  or diagnosis at time of index surgery,  between patients with normal and abnormal cervical cytology results.  The group with abnormal cervical cytology presented a greater prevalence of VM imbalance,  hrHPV positivity,  and disease recurrence.  No differences were observed in surgical modality or diagnosis at time of index surgery between the recurrence group and the nonrecurrence group.  However,  patients in the recurrence group were older,  had a greater proportion of VM imbalance,  and presented higher rates of hrHPV positivity and cytological abnormalities.  These findings suggest a significant association between VM,  hrHPV,  cervical cytology,  and disease recurrence in postsurgical patients.  Regardless of the chosen endpoint for analysis,  VM is clearly associated with postsurgical patients' diseases.  Therefore,  a detailed analysis of the individual indices of VM and their interrelationships was conducted. As shown in Table 3,  no correlation was found between bacterial density,  flora diversity,  or the presence of BV and postsurgical hrHPV,  cervical cytology,  or disease recurrence;  similarly,  neither the leukocyte count nor the presence of leukocyte esterase was associated with these three conditions.  Conversely,  a significant correlation was observed between the vaginal pH and lactobacilli proportion with hrHPV,  cervical cytology,  and recurrence.  Specifically,  patients with a greater vaginal pH (above 4. 5) presented significantly higher rates of hrHPV infection (45. 7 % vs.  15. 7 %,  P < 0. 001),  abnormal cervical cytology (20. 0 % vs.  4. 9 %,  P = 0. 001),  and recurrence (19. 0 % vs.  4. 9 %,  P = 0. 002) than patients with a lower vaginal pH.  Patients with a greater proportion of lactobacilli (above 50 %) had significantly lower incidences of hrHPV infection (14. 6 % vs.  41. 6 %,  P < 0. 001),  abnormal cervical cytology (6. 1 % vs.  16. 8 %,  P = 0. 023),  and recurrence (4. 9 % vs.  16. 8 %,  P = 0. 010) than those with a lower proportion of lactobacilli.  In addition,  the presence of AV was significantly associated with hrHPV infection (62. 5 % vs.  43. 4 %,  P = 0. 011) and recurrence (68 % vs.  46. 7 %,  P = 0. 046) but not with abnormal cervical cytology (65. 4 % vs.  47. 0 %,  P = 0. 079).  With respect to pathogen detection,  no significant correlations were found between the presence of fungal spores or hyphae and the three aforementioned conditions.  However,  a significant association was noted between the detection of trichomonads and hrHPV infection (Table 3). The impact of VM and specific indices on the risk of postsurgical hrHPV,  cervical cytology,  and disease recurrence was further analyzed.  As shown in Table 4 and Fig.  2,  VM imbalance,  pH above 4. 5,  and a lactobacilli proportion below 50 % were identified as risk factors for postsurgical hrHPV infection,  abnormal cervical cytology,  and recurrence.  The presence of AV was also found to be a risk factor for postsurgical hrHPV infection and recurrence but not for abnormal cervical cytology.  Trichomoniasis infection was observed to be a risk factor solely for hrHPV infection. \", 'Discussion': 'DiscussionSurgery plays a crucial role in the treatment of CC and CIN.  The use of antibiotics and irrigation both before and after surgery may lead to presurgical VM failing to accurately represent the postsurgical condition.  Despite extensive research outlining the link between VM and cervical-related diseases,  substantial prospective clinical trials exploring the connection between postsurgical VM and CIN/CC are lacking.  There are very few accompanying data that provide some insights,  making it difficult to provide direct guidance for clinical practice.  Hence,  we undertook this study to address these gaps. In our study,  postsurgical VM imbalance was significantly associated with hrHPV infection,  abnormal cervical cytology and disease recurrence.  Among the specific VM indices,  a lactobacilli proportion less than 50 % was a significant risk factor for these conditions.  Our findings align with those of an observational study conducted by.  In that study,  postsurgical hrHPV status was associated with postsurgical vaginal microbial types in patients who underwent surgical treatment for CIN.  The proportion of nonlactobacilli-dominant VM was approximately 29 % in hrHPV-positive patients,  which was greater than the 10 % reported in hrHPV-negative patients;  however,  the difference was not statistically significant.  In the present study,  the percentage of nonlactobacilli-dominant VMs in the hrHPV-positive group was approximately 81. 3 %,  which was significantly greater than the 51. 0 % reported in the hrHPV-negative group (P < 0. 001). Trichomonads are protozoan parasites responsible for trichomoniasis,  the most common nonviral sexually transmitted infection,  which severely damages the reproductive system.  Trichomonad infection increases the risk of hrHPV infection,  thereby promoting the development of cervical cancer.  This study also confirmed that trichomonad infection is a risk factor for hrHPV infection in postsurgical CIN and CC patients.  However,  trichomonad infection was not observed to increase the risk of cervical cytological abnormalities or recurrence in these patients,  which may be attributed to the nature of the study.  This was a single-center retrospective study,  and the study\\x80\\x99s design was its main limitation,  as it inherently limited the number of cases included. A significant negative correlation between vaginal lactobacilli and hrHPV infection,  as well as the development and progression of CIN/CC,  has been reported previously,  suggesting a potential protective role of lactobacilli in female health.  This finding is consistent with the results of our observational study.  Furthermore,  various probiotics have been shown to inhibit vaginal pathogens,  improve the microenvironment,  and even exhibit anticancer effects in vitro.  Therefore,  investigating whether VM intervention in patients with CIN or CC who have been surgically treated can aid in the recovery of these patients could be a meaningful and valuable research direction in the future. The evaluation methods used for lower genital tract infections in women vary in China,  including traditional cleanliness assessments commonly used in clinical practice,  the VM evaluation employed in this study,  and the prevalent 16S rRNA sequencing used in scientific research.  Among these methods,  the cleanliness assessment is the most historically utilized approach.  However,  this technique is limited to the examination of secretions under a low-power microscope and lacks staining microscopy and enzyme activity analysis,  thus resulting in significant constraints.  The 16S rRNA sequencing technique,  driven by high-throughput sequencing,  represents a novel method capable of comprehensive bacterial composition and quantification analysis within vaginal secretions.  However,  this method lacks direct observational outcomes,  precluding visual inspection of eukaryotes such as cells,  trichomonads,  and fungi,  as well as enzyme activity and pH data acquisition.  Compared with cleanliness assessment and high-throughput sequencing,  the VM assessment method utilized in this study integrates high-power microscopic observation,  Gram staining,  and enzyme activity analysis,  presenting a more comprehensive and meticulous approach.  On the basis of the results from Gram staining and oil microscopy,  we can conduct quantitative Donders and Nugent scoring for the straightforward diagnosis of AV and BV.  Furthermore,  our analysis indicated that AV serves as a significant risk factor for postsurgical recurrence among patients (Table 4 and Fig.  2). This study is a retrospective investigation,  inevitably accompanied by certain limitations,  such as selection bias.  Our inclusion/exclusion criteria resulted in missing data for patients who did not adhere to regular visits because of factors such as distance from medical centers,  economic status,  and educational level.  Additionally,  the limited number of cases in this study precluded more detailed subgroup analyses,  such as grouping patients who underwent complete versus incomplete cervical excision during surgery and comparing patients with carcinoma to those with dysplasia.  These subgroup analyses would have been instrumental in further elucidating the mechanisms underlying the interaction between the vaginal microbiome and disease recurrence in postsurgical patients with CIN/CC. ', 'Conclusion': '', 'References': []}\n",
      "<pymongo.synchronous.cursor.Cursor object at 0x00000231822E5750>\n",
      "{'_id': ObjectId('66f3feec53ac3a0e06052eb2'), 'pmc_id': 'PMC11419422', 'title': 'Dissecting causal relationships between gut microbiome, immune cells, and brain injury: A Mendelian randomization study', 'abstract': 'No Abstract Available', 'authors': ['Xian L', 'Xu X', 'Mai Y', 'Guo T', 'Chen Z', 'Deng X'], 'publication_date': '2024 Sep 20', 'journal_name': 'Medicine (Baltimore)', 'doi': '10.1097/MD.0000000000039740', 'keywords': [], 'score': 10, 'filled_sections_count': 0.8333333333333334, 'introduction': '1.  IntroductionBrain injury (BI),  including traumatic BI (TBI),  ischemic and hemorrhagic stroke,  anoxic BI,  drug-related BI,  and status epilepticus,  all stand out as global health burdens,  that culminate in lifelong disabilities or mortalities,  resulting in a detrimental impact on injured people,  their family and the whole society.  Despite a segment of preclinical researchers who have made substantial strides in developing efficacious therapeutic approaches to avoid neuronal death and preserve neurological function,  patients with BI have higher disabilities and poor prognosis.  Additionally,  treatment approaches to alleviate secondary BI following BI can culminate in deleterious off-target impacts on distant organ systems which may influence the the severity and prognosis of BI.  Thus,  further understanding the molecular physiology of disease and identifying novel effective treatment targets,  which can prevent several BI concurrences or block secondary BI in survivors,  is imperative. Research delving into the function of the gut microbiome (GM) in regulating brain function,  such as neurogenesis,  neuroinflammation,  and behavior,  has exponentially heightened over the past decade,  albeit predominantly in animal models.  Pieces of literature have indulged in determining the bidirectional communication pathways between the microbiota of the central nervous system (CNS) and gut microbiota particularly recognized as the microbiota-gut\\x80\\x93brain axis.  Dysregulation of the aforementioned axis has been exceedingly linked to the pathophysiology of neurological disorders like Alzheimer disease,  BI,  Parkinson disease,  stroke,  and so on.  For this reason,  several researchers call the GM the \\x80\\x9csecond brain. \\x80\\x9d Therefore,  further comprehension of the effects of the microbiota-gut\\x80\\x93brain axis in BI may introduce benefits clinically. An observational study,  recruiting 10 and 24 healthy control volunteers and TBI patients,  correspondingly,  investigated the characteristics of GM and stated that the abundance of Enterococcus,  Akkermansia,  Parabacteroides,  and Lachnoclostridium was substantially increased,  whereas the abundance of Bifidobacterium as well as Faecalibacterium were diminished among the TBI patients.  Li et al found that ischemic stroke patients had significant GM dysbiosis with an increase in the abundance of short-chain fatty acids (SCFAs)-synthesizing bacteria,  for instance,  Akkermansia and Odoribacter,  which was markedly linked to stroke outcome.  Additionally,  several animal\\x80\\x99 studies ascertained that GM changes may control a pro-inflammatory response after BI and can worsen/relieve secondary BI.  For example,  Celorrio et al propounded that antibiotic-induced GM dysbiosis occurring prior to TBI remarkably exacerbated neuronal loss,  minimized cortical infiltration of Ly6Chigh monocytes in addition to T lymphocytes,  enhanced microglial pro-inflammatory markers,  and impeded neurogenesis following TBI.  The above pieces of literature bolster the concept of modified GM composition playing a critical role in the development of BI.  Nonetheless,  the field is nascent,  and data interpretation is frequently challenging provided that the microbiome composition is impacted by distinct factors encompassing exercise and diet.  There is a certainty that these few observational studies were predisposed to reverse causation and confounding bias.  It is important to ascertain if such links indicate spurious correlations or causal relations as a result of bias.  Human randomized controlled trials (RCTs) are important in determining if targeting the specific GM can lead to the discovery of novel avenues for treatment. In light of the lack of RCTs,  Mendelian randomization (MR),  conquers the bias because of 2 factors.  These 2 factors are reverse causation and confounding.  MR is a markedly acknowledged technique to assess more robust causal inferences between exposure and clinical outcomes by utilizing genetic variants like instrumental variables (IVs).  Additionally,  increasing literature illustrates the significance of utilizing human genetic data of GM features for clinical evaluations,  which allowed us to deploy an MR approach to ascertain the mutually causal relations of BI and GM.  In the present study,  we execute a thorough MR analysis to ascertain the causal relationship between GM and BI for the first time.  Additionally,  animal and clinical studies have validated that GM may impact BI by controlling the immune cells,  such as neutrophils,  macrophages,  and T cells.  Thus,  we projected that there might be causal relationships between immune cells,  GM,  and BI,  then used mediation MR analysis to clarify the mediating function of these immune cells in the link between GM and BI. ', 'methods': '2.  Materials and methods2. 1.  Study design overviewOur analytical strategy was as follows: first,  we utilized a 2-sample bidirectional MR technique to examine the causal relationships among 211 GM and BI (2 subtypes) utilizing summary-level data gleaned from GWASs;  second,  we studied pieces of evidence on the causal relationships between 731 immune cells and BI (2 subtypes).  Ultimately,  we executed a mediation analysis to measure the percentage of the total effect of specific GM taxa on BI (2 subtypes) that was triggered by immune cells trait.  Figure 1 exhibits a flow chart of the study design.  The STROBE-MR (Strengthening the Reporting of Observational Studies in Epidemiology using Mendelian randomization) checklist was completed for this observational study.  The original GWASs referred to herein have been ethically reviewed and approved by the relevant institutional review committee,  therefore no additional ethics approval or participant consent is required for this research. 2. 2.  Data sources2. 2. 1.  GWAS data sources for GMSummary data of GM in the MiBioGen consortium (https://mibiogen. gcc. rug. nl) encompassed 18, 340 participants of varied ancestries derived from 24 cohorts,  predominantly of European descent.  The MiBioGen consortium curated as well as evaluated the most comprehensive genome-wide meta-analysis amalgamating the 16S fecal microbiome and human genome-wide genotypes from participants.  We only utilized the taxa that manifested in more than 10% of samples to detect genetic loci that influenced relative abundance (microbiome quantitative trait loci),  leading to 211 taxa: 9 phyla,  16 classes,  20 orders,  35 families,  and 131 genera.  The GWAS data of all-inclusive cohorts were adjusted for variables,  including age,  sex,  genetic main components,  and other relevant factors. 2. 2. 2.  GWAS data sources for immune cellsGWAS summary statistics of immunological traits are available publicly from the GWAS Catalog (accession numbers GCST90001391 to GCST90002121).  This dataset includes 731 distinct immune phenotypes,  such as morphological parameters (MP) (n\\x80 =\\x80 32),  absolute cell (AC) counts (n\\x80 =\\x80 118),  relative cell (RC) counts (n\\x80 =\\x80 192),  and median fluorescence intensities (MFI) expressing surface antigen levels (n\\x80 =\\x80 389) were incorporated.  In particular,  the MFI,  AC,  and RC features have B cells,  myeloid cells,  dendritic cells (DCs),  monocytes,  mature stages of T cells,  TBNK (T cells,  B cells,  natural killer cells),  and Treg panels,  whereas the MP feature have TBNK and DC panels.  The initial GWAS on immune traits was executed utilizing data from 3757 individuals of European origin and overlapping cohorts were absent.  Nearly 22 million single-nucleotide polymorphisms (SNPs) genotyped that had high-density arrays were ascribed into the reference panel of the Sardinian sequence and associations were evaluated following adjusting for covariates (i. e.,  age,  sex,  and age2). 2. 2. 3.  GWAS data sources for BILarge-scale GWAS data on FBI (GWAS ID: finn-b-ST19_FOCAL_BRAIN_INJURY) and TBI (GWAS ID: finn-b-TRAUMBRAIN_NONCONCUS) were obtained from the FinnGen Consortium.  The FBI GWAS dataset was available from 137, 641 European individuals (Ncase\\x80 =\\x80 1065,  Ncontrol\\x80 =\\x80 136, 576),  with approximately 16. 4 million variants examined postimputation and quality control.  The TBI GWAS dataset was available from 218, 792 European individuals (Ncase\\x80 =\\x80 3193,  Ncontrol\\x80 =\\x80 215, 599),  with approximately 16. 4 million variants examined after imputation and quality control. 2. 3.  Selection of IVs and data harmonizationThe IVs in MR refer to genetic variants strongly associated with the exposure and not confounded by other factors that influence the outcome,  which can adequately suppress the effect of confounding factors.  There are 3 basic assumptions for a valid MR study that ought to be satisfied for the IVs (Fig.  1): (1) Relevance Assumption: SNPs markedly(P\\x80 <\\x80 5\\x80 Ã\\x97\\x80 10\\x88\\x928 or P\\x80 <\\x80 1\\x80 Ã\\x97\\x80 10\\x88\\x925,  F-statistic\\x80 >\\x80 10) linked to exposures are utilized as IVs;  (2) Independence Assumption: SNPs (IVs) are not linked to pertinent confounding factors \\x80\\x93 that is,  factors connected to both the exposure as well as the respective outcome;  (3) Exclusivity Assumption: SNPs (IVs) influence outcome directly susceptibility via exposure and are not otherwise linked to outcome. SNPs associated with GM and immunological traits were selected at the P\\x80 <\\x80 1\\x80 Ã\\x97\\x80 10\\x88\\x925,  as used in previous MR studies (given that only a small number of SNPs attained a degree of genome-wide significance (P\\x80 <\\x80 5\\x80 Ã\\x97\\x80 10\\x88\\x928)).  SNPs connected with BI were chosen at GWAS conventional thresholds (P\\x80 <\\x80 5\\x80 Ã\\x97\\x80 10\\x88\\x928).  Further,  the examination of the linkage disequilibrium (LD) among SNPs relied on European ancestry reference data derived from the 1000 Genomes Project.  We picked SNPs utilizing an LD coefficient (r2 < 0. 001) and situated over 10Mb,  to identify SNPs with independent genetic effects. Additionally,  in cases where there were no common SNPs between the outcome and exposure,  proxies SNPs in LD (r2\\x80 \\x89¥\\x80 0. 8) were added.  To eliminate the impact of weak IV bias,  we incorporated SNPs whose F-statistic was\\x80 <\\x80 10 (a measurement of these IVs\\x80\\x99 strength). Noticeably,  bidirectional MR evaluation between GM and BI was executed upon excluding IVs linked to immunological traits,  MR analysis between immunological traits and BI was employed upon eliminating IVs linked to GM,  MR evaluation between GM and immunological traits was employed upon eliminating IVs linked to BI. 2. 4.  Statistical analysisThe R version 4. 3. 1 of the package TwoSampleMR (version 0. 5. 6) executed the MR analyses.  Unless a specified P-value is indicated,  a 2-sided P\\x80 <\\x80. 05 connoted statistical significance. 2. 4. 1.  Primary MR analysisMR analysis was employed to ascertain the causal relationships between GM and FBI to generate FBI-specific GM,  between GM and TBI generate TBI-specific GM. Regarding characteristics influenced by 2 or more SNPs,  the random-effects inverse-variance weighted (IVW) models,  which have stable,  as well as,  balanced pleiotropic impacts,  were utilized as the primary approach to estimate causal effects.  Effect estimates for trails controlled by a single SNP were derived utilizing the Wald ratio.  The MR estimates are presented as odds ratio (OR) with a 95% CI for the dichotomous data or beta value with standard error (SE) for the continuous variables.  To improve the reliability and strength of the causal relationship between specific GM and BI,  we take the intersection of FBI-specific GM and TBI-specific GM,  namely BI-specific GM. 2. 4. 2.  Reverse MR analysisUsing identical MR methods,  we examined if FBI/TBI causally impacts GM traits and studied the possibility of reverse causation.  FBI/TBI was taken into account as the exposure factor,  and FBI-specific GM traits or TBI-specific GM were considered outcomes in this reverse MR analysis. 2. 4. 3.  MR mediation analysisA stepwise MR evaluation technique identified the presence of mediation effects of immunological traits between FBI-specific GM traits and FBI,  between TBI-specific GM traits and TBI.  The 2-step MR presumed that no interaction between exposure and that of the mediator.  The fundamental effect approximation of GM on BI (FBI and TBI) obtained from the univariate MR analyses,  the total effect was defined as Î²1.  Additionally,  2 other estimates need to be computed: (1) the causal exerted by the mediator (731 immune cells) on BI (FBI and TBI) was defined as Î²2,  to improve the reliability and strength of the causal relationship between specific immunological traits and BI,  we take the intersection of FBI-specific immune cells and TBI-specific immune cells,  namely BI-specific immune cells;  and (2) exposure causal effect (FBI-specific GM taxa or TBI-specific GM taxa) on the mediator (BI-specific immune cells) was denoted as Î²3.  Î²2Ã\\x97Î²3 and Î²2Ã\\x97Î²3/Î²1 connoted the mediating and the percentage of the mediating effects,  correspondingly. 2. 4. 4.  MR sensitivity and heterogeneity analysisTo ascertain if MR impact estimates are resilient to possible invalid genetic variants,  we executed MR-Egger regression,  weighted median (WM),  simple mode,  and weight mode as sensitivity analyses.  In comparison to the IVW method,  which presumes that all the SNPs are valid IVs when the Instrument Strength must be Independent of the Direct Effect (InSIDE) assumption is true,  the MR-Egger regression test has the potential to produce a reliable estimate where every genetic instruments are deemed invalid.  The WM model stands out as a robust method,  capable of availing consistent estimate results when over half of the genetic instruments are deemed valid.  For any possible heterogeneity,  we applied Cochran Q statistic derived from the MR-Egger regression and IVW methods.  They were visualized using funnel plots,  while the MR-Egger intercept test tested horizontal pleiotropy,  a threshold of P\\x80 <\\x80. 05 for both was utilized.  Finally,  we performed a leave-one-out (LOO) analysis to re-calibrate the overall effect size and explored whether the association can be affected by a single SNP,  by eliminating 1 exposure-linked SNP at a time. ', 'results': '3.  Results3. 1.  Genetic instruments for GM and immune cellsBy applying the indicated significance level (P\\x80 <\\x80 1\\x80 Ã\\x97\\x80 10\\x88\\x925),  LD clumping,  harmonization,  and F-statistics\\x80 >\\x80 10,  genetic IVs for 211 GM taxa and 731 immune cells were identified.  Consequently,  our study certainly exhibited negligible instrument bias. 3. 2.  Causal effects of GM on BIWhen examining the causal effects of GM on FBI,  we screened 1423 SNPs as IVs from 211 GM taxa (Table S1,  Supplemental Digital Content,  http://links. lww. com/MD/N616).  A total of 6 GM taxa (FBI-specific GM) yielded causal effects on FBI using the IVW method (P\\x80 <\\x80. 05) (Table S2,  Supplemental Digital Content,  http://links. lww. com/MD/N616 and Figure S1,  Supplemental Digital Content,  http://links. lww. com/MD/N615).  Sensitivity analysis showed that only 5 GM taxa using the IVW method had the same direction as the findings of MR-Egger regression,  WM,  simple model,  and weight model (Table S2,  Supplemental Digital Content,  http://links. lww. com/MD/N616 and Figure S2,  Supplemental Digital Content,  http://links. lww. com/MD/N615).  When examining the causal effects of GM on TBI,  we identified 1423 SNPs as IVs from 211 GM taxa (Table S3,  Supplemental Digital Content,  http://links. lww. com/MD/N616).  A total of 8 GM taxa (TBI-specific GM) generated causal effects on TBI using the IVW method (P\\x80 <\\x80. 05) (Table S4,  Supplemental Digital Content,  http://links. lww. com/MD/N616 and Figure S3,  Supplemental Digital Content,  http://links. lww. com/MD/N615).  Sensitivity analysis showed that only 6 GM taxa using the IVW method had the same direction as the findings of MR-Egger regression,  WM,  simple model,  and weight model (Table S4,  Supplemental Digital Content,  http://links. lww. com/MD/N616 and Figure S4,  Supplemental Digital Content,  http://links. lww. com/MD/N615).  To further improve the reliability and strength of the causal interrelationship between specific GM and BI,  we take the intersection of FBI-specific GM and TBI-specific GM and consider the direction of action of the GM.  As a result,  3 taxa (Anaerofilum,  LachnospiraceaeNC2004group,  and RuminococcaceaeUCG004) were causally associated with FBI and TBI (Fig.  2).  In FBI,  host-genetic-driven increase in Anaerofilum (OR\\x80 =\\x80 0. 74,  95% CI\\x80 =\\x80 0. 57\\x80\\x930. 97,  PIVW = 0. 028),  LachnospiraceaeNC2004group (OR\\x80 =\\x80 0. 75,  95% CI\\x80 =\\x80 0. 58\\x80\\x930. 98,  PIVW = 0. 036),  Escherichia. Shigella (OR\\x80 =\\x80 0. 72,  95% CI\\x80 =\\x80 0. 53\\x80\\x930. 98,  PIVW = 0. 037) exhibited significant protective effects on FBI,  while genetically predicted higher relative abundances of RuminococcaceaeUCG004 (OR\\x80 =\\x80 1. 55,  95% CI\\x80 =\\x80 1. 15\\x80\\x932. 08,  PIVW = 0. 004) (Table 1 and Fig.  3A) exhibited significant negative effects on FBI.  In TBI,  genetically predicted potentiates increase in Anaerofilum (OR\\x80 =\\x80 0. 84,  95% CI\\x80 =\\x80 0. 74\\x80\\x930. 97,  PIVW = 0. 016),  FamilyXIIIAD3011group (OR\\x80 =\\x80 0. 78,  95% CI\\x80 =\\x80 0. 63\\x80\\x930. 95,  PIVW = 0. 016),  LachnospiraceaeNC2004group (OR\\x80 =\\x80 0. 84,  95% CI\\x80 =\\x80 0. 73\\x80\\x930. 98,  PIVW = 0. 029) exhibited significant protective functions on TBI,  while host-genetic-driven elevation in RuminococcaceaeUCG004 (OR\\x80 =\\x80 1. 22,  95% CI\\x80 =\\x80 1. 03\\x80\\x931. 45,  PIVW = 0. 023) (Table 1 and Fig.  3B) exhibited significant negative effects on TBI. There was no obvious heterogeneity in the Cochran Q test utilizing MR-Egger regression and IVW (all P\\x80 >\\x80. 05,  Table 1,  Tables S5 and S6,  Supplemental Digital Content,  http://links. lww. com/MD/N616,  and Figures S5 and S6,  Supplemental Digital Content,  http://links. lww. com/MD/N615).  The MR-Egger regression intercepts failed to deviate from zero,  a clear indication that horizontal pleiotropy was lacking (P\\x80 >\\x80. 05 for all intercepts,  Table 1,  Tables S7 and S8,  Supplemental Digital Content,  http://links. lww. com/MD/N616).  Moreover,  the LOO analyses ascertained that none of the detected causal interrelationships were driven by any individual IV (Figures S7 and S8,  Supplemental Digital Content,  http://links. lww. com/MD/N615). 3. 3.  Reverse MR analysis-causal effects of BI on GMTo mitigate the impact of reverse causality on the above findings,  we executed a reverse MR analysis with FBI/TBI-specific GM traits as the outcome and FBI/TBI as the exposure variables.  By utilizing the indicative significance level (P\\x80 <\\x80 5\\x80 Ã\\x97\\x80 10\\x88\\x928),  LD clumping,  harmonization,  and F-statistics\\x80 >\\x80 10,  a total of 13 and 103 SNPs were identified for FBI and TBI,  respectively.  MR analysis discovered no substantial reverse causal association of BI on FBI/TBI-specific GM was detected. 3. 4.  Causal functions of the mediator (immune cells) on BIWhen evaluating the causal functions of immune cells on FBI,  we screened 16, 935 SNPs as IVs from 731 immune cell traits (Table S9,  Supplemental Digital Content,  http://links. lww. com/MD/N616).  A total of 30 immunophenotypes (FBI-specific immune cells) yielded causal effects on FBI using the IVW method (P\\x80 <\\x80. 05) (Table S10,  Supplemental Digital Content,  http://links. lww. com/MD/N616 and Figure S9,  Supplemental Digital Content,  http://links. lww. com/MD/N615).  Sensitivity analysis showed that only 27 immunophenotypes using the IVW method had the same direction as the findings of MR-Egger regression,  WM,  simple model,  and weight model (Table S10,  Supplemental Digital Content,  http://links. lww. com/MD/N616 and Figure S10,  Supplemental Digital Content,  http://links. lww. com/MD/N615).  When evaluating the causal effects of immune cells on TBI,  we identified 16, 935 SNPs as IVs from 731 immune cell traits (Table S11,  Supplemental Digital Content,  http://links. lww. com/MD/N616).  A total of 42 immunophenotypes (TBI-specific immune cells) generated causal effects on TBI using the IVW method (P\\x80 <\\x80. 05) (Table S12,  Supplemental Digital Content,  http://links. lww. com/MD/N616 and Figure S11,  Supplemental Digital Content,  http://links. lww. com/MD/N615).  Sensitivity analysis showed that only 39 immunophenotypes using the IVW method had the same direction as the findings of MR-Egger regression,  WM,  simple model,  and weight model (Table S12,  Supplemental Digital Content,  http://links. lww. com/MD/N616 and Figure S12,  Supplemental Digital Content,  http://links. lww. com/MD/N615).  To further improve the reliability and strength of the causal interrelationship between specific immune cells and BI,  we take the intersection of FBI-specific immune cells and TBI-specific immune cells and consider the direction of action of the immune cells.  As a result,  9 immune phenotypes (IgD\\x80 +\\x80 CD24- B cell %lymphocyte,  BAFF-R on IgD\\x80 +\\x80 CD38- naive B cell,  BAFF-R on IgD- CD38- B cell,  BAFF-R on memory B cell,  BAFF-R on switched memory B cell,  CD25 on CD20- CD38- B cell,  CD123 on plasmacytoid DC,  CD123 on CD62L\\x80 +\\x80 plasmacytoid DC,  and CCR2 on myeloid DC) were causally associated with FBI and TBI (Fig.  4).  In FBI,  host-genetic-driven increase in IgD\\x80 +\\x80 CD24- B cell %lymphocyte (OR\\x80 =\\x80 0. 84,  95% CI\\x80 =\\x80 0. 73\\x80\\x930. 97,  PIVW = 0. 019),  BAFF-R on IgD\\x80 +\\x80 CD38- naive B cell (OR\\x80 =\\x80 0. 93,  95% CI\\x80 =\\x80 0. 88\\x80\\x930. 98,  PIVW = 0. 009),  CD123 on plasmacytoid DC (OR\\x80 =\\x80 0. 95,  95% CI\\x80 =\\x80 0. 91\\x80\\x930. 99,  PIVW = 0. 021),  CD123 on CD62L\\x80 +\\x80 plasmacytoid DC (OR\\x80 =\\x80 0. 95,  95% CI\\x80 =\\x80 0. 91\\x80\\x930. 99,  PIVW = 0. 016),  and CCR2 on myeloid DC (OR\\x80 =\\x80 0. 90,  95% CI\\x80 =\\x80 0. 84\\x80\\x930. 96,  PIVW = 0. 002) exhibited significant protective effects on FBI,  while genetically predicted higher relative abundances of BAFF-R on IgD- CD38- B cell (OR\\x80 =\\x80 1. 07,  95% CI\\x80 =\\x80 1. 01\\x80\\x931. 14,  PIVW = 0. 028),  BAFF-R on memory B cell (OR\\x80 =\\x80 1. 08,  95% CI\\x80 =\\x80 1. 00\\x80\\x931. 16,  PIVW = 0. 041),  BAFF-R on switched memory B cell (OR\\x80 =\\x80 1. 07,  95% CI\\x80 =\\x80 1. 00\\x80\\x931. 14,  PIVW = 0. 039),  CD25 on CD20- CD38- B cell (OR\\x80 =\\x80 1. 11,  95% CI\\x80 =\\x80 1. 00\\x80\\x931. 22,  PIVW = 0. 043) (Table 2 and Fig.  5) exhibited significant negative effects on FBI.  In TBI,  genetically predicted increases increase in IgD\\x80 +\\x80 CD24- B cell %lymphocyte (OR\\x80 =\\x80 0. 90,  95% CI\\x80 =\\x80 0. 83\\x80\\x930. 98,  PIVW = 0. 012),  BAFF-R on IgD\\x80 +\\x80 CD38- naive B cell (OR\\x80 =\\x80 0. 95,  95% CI\\x80 =\\x80 0. 92\\x80\\x930. 99,  PIVW = 0. 005),  CD123 on plasmacytoid DC (OR\\x80 =\\x80 0. 98,  95% CI\\x80 =\\x80 0. 95\\x80\\x930. 99,  PIVW = 0. 039),  CD123 on CD62L\\x80 +\\x80 plasmacytoid DC (OR\\x80 =\\x80 0. 98,  95% CI\\x80 =\\x80 0. 95\\x80\\x930. 99,  PIVW = 0. 039),  and CCR2 on myeloid DC (OR\\x80 =\\x80 0. 96,  95% CI\\x80 =\\x80 0. 93\\x80\\x930. 99,  PIVW = 0. 040) exhibited significant protective effects on TBI,  while host-genetic-driven increase in BAFF-R on IgD- CD38- B cell (OR\\x80 =\\x80 1. 04,  95% CI\\x80 =\\x80 1. 01\\x80\\x931. 08,  PIVW = 0. 024),  BAFF-R on memory B cell (OR\\x80 =\\x80 1. 05,  95% CI\\x80 =\\x80 1. 01\\x80\\x931. 09,  PIVW = 0. 008),  BAFF-R on switched memory B cell (OR\\x80 =\\x80 1. 04,  95% CI\\x80 =\\x80 1. 01\\x80\\x931. 08,  PIVW = 0. 012),  CD25 on CD20- CD38- B cell (OR\\x80 =\\x80 1. 07,  95% CI\\x80 =\\x80 1. 01\\x80\\x931. 13,  PIVW = 0. 015) (Table 2 and Fig.  6) exhibited significant negative effects on TBI. Obvious heterogeneity in the Cochran Q test was lacking when utilizing IVW and MR-Egger regression (all P\\x80 >\\x80. 05,  Table 2,  Tables S13 and S14,  Supplemental Digital Content,  http://links. lww. com/MD/N616 and Figure S13 and S14,  Supplemental Digital Content,  http://links. lww. com/MD/N615).  The MR-Egger regression intercepts failed to deviate from zero,  signifying no indication of horizontal pleiotropy (P\\x80 >\\x80. 05 for all intercepts,  Table 2,  Tables S15 and S16,  Supplemental Digital Content,  http://links. lww. com/MD/N616).  Moreover,  the LOO analyses affirmed that none of the detected causal interrelationships were driven by any individual IV (Figures S15 and S16,  Supplemental Digital Content,  http://links. lww. com/MD/N615). 3. 5.  Causal effects of the exposure (GM) on the mediator (immune cells)After identifying the mediator (9 immunophenotypes),  we investigate the causal associations of FBI/TBI-specific GM on it.  As a result,  Escherichia. Shigella was significantly positively associated with CD123 on CD62L\\x80 +\\x80 plasmacytoid DC (OR\\x80 =\\x80 1. 31,  95% CI\\x80 =\\x80 1. 02\\x80\\x931. 69,  PIVW = 0. 035),  and CD123 on plasmacytoid DC (OR\\x80 =\\x80 1. 31,  95% CI\\x80 =\\x80 1. 02\\x80\\x931. 69,  PIVW = 0. 034) (Table 3).  FamilyXIIIAD3011group was significantly positively correlated with CCR2 on myeloid DC (OR\\x80 =\\x80 1. 39,  95% CI\\x80 =\\x80 1. 05\\x80\\x931. 84,  PIVW = 0. 021) (Table 3).  Importantly,  the results of the other 4 analytical methods concurred with the IVW direction,  unifying the cause-effect interrelationship (Table S17,  Supplemental Digital Content,  http://links. lww. com/MD/N616).  No evidence of heterogeneity and horizontal pleiotropy (all P\\x80 >\\x80. 05) was observed among the IVs,  denoting the reliability of the results (Table 3).  The LOO analyses verified that no individual IV markedly influenced the detected causal associations. 3. 6.  Mediating effects of immune cell traits on GM\\x86\\x92BI effectsBased on \\x80\\x9cGM\\x86\\x92FBI/TBI, \\x80\\x9d \\x80\\x9cimmune cells\\x86\\x92FBI/TBI\\x80\\x9d and\\x80\\x9cGM\\x86\\x92immune cells\\x80\\x9d in the previous analysis,  we considered immune cells to mediate the relationship between GM and FBI/TBI (Fig.  7).  Specifically,  CD123 on CD62L\\x80 +\\x80 plasmacytoid DC and CD123 on plasmacytoid DC mediated the effects of Escherichia. Shigella on FBI,  with mediating impacts of \\x88\\x920. 0136,  and \\x88\\x920. 0136,  denoting 4. 21%,  and 4. 21% of the total effects,  correspondingly (Table 4).  CCR2 on myeloid DC mediated the effects of FamilyXIIIAD3011group on TBI,  with mediating effects of \\x88\\x920. 0129,  reflecting 5. 07% of the total effects (Table 4). ', 'discussion': '4.  DiscussionThe present study utilized large-scale GWAS summary data and a 2-sample MR technique to ascertain the relationship between 211 GM,  731 immune cell subtypes,  and FBI/TBI.  Our genetic analyses found several potential causal associations between GM and FBI/TBI,  and further analysis revealed that immunophenotypes mediated a partial proportion of the causal effect of GM on the FBI/TBI risk.  Our findings indicated that different GM taxa exert different effects on FBI/TBI,  underlying the necessity of controlling host\\x80\\x93microbe balance in the prevention and management of FBI/TBI. Accumulating evidence supported an association between GM abundance and BI,  however,  there are deficient direct pieces of evidence affirming a causal correlation.  Considering that BI comprises a heterogeneous group with diverse disease entities (such as TBI,  and FBI),  each accompanied by individual pathophysiology and variations.  MR analysis investigated the causal relationship between 211 GM taxa and TBI and FBI,  respectively.  As a result,  a total of 5 genetically predicted GM taxa (RuminococcaceaeUCG004,  Anaerofilum,  LachnospiraceaeNC2004group,  Escherichia. Shigella,  Coprococcus3) and 6 genetically predicted GM taxa (RuminococcaceaeUCG004,  Anaerofilum,  LachnospiraceaeNC2004group,  FamilyXIIIAD3011group,  Coprococcus3,  Gordonibacter) significantly affected FBI and TBI,  respectively,  using IVW method had the same direction as the findings of MR-Egger regression,  WM,  simple model and weight model.  It is worth noting that there are 3 shared GM taxa (RuminococcaceaeUCG004,  Anaerofilum,  LachnospiraceaeNC2004group) that have a significant causal relationship with FBI and TBI.  Within the Ruminococcaceae family,  RuminococcaceaeUCG004 represents a genus of anaerobic bacteria.  Urban et al employed a 2-site trial to ascertain the function of the gut\\x80\\x93brain axis in comorbidities connected with chronic TBI and compared the microbiome derived from fecal of 22 moderate/severe TBI patients profiles with 18 healthy age-matched control subjects.  The result found that bacteria in the Ruminococcaceae family were more abundant in chronic TBI in comparison to control profiles.  However,  Opeyemi et al observed that the microbial diversity is progressively diminished in the gut upon TBI,  and bacteria with an origin traced to the Ruminococcaceae families were exhausted compared with control groups.  In our study,  a host-genetic-driven increase in RuminococcaceaeUCG004 exhibited significant negative effects on FBI and TBI.  Liu et al performed an observational study to analyze the 16S rRNA intestinal microbiome between the delayed neurocognitive recovery (dNCR) group and the non-dNCR group and found that the abundance of Anaerofilum was significantly enriched in the dNCR group.  However,  no Anaerofilum and BI have been reported to date.  A recent study uncovered that the genus Anaerofilum was the only bacterial group identified that had a significant negative association with carcinoid syndrome (such as diarrhea).  This study first revealed the protective role of Anaerofilum in FBI and TBI.  Genus LachnospiraceaeNC2004group is an anaerobic bacteria belonging to the Lachnospiraceae family.  Opeyemi et al reported that bacteria from Lachnospiraceae significantly decrease in the taxonomic composition of fecal samples post-TBI.  These findings align with our study,  which confirms the potential protective effects of LachnospiraceaeNC2004group in TBI and FBI.  A possible explanation is that the Lachnospiraceae families entail dominant SCFA producers in mice and humans,  because of their anti-inflammatory qualities and ability to affect immunological control,  SCFAs may help lessen the impact of inflammation following BI.  Additionally,  research has shown that when drinking water is supplemented with SCFA,  spatial learning is improved following TBI.  Our results ascertained the distinct profiles of dominant taxa in FBI and TBI.  These 2 share common characteristics despite also having differences.  For example,  genetically predicted higher relative abundances of Coprococcus3 are significantly negatively associated with FBI,  while positively associated with TBI.  The difference in results is most likely due to 2 different BIs caused by different causes.  Finally,  we executed a reverse MR analysis with FBI/TBI-specific GM traits as the outcome and FBI/TBI as the exposure variables and found no significant reverse causal association of BI on FBI/TBI-specific GM.  The reliability of the results is further improved.  To sum up,  further characterization of GM taxa,  such as RuminococcaceaeUCG004,  Anaerofilum,  LachnospiraceaeNC2004group,  Escherichia. Shigella,  and FamilyXIIIAD3011group will avail a novel perspective for the development and progress of FBI/TBI and will aid in finding potential therapy for BI patients. Several observational studies and animal experiments have found that immune cell infiltration (such as neutrophils,  monocytes,  microglia,  astrocytes,  T cells,  and B cells) functions in the genesis and progression of BI,  and rejuvenating the immune system supports brain repair after injury.  Generally speaking,  neutrophil numbers commence to diminish after BI,  activated microglia in conjunction with astrocytes assemble at the injury site \\x80\\x93 isolating tissues that are injured from healthy tissues and allowing the restorative processes.  Monocyte infiltration to the injury site is instrumental in the injured brain repair.  The recruitment of T and B cells to the brain pathology locations at later time points can vary depending on the severity of the BI.  In our study,  we performed MR analysis to investigate the causal interrelationship between 731 immune phenotypes and TBI and FBI,  respectively,  to further elucidate the direct causal relationship between them.  As a result,  a total of 27 genetically predicted immunophenotypes and 39 genetically predicted immunophenotypes (such as B cell subtypes,  T cell subtypes,  Treg cells subtypes,  DC subtypes,  granulocyte,  plasma cell,  basophil) significantly affected FBI and TBI,  respectively,  using IVW method had the same direction as the findings of MR-Egger regression,  WM,  simple model and weight model.  Notably,  there are 9 shared immunophenotypes (only B cell and DC subtypes) that have a significant causal relationship with FBI and TBI.  In contrast to well-known immune cell infiltrations (monocytes and T cells),  our findings point out the causal functions of B cells and DC in BI and the related studies are very few so far.  A B cell marker,  OX33,  has been affirmed for 4 to 6 days following experimental injury.  Meissner et al ascertained that a chemokine linked to B cell chemotaxis (CCL20) is potentiated within 4 hours and continues out to 3 days postexperimental BI.  There are scarce pieces of clinical studies that examine the effect of BI on B cell populations.  Mrakovcic-Sutic et al delved into examining B cell (CD5+/CD19+) populations in 20 patients suffering from severe TBI by utilizing their peripheral blood and discovered that there were no substantial differences when contrasted with healthy controls.  A study by 2 groups of scientists inferred that autoantibodies precise to CNS proteins were identified in BI patients\\x80\\x99 serum,  implying a broken tolerance.  B cells initiate an immune response against antigens derived from the brain upon injury and a possible pathophysiological function for B cells throughout BI the recovery phase,  which concurs with this research that propounds that various B cells (for instance IgD+ CD24\\x88\\x92 B cell %lymphocyte,  BAFF-R on IgD\\x88\\x92 CD38\\x88\\x92 B cell,  BAFF-R on IgD+ CD38\\x88\\x92 naive B cell,  and BAFF-R on memory B cell) have a protective/negative role in brain damage.  The involvement of DCs following BI has not been studied deeply in comparison to the other perspectives of immunity.  DCs believed to function in neuroinflammation via processing antigens as well as presenting these antigens to T cells to start an immune response.  DCs occur in limited numbers in the brain parenchyma in normal conditions,  in particular,  predominantly situated adjacent to cerebrospinal fluid,  and continuously monitor the immune microenvironment of the CNS.  In summary,  our current analysis identifies new immunological directions in BI,  where the host-genetic-driven change in B cell and DC infiltration significantly affect FBI and TBI.  Further research is necessary in the future to provide new strategies for BI treatment. Pieces of evidence have affirmed that GM functions in peripheral immune response and neuroinflammation following BI,  modifying the lymphocyte populations.  Numerous immune cells,  like lymphocytes,  participate in dynamic and complex immune responses in various BIs and maybe a central moderator in the GM and BI immunomodulation.  Mediation analyses availed genetic evidence that various distinct immune cells mediate the causal impacts of the GM on BI.  Escherichia. Shigella exerts its protective effects against FBI by increasing the abundance of infiltrating CD123 on CD62L+ plasmacytoid DC and CD123 on plasmacytoid DC.  FamilyXIIIAD3011group increases the proportion of infiltrating CCR2 on myeloid DC to protect against TBI.  In brief,  DC may be an important bridge of immune regulation between GM and BI.  In the intestines,  the intestinal DCs exhibit a local immunomodulatory function.  DCs that have intestinal tolerability and manifest greater levels of CD103,  initiate differentiation of intestinal homing Treg cells in mesenteric lymph nodes (mLN).  Following BI,  the GM ultimately imparts a neuroprotective impact via inducing DCs in the mesenteric mLN to express remarkable levels of CD103 as well as inducing Treg cells to differentiate and produce IL-10 to suppress IL-17+Î³Î´T cells.  These findings in conjunction with the findings of our study offer enlightenment into the causal relationship between the GM and the BI,  specifically pertaining to the mediating effects of DC.  There are limited shreds of literature connected with the function of DCs in BI via the brain\\x80\\x93gut axis,  which highlights the need for further investigation into the interaction between the BI,  GM,  and DC modulation in future studies. The utilization of summary-level data gleaned from extensive GWASs and bidirectional MR design stood out as the current study\\x80\\x99s strengths.  The aforementioned design particularly averted bias as a result of reverse causation and confounding in order to derive accurate findings under the assumptions of MR.  Moreover,  consistent findings from numerous sensitivity analyses and validation of different data set results indicate the robustness of our findings.  In spite of this remarkable sense,  it is inevitable that limitations also existed in our study.  Firstly,  in the MiBioGen consortium,  microbiome profile characterization employs 16S ribosomal RNA gene sequencing,  that only permits resolution from the genus to phylum.  More information about a particular species is availed by metagenomic sequencing.  However,  a prior MR analysis of the GM ascertained that the P-values were occasionally more remarkable for higher taxonomic units,  encompassing genera or phyla,  implying that species provided similar functions.  Secondly,  our findings failed to withstand a strict Bonferroni correction for multiple comparisons.  However,  as a hypothesis-driven approach,  the MR study with some biological evidence was used to test epidemiologically established associations,  irrespective of Bonferroni corrected P values.  As a result of the causal relationship between GM,  immune cells,  and BI,  we are taking the intersection of 2 BI GWAS datasets,  which can be mutually verified,  improving the reliability of the results.  Thirdly,  since more than 70 % of the study population constituted only European ancestry and variations exist in terms of host metabolism,  lifestyle,  and resident GM among people globally,  there is a need to ascertain whether our findings have the likelihood of applying to a distinct ethnic group.  Unequal distribution of genetic variants among distinct racial and ethnic groups can cause population stratification,  and this can introduce potential bias in the outcomes of the study.  There is a need to take caution when executing the generalizability of the study findings on other racial or ethnic groups.  It is crucial for future studies to incorporate a more diverse population to improve the generalizability of the result.  Fourthly,  MR relies on the premise that a linear connection between exposure and outcome,  however in reality,  this relationship may be more intricate,  encompassing nonlinear relationships as well as interactions with other genetic and environmental factors.  For instance,  a few genetic variants may exhibit a substantial impact on the outcome at heightened or diminished levels of exposure,  or the influence of the exposure on the outcome might be mediated or moderated by other factors.  Therefore,  forthcoming MR studies should meticulously account for possible nonlinear and interaction impacts between the immunophenotypes,  GM,  and BI. ', 'conclusion': '5.  ConclusionTo our knowledge,  this study is the first to comprehensively examine the causal relationships between GM,  immune cells,  and FBI/TBI.  Our findings supported several potential causal links between several specific GM taxa with FBI/TBI.  Additionally,  CD123 on plasmacytoid DC and CCR2 on myeloid DC may have a key function in gut microbiota-host crosstalk in FBI and TBI,  correspondingly.  More research is paramount in the underlying mechanisms of GM in the occurrence of FBI/TBI for future development of probiotics to prevent and treat FBI/TBI. ', 'references': []}\n",
      "<pymongo.synchronous.cursor.Cursor object at 0x00000231822E5750>\n",
      "{'_id': ObjectId('66f3feec53ac3a0e06052eb3'), 'pmc_id': 'PMC11418651', 'title': 'Unde venis? Bacterial resistance from environmental reservoirs to lettuce: tracking microbiome and resistome over a growth period', 'abstract': 'No Abstract Available', 'authors': ['Gekenidis MT', 'Vollenweider V', 'Joyce A', 'Murphy S', 'Walser JC', 'Ju F', 'Bürgmann H', 'Hummerjohann J', 'Walsh F', 'Drissner D'], 'publication_date': '2024 Aug 31', 'journal_name': 'FEMS Microbiol Ecol', 'doi': '10.1093/femsec/fiae118', 'keywords': [], 'score': 10, 'filled_sections_count': 0.8333333333333334, 'introduction': 'IntroductionSince the discovery of antibiotics,  antibiotic resistance in pathogenic bacteria has increased dramatically due to over- and misuse in human and veterinary medicine (Davies and Davies).  As a result,  antibiotic-resistant bacteria (ARB) are being detected more frequently,  not only in clinics but also in the environment including soils and water bodies (Guenther et al.,  Graham et al.,  Gekenidis et al.,  Cerqueira et al. ).  In a nonclinical setting,  transmission of ARB to humans occurs indirectly through handling of contaminated animal and natural products as well as directly via consumption of contaminated foods such as meat,  fish,  dairy products,  or plant-based foods (Thanner et al. ).  Fresh produce is of particular interest since its popularity as part of a healthy diet has risen in recent years (Food and Agriculture Organization of the United Nations),  and the frequent raw consumption of such products increases the probability of ARB transfer to the consumer (Rahman et al. ).  This is all the more critical,  as produce has been shown to harbor an impressive diversity of self-transmissible plasmids carrying multiple resistance genes (Blau et al. ). The cultivation environment of fresh produce,  including soil,  irrigation water,  and fertilizer,  is a reservoir of ARB and antibiotic resistance genes (ARGs),  which act as a potential source of contamination (Thanner et al. ).  Soil is considered a natural reservoir of ARB and ARGs,  where resistance determinants accumulate due to agricultural land-use and from where drainage into surface and groundwater can occur (Walsh and Duffy).  In turn,  contaminated water reservoirs can serve as sources of ARB and ARGs when used for irrigation of fresh produce (Gekenidis et al. ).  Strikingly,  wastewater treatment plants (WWTPs),  which continuously release their effluent into nearby streams,  are not able to completely eliminate antibiotics or resistance determinants and have been shown to enrich ARB and ARGs (Czekalski et al. ).  As a result,  WWTP discharge can lead to increased levels of ARB and ARGs in receiving rivers (Lekunberri et al.,  Lee et al. ),  and combined sewer overflows can increase the contamination considerably (Lee et al. ).  Further,  organic fertilizers such as manure\\x80\\x94especially when originating from animals having received antibiotic treatment in the past\\x80\\x94can add to the soil resistome (Marti et al. ),  thereby increasing the risk of fresh produce contamination with ARB and ARGs (Blau et al.,  Sanz et al. ).  Finally,  metal and biocide resistance should be considered jointly with antibiotic resistance,  since co- and cross-resistance (i. e.  distinct resistance determinants colocated physically in a bacterial genome or one resistance determinant conferring resistance to anti-infective compounds of several classes,  respectively) is described frequently and is of clinical concern (Pal et al.,  CÃ¢ndido et al. ). It is well established that antibiotic resistance in the environment is an ancient phenomenon,  while the more recently observed increased resistance levels are related to human activities (Rothrock Jr.  et al.,  Iwu et al. ).  There have been efforts to determine the natural occurrence of antibiotic resistance in pristine environments such as soil,  seawater,  or pristine plants,  as opposed to antibiotic resistance of anthropogenic origin.  A multitude of studies has demonstrated that soils contain high levels of native ARB carrying a broad arsenal of ARGs (Cytryn).  A study isolating bacteria from the deep terrestrial subsurface found resistance towards at least one of 13 antibiotics tested in 90% of isolated strains,  with nalidixic acid,  mupirocin,  and ampicillin resistance being the most frequent (Brown and Balkwill).  A shotgun metagenomic study in 17 pristine and remote Antarctic soils identified 177 naturally occurring ARGs,  the majority of which encoded single or multidrug efflux pumps (Van Goethem et al. ).  Other common mechanisms related to aminoglycoside,  chloramphenicol,  and Î²-lactam antibiotic resistance.  Interestingly,  the researchers describe the lack of mobile genetic elements (MGEs) flanking ARGs.  A metagenome-wide analysis comparing samples from pristine soils and seawater to human,  chicken,  and pig gut samples found a much higher concentration of intrinsic ARGs in pristine environments,  while the abundance of total ARGs was significantly higher in gut samples including ARGs against last resort antibiotics (Zeng et al. ).  The two dominant ARG types in both pristine environments were multidrug and aminocoumarin.  In another study,  psychrotrophic bacteria from Antarctic Marine waters were all resistant to ampicillin and more than half to chloramphenicol and streptomycin (De Souza et al. ).  Finally,  an investigation of 111 Enterobacteriaceae from pristine freshwater showed that resistance to multiple antibiotics was common (61%),  most frequently toward Î²-lactams and chloramphenicol (Lima-Bittencourt et al. ).  In contrast to soil or water,  studies on the innate resistome of plants are rare.  An investigation on the effect of struvite-application on the resistome of the Brassica microbiome revealed the presence of 25 ARGs from eight different classes unique to the phyllosphere [mainly macrolide\\x80\\x93lincosamide\\x80\\x93streptogramin B ARGs (MLSB) and multidrug],  making them potentially phyllosphere-specific ARGs (Chen et al. ).  A more extensive investigation of the same group comprising 12 plant species among which lettuce (Lactuca sativa),  resulted in 172 ARGs unique to the plant phyllosphere while the plant host significantly affected ARG profiles (Chen et al. ).  The ARGs shared between all samples conferred resistance to aminoglycoside,  Î²-lactam,  MLSB,  multidrug,  tetracycline,  and vancomycin.  Another work investigating Sphagnum moss as a plant growing in a pristine environment found a highly diverse resistome,  targeting 29 antibiotics and covering all major resistance mechanisms with an extraordinarily high abundance of efflux pumps (\\x89¤96%) (Obermeier et al. ).  As another environment with least anthropogenic influence,  the primary vegetation of a retreating glacier was studied revealing a core set of phyllosphere ARGs across the successional sequence of plants,  where multidrug and aminoglycoside ARGs were the most abundant (Li et al. ).  Finally,  a study of the native microbiome of two model indoor plants detected a variety of ARGs,  with multidrug resistance (MDR) as the most prevalent resistance category followed by MLSB (Wicaksono et al. ). The prevalence of a broad variety of ARB and ARGs in fresh produce including lettuce has been described in many studies (Blaak et al.,  Vital et al.,  Rahman et al.,  Yin et al.,  KlÃ¤ui et al. ).  In 2017,  the World Health Organization defined a priority list of AR pathogens for which new antibiotics are urgently needed,  divided into three priority classes: (1) critical priority,  including carbapenem-resistant (CR) Acinetobacter baumannii,  CR Pseudomonas aeruginosa,  and CR and extended-spectrum Î²-lactamase (ESBL)-producing Enterobacteriaceae;  and (2) high priority,  including vancomycin-resistant Enterococcus faecium and methicillin-resistant Staphylococcus aureus,  among others.  Of these clinically significant ARB,  many have also been isolated from food or environmental sources.  For example,  ESBL-producing Enterobacteriaceae and vancomycin-resistant Enterococcus spp.  (VRE) have been isolated from a variety of fresh produce (Reuland et al.,  Kim et al. ).  Resistant Acinetobacter spp.  including A.  baumannii have been isolated from hospital tap water,  raw meat,  and cattle manure (Shamsizadeh et al.,  Malta et al. ).  Pseudomonas aeruginosa including CR isolates have been detected in fresh produce (KlÃ¤ui et al. ).  Multidrug-resistant as well as CR and ESBL-producing Klebsiella pneumoniae,  a member of the Enterobacteriaceae family,  have been isolated from raw milk,  various meats,  fruits,  and raw vegetables (Hartantyo et al.,  Junaid et al. ).  Nevertheless,  more research is needed to better understand: (1) which environmental reservoirs contribute to the transfer of ARB and ARGs to foods and to what extent (especially to fresh produce,  which is exposed to a wide variety of potential contamination sources;  Drissner and Gekenidis),  (2) which bacterial genera and resistance genes are involved,  and (3) whether resistance determinants persist until harvest in case of an early contamination event. The aim of the present study was to identify the changes in microbiome composition of lettuce as well as potential source microbiomes in soil,  manure,  and irrigation water,  and to track ARB and ARGs from the investigated environmental reservoirs to the beginning of the plant food chain in order to identify the main sources of fresh produce contamination.  As a model system,  we used lettuce (L.  sativa,  Salanova® Barlach),  grown on a field of a conventional lettuce farm.  The lettuce was grown under four conditions,  representing all combinations of conventional,  that is,  nonmanured field soil (common practice) or manure-amended field soil with untreated or UV-treated river water irrigation.  In a culture-based approach,  we quantified total bacterial numbers and presumptive ARB present in the four environments lettuce leaves,  soil,  water,  and manure,  and monitored their development throughout the lettuce growth period.  Our target bacteria included ESBL-,  carbapenemase-producing,  and fluoroquinolone-resistant Escherichia coli and K.  pneumoniae;  VRE;  and A.  baumannii and P.  aeruginosa.  We further characterized the microbiome of the same four environments using amplicon sequencing,  and investigated their resistome including resistance to antibiotics,  biocides,  and metals.  Such insights are essential to establish comprehensive resistance-monitoring programs along the plant food chain and to develop recommendations for improved agricultural practices. ', 'methods': 'Material and methodsField site and experimental treatmentsThe field trial was conducted in summer 2017 (July and August) on a conventional salad farm in the canton of Berne (Switzerland),  in a region known for its fertile peat soils.  More specifically,  the soil of the selected field site has been classified as calcaric cambisol.  The selected field site is used for lettuce cultivation all year long and is shaded by a foil tunnel,  which reduces the impact of environmental factors such as rain,  wind,  and UV radiation on the plants.  The common practices of the farm included opening the foil tunnel on very hot days to avoid overheating,  plowing the soil before each new culture,  and sprinkler irrigating with water pumped from a nearby river (Saane) receiving up to 9. 1% of effluents from a WWTP located about 2 km upstream (Federal Office of the Environment).  For the present trial,  half of the plants were irrigated with UV-disinfected river water.  To this purpose,  a water disinfection device AQUASTERA/Aqua UVtron ASUV46 was used (aquatec solution GmbH,  Schwarzenbach,  Switzerland),  providing a minimal UVC irradiance of 26. 4 W/m2.  Lettuce seedlings (L.  sativa,  Salanova® Barlach) of 3 g were purchased from a local supplier.  Stored pig manure was obtained from a private pig farm and used for manual soil amendment after lettuce planting while taking special care not to contaminate the seedlings with manure.  The conventional,  that is,  nonmanured field soil (common practice) will be termed \\x80\\x98conventional soil\\x80\\x99 in the following.  Overall,  the lettuce was managed according to the farmer\\x80\\x99s recommended practice for planting,  irrigation,  and temperature regulation. The selected field site consisted of two adjacent patches (total: 100 m Ã\\x97 4 m),  and each was planted with four rows of lettuce heads (35 cm distance between heads).  The patches were divided into four sectors from front to rear with buffer zones in between them,  to compare the common practice to three alternative practices (four treatments A\\x80\\x93D;  Fig.  S1): (A) river water and conventional soil (common practice),  (B) river water irrigation and manure-amended soil,  (C) UV-treated river water irrigation and manure-amended soil,  and (D) UV-treated river water irrigation and conventional soil.  Each of the four treatment sectors was further divided into 18 sampling plots to provide three harvest replicates for 6 weeks,  each containing eight lettuce heads (576 lettuce heads in total).  At the end of each treatment sector,  four plots containing eight lettuce heads each served as buffer zones between treatments. Sampling campaignsPrior to the field trial,  lettuce seedlings,  conventional field soil,  manure,  and river irrigation water were sampled (week 0 or initial condition Z) as described in the respective section below.  Starting 1 week after planting (week 1),  lettuce,  soil,  and irrigation water were sampled weekly during the 6-week lettuce growth period,  to study microbiome and resistome progression over time.  Three biological replicates per treatment and environment (lettuce,  soil,  water,  and manure) were collected each week.  Plots were randomly selected for each time point using R (version 3. 4. 0).  Gloves were worn at all times and changed between treatments.  Samples were collected in order of cleanness,  starting with the UV-treated river water irrigation/conventional soil treatment (D > A > C > B).  Lettuces were always harvested before soil to avoid contamination of lettuce with soil.  All samples were cooled during transport.  Lettuce and soil samples were processed within 8 h,  water samples within 36 h. Lettuce harvestingA stainless steel knife was used for harvesting and cleaned with 80% ethanol between lettuce heads.  Per biological replicate,  eight lettuce heads were cut ~1 cm above the soil,  excluding the lower leaves.  Special care was taken to ensure that the lettuce was not contaminated with soil during sampling.  The lettuce leaves were directly transferred into sterile plastic bags for subsequent analysis.  The plastic bags were sealed,  including some air to minimize squeezing of lettuce leaves during transport. Soil core collectionAfter lettuce heads were harvested,  soil samples were collected from the top 10 cm of soil within 10 cm distance of the harvested lettuce heads using a soil corer (2. 5 cm diameter).  Between treatments,  the soil corer was cleaned and disinfected with 80% ethanol.  Per biological replicate,  eight soil cores were pooled in a sterile plastic bag and mixed thoroughly. Irrigation water samplingBefore sampling the untreated river water,  5-l sampling bottles were disinfected with 80% ethanol and rinsed twice with the water to be sampled.  Additionally,  two sterile 1-l bottles were filled with UV-disinfected river water by drawing off water at the outflow of the UV disinfection device. Sample processingLettuce leaf washesThe plastic bags containing the lettuce leaves were shaken gently to ensure proper mixing.  Eighty grams per replicate were weighed into sterile plastic containers and gently transferred into a plastic bottle containing 720 ml of phosphate buffered saline (PBS;  8 g NaCl,  0. 2 g KCl,  1. 15 g Na2HPO4,  and 0. 2 g KH2PO4 in 1 l distilled water;  pH 7. 3).  The bottles were then sonicated for 7 min,  rotating them after 3. 5 min to ensure homogenous sonication.  The resulting leaf wash was sieved into sterile glass bottles to avoid further soaking and disruption of the lettuce.  Notably,  this procedure was chosen to recover surface-attached bacteria while omitting internalized bacteria,  since internalization from contaminated field soil has been described to be small or absent (Detert and Schmidt),  ARG numbers detected in root and leaf endophytes have been found to be markedly lower than those on the leaf surface (Zhang et al. ),  and internalized pathogens seem to persist only in the short term (Erickson).  A 30-ml aliquot of each leaf wash was stored on ice for later cultivation.  For DNA extraction,  200 ml of leaf wash were filtered through polycarbonate (PC) filters (0. 2 µm pores,  47 mm diameter;  Whatman plc,  Buckinghamshire,  UK).  Each filter was then carefully transferred into DNeasy PowerWater Kit bead tubes (Qiagen,  Venlo,  Netherlands),  top side facing inwards.  Finally,  the DNA tubes were shock frozen in liquid nitrogen and stored at \\x88\\x9280°C until further processing. Soil and manure processingThe plastic bags containing the soil samples were shaken gently to ensure proper mixing.  Soil was sieved (mesh size 2 mm) onto clean aluminum foil,  and the sieve was cleaned between samples using 80% ethanol.  Thereof,  20 g (wet weight) were dissolved in 180 ml of PBS,  vigorously shaken for 1 min,  and allowed to settle for 5 min.  The supernatant was transferred into 100 ml glass bottles and left to settle for another 5 min.  The supernatant thereof was stored at 4°C for cultivation.  Finally,  three times 250 mg of soil (wet weight) were aliquoted into DNeasy PowerSoil Kit bead tubes (Qiagen),  shock frozen,  and stored at \\x88\\x9280°C for later DNA extraction.  Manure samples (only sampled in week 0 = initial condition Z) were treated in the same manner as soil samples. Irrigation water filteringFor bacterial culturing,  200 ml of river or UV-treated river water were filtered in duplicates through cellulose acetate (CA) filters (0. 2 µm pores,  47 mm diameter;  Sartorius AG,  GÃ¶ttingen,  Germany).  For DNA extraction,  700 ml of water were filtered using two PC filters (300 ml and 400 ml,  respectively),  to avoid filter clogging.  The filters were transferred into DNeasy PowerWater Kit bead tubes,  shock frozen,  and stored at \\x88\\x9280°C. Bacterial cultivationFor determination of total heterotrophic bacteria,  R2A agar was used (Merck Millipore,  MA,  USA).  ARB were cultured using commercially available media.  BrillianceTM ESBL and CRE agar (ESBL and CRE;  Thermo Fisher Scientific,  Waltham,  MA,  USA) were used for isolation of ESBL-producing and CR E.  coli and Klebsiella spp.  Fluoroquinolone-resistant E.  coli or Klebsiella spp.  were cultured on BrillianceTM E.  coli/coliform Selective Agar (CM;  Thermo Fisher Scientific) or Simmons Citrate Agar with 1% (g/g) myo-inositol (SCA;  Sigma-Aldrich,  St.  Louis,  MI,  USA) supplemented with ciprofloxacin (1 mg/l),  respectively.  BrillianceTM VRE agar (VRE;  Thermo Fisher Scientific) was used for isolation of VRE.  Finally,  Acinetobacter spp.  and Pseudomonas spp.  were cultured on Leeds Acinetobacter Medium (LEE;  Hardy Diagnostics,  Santa Maria,  CA,  USA) and Cetrimide Agar with 1% (g/g) glycerol (CTM;  Thermo Fisher Scientific),  respectively. From the prepared leaf and soil washes,  100 µl of each sample were spread-plated in duplicate on each selective agar type.  For R2A,  10-fold dilutions in PBS were prepared and the most appropriate two dilutions were plated.  For water,  duplicate CA filters were transferred to each agar directly after water filtering.  All plates were incubated at 37°C for 24 h,  except for LEE (35°C,  24 h) and R2A (25°C,  up to 5 days).  Thereafter,  plates were inspected for growth and when no growth was visible,  CRE,  VRE,  and SCA plates were incubated for another 24 h according to instructions.  Total colonies as well as target colonies displaying coloration typical for each selective medium were enumerated. MALDI-TOF biotypingRandomly selected bacterial colonies displaying typical morphology as well as secondary colonies were identified by Matrix-Assisted Laser Desorption/Ionization Time-of-Flight (MALDI-TOF) biotyping with the direct smearing technique described previously (Gekenidis et al. ).  Identifications with a score above 2. 0 were considered reliable on species level while identifications with a score between 1. 7 and 2. 0 were considered reliable on genus level,  as recommended. DNA extraction,  amplicon,  and shotgun metagenomic sequencingTotal DNA was extracted from the water- and leaf-wash-filters using the DNeasy PowerWater Kit,  while the DNeasy PowerSoil Kit was used on the prealiquoted soil and manure samples,  using the kit best adapted to each sample type in order to maximize DNA yield and quality.  Both kits included a harsh lysis step using PowerBead tubes and specially formulated cell lysis buffers.  All DNA extracts were checked using a Quant-iTTM High-Sensitivity dsDNA Assay Kit on a Qubit 3. 0 Fluorometer (concentration measurement;  Thermo Fisher Scientific) as well as a NanoDrop One Spectrophotometer (purity measurement;  Thermo Fisher Scientific). To study the microbial communities,  Illumina MiSeq® sequencing of the 16S rRNA gene V3\\x80\\x93V4 hypervariable regions was performed,  using universal primers 341F and 805R (Herlemann et al. ).  DNA extracts from soil and lettuce were diluted to a final concentration of 5 ng/µl,  while low-concentrated DNA extracts from water were concentrated on an Eppendorf Concentrator plus centrifuge (Eppendorf,  Hamburg,  Germany).  Sequencing libraries were produced following Illumina\\x80\\x99s 16S Metagenomic Sequencing Library Preparation guide,  using Phusion Hot Start II High-Fidelity Master Mix (Thermo Fisher Scientific) for amplification.  The libraries were pooled for paired-end sequencing on a MiSeq v3 cartridge (2 Ã\\x97 300 bp;  Illumina,  San Diego,  CA,  USA). For resistome analysis,  samples from weeks 0,  1,  and 5 were selected.  Biological replicates were pooled and metagenomic libraries were prepared using a NEBNext® Ultra\\x84¢ II FS DNA Library Prep Kit for Illumina (New England Biolabs,  Ipswich,  MA,  USA),  following the high-input protocol including size selection.  First,  enzymatic digestion was performed during 10 min.  After library preparation,  fragment size distribution was checked on a D1000 ScreenTape System (Agilent Technologies,  Santa Clara,  CA,  USA).  Samples were barcoded using NEBNext Multiplex Oligos for Illumina (Unique Dual Index Primer Pairs;  New England Biolabs),  adjusted to 4 nM,  and pooled equimolarly.  The pool was sequenced on a NovaSeqTM 6000 System (2 Ã\\x97 250 bp,  SP-type flow cell;  Illumina). BioinformaticsAmplicon sequencing: microbial community data analysisThe demultiplexed 16S rRNA gene amplicon sequences were processed using USEARCH (detailed protocol: AmpSeq_DataPrepReport,  see online Supplementary material).  Briefly,  in a first step,  the 3\\x80² ends of the reads were cut to improve the subsequent merging.  In a next step,  the primer regions of the merged reads were removed.  Before clustering,  size selection and quality filtering were performed.  The amplicon sequences were then clustered into operational taxonomic units (OTUs) with 97% identity or into zero-radius zOTUs.  In addition,  the zOTUs were clustered at an identity of 99%,  98% and 97%.  The SILVA SSU v128 database was used as a reference for taxonomic predictions.  After annotation,  clusters classified as mitochondria or chloroplasts were excluded from downstream analysis,  while Archaea were retained for downstream analysis (detailed read statistics: AmpSeq_ReadReport,  see online Supplementary material). Microbial community compositions were investigated using the web-application Calypso (v8. 84) (Zakrzewski et al. ) and R (v4. 0. 2) implemented in RStudio (RStudio Team. ).  Upon import into Calypso,  data was filtered as recommended by allowing only samples with more than 1000 sequence reads and taxa with less than 0. 001% relative abundance across all samples,  including maximally the top 20\\x80\\x89000 taxa (filtered by mean).  No samples were lost by the applied filtering.  Additionally,  cumulative-sum scaling with log2-transformation (Paulson et al. ) was applied to normalize the data.  All downstream Calypso tools were used with default settings,  unless otherwise indicated.  Principal coordinates analysis (PCoA) was conducted for visualization in RStudio. Metagenomics: resistome analysisTwo major approaches were used to analyse metagenomics read data: a read-based analysis to detect resistance genes after minimal data processing and thereby least information loss,  and an assembly-based analysis to explore the genetic context of resistance genes including potential for genetic mobility. Read-based analysisFirst,  resistance genes were annotated to minimally processed reads using two reference databases.  To do so,  Illumina adapters were removed from raw paired-end reads using Cutadapt (v1. 15) (Martin),  allowing 10% errors.  Sequences were then trimmed with sickle (v1. 33;  paired-end,  \\x80\\x93qual-type Sanger) using a minimum quality score per window of 20 and a length threshold of 20 (Joshi and Fass).  The obtained trimmed reads were processed with DeepARG-SS (identity: 70%,  e-value: 1e\\x88\\x9210,  target coverage: 60% and probability: 0. 7) (Arango-Argoty et al. ) and ARGs-OAP (v2. 0) (Yin et al. ) for ARG annotation.  The resulting ARG tables were analysed using Microsoft Excel and RStudio (summary tables,  PCoA,  Venn diagram).  PCoA was performed on 16S-normalized data using the ARGs-OAP v2. 0 pipeline in RStudio. Assembly-based analysisSecond,  raw reads were assembled into contigs,  open-reading frames (ORFs) predicted,  and resistance genes and genetic mobility elements annotated using several reference databases (detailed protocol: MetaSeq_DataPrepReport,  see online Supplementary material).  Briefly,  the raw paired-end reads were first processed to remove adapters,  low complexity and low quality reads,  correct overlaps of at least 30 bp,  and discard reads shorter than 100 bp.  Lactuca sativa (chromosome and chloroplast) DNA was then removed from all samples.  The obtained clean reads were assembled,  and protein-coding genes were predicted while retaining only ORFs longer than 50 amino acids.  The obtained ORFs were searched against the profile hidden Markov model databases Pfam,  Resfams (Gibson et al. ),  and TIGRFAMs (Haft et al. ),  as well as the BacMet database (Pal et al. ) containing antibacterial biocide and metal resistance genes (BMRGs).  Only hits with bit score >50 and target coverage >60% were kept.  All results tables were filtered to retain only the best annotation for each ORF.  TIGRFAMs hits were additionally filtered using a list of keywords (transposase,  transposon,  conjugative,  integrase,  integron,  recombinase,  conjugal,  mobilization,  recombination,  and plasmid) to retain annotations related to MGEs only,  as described by Forsberg et al..  Finally,  plasmid tags were annotated.  For better inter sample comparison,  all counts were normalized by number of reads per sample to obtain counts per one million reads (CPM). A Circos plot for visualization of ARG distribution among environments was produced using a freely available online tool (Krzywinski et al. ).  MDR contigs were defined as contigs carrying three or more ARGs assigned from the Resfams database.  Nonmetric multidimensional scaling (NMDS) of all samples was conducted in RStudio using Bray\\x80\\x93Curtis distance measure and dimensions k = 3. All R scripts used for data analysis are available from GitHub (https://github. com/maria-gekenidis/lettuce-resistome).  The main R-packages used include ape,  DESeq2,  labdsv,  microbiome,  phyloseq,  phyloseq. extended,  plotly,  and vegan. Statistical analysisFor culture data analysis,  bacterial counts were log10-transformed and statistically compared to detect an effect of time or treatment using two-way analysis of variance (two-way ANOVA) and adjusting P-values using Tukey\\x80\\x99s or Å Ã\\xaddÃ¡k\\x80\\x99s multiple comparison tests (GraphPad PRISM 8,  GraphPad Software Inc.,  San Diego,  CA).  Where no visible growth was observed,  half the limit of detection (LOD/2) was used.  Statistical significance was assigned at P <. 05. In microbial community data analysis,  differentially abundant (DA) taxa on lettuce were identified using Calypso at different taxonomic levels by pairwise comparisons {Student\\x80\\x99s t-test;  P-values adjusted for multiple testing using Benjamini\\x80\\x93Hochberg [False Discovery Rate (FDR)]-correction}.  Additionally,  the top 10 DA taxa at each level were identified using the R-package DESeq2 (FDR-adjusted P <. 05).  Alpha-diversity of lettuce communities was estimated in Calypso using different measures including Chao1,  Shannon Diversity index,  and Inverse Simpson\\x80\\x99s index (measuring species richness,  richness and evenness,  and dominance,  respectively),  and significant differences between time points or between treatments were detected by one-way ANOVA.  Beta-diversity of lettuce communities was measured by Bray\\x80\\x93Curtis,  Jaccard,  Unifrac,  and weighted Unifrac distances (each taking into account presence/absence,  taxa abundance,  and/or phylogeny) and compared between time points and between treatments in RStudio by permutational analysis of variance (PERMANOVA) with 9999 permutations,  for diversity measures with homogeneous multivariate dispersions among groups.  All P-values were adjusted using FDR-correction. To quantify the extent to which the observed resistome profiles in lettuce correlated with the respective microbial community structure,  Procrustes analysis was performed using ResistoXplorer (default parameters;  Dhariwal et al. ) and RStudio.  For the microbiomes,  zOTUs clustered at 99% and annotated with the SILVA database were used as input,  after removing samples not represented in the resistome data,  rarefying to the lowest depth,  and merging biological replicates by summing in order to match the sample structure of the resistome dataset.  For the resistome,  abundance tables based on Resfams/BacMet annotations normalized by CPM were used. Data accessibilityThe Illumina MiSeq raw paired-end reads (ERR4552817\\x80\\x93ERR4552843;  ERR4552847\\x80\\x93ERR4552882,  ERR4552890\\x80\\x93ERR4552950,  ERR4554840\\x80\\x93ERR4554892,  and ERR4555363\\x80\\x93ERR4555379) were submitted to the European Nucleotide Archive (ENA) under project number PRJEB36754. The Illumina NovaSeq\\x84¢ raw paired-end reads (ERR3929355\\x80\\x93ERR3929377) as well as the trimmed reads (ERR3943975\\x80\\x93ERR3943997) were deposited in ENA under project number PRJEB36754. ', 'results': 'ResultsCulturing and species identificationHeterotrophic bacteria and bacterial counts on antibiotic-containing media show temporal variation on lettuce and a strong impact of UV-treatment in waterFirst,  the change in number of total heterotrophic bacteria on young compared to mature lettuce was assessed by counting colony forming units (CFU) on R2A plates for lettuce of all treatments harvested at weeks 1 and 5,  but no systematic differences were observed.  All counts were between 3 Ã\\x97 105 and 2 Ã\\x97 106 CFU/g.  In soil,  counts were significantly higher in week 5 than in week 1 for all treatments,  increasing on average from 1 Ã\\x97 106 ± 3 Ã\\x97 105 to 4 Ã\\x97 106 ± 5 Ã\\x97 105 CFU/g wet weight (A: P <. 0001,  B: P =. 0019,  C: P =. 0015,  and D: P =. 0006).  Finally,  as expected UV-treatment resulted in a reduction of total heterotroph counts (week 1: from 4 Ã\\x97 103 ± 1 Ã\\x97 103 to 9 Ã\\x97 102 ± 7 Ã\\x97 102 CFU/100 ml,  ns;  week 5: from 7 Ã\\x97 105 ± 5 Ã\\x97 105 to 7 Ã\\x97 103 ± 4 Ã\\x97 103 CFU/100 ml,  P =. 0029). In a next step,  bacterial counts on chromogenic plates selective for ESBL-producing Enterobacteriaceae,  CRE,  and VRE were determined by plating and counting total CFU.  Additionally,  bacterial colonies displaying coloration typical of the target bacteria for each selective plate were enumerated as presumptive target bacteria,  since typical coloration may be displayed by nontarget bacteria as well.  Bacteria targeted by ESBL and CRE agar may include the Klebsiella spp.,  Enterobacter spp.,  Serratia spp.,  and Citrobacter spp.  (KESC) group as well as E.  coli.  In the following,  these counts are collectively referred to as ARB counts,  although intrinsically resistant bacteria can grow on the mentioned selective plates when plating environmental samples. Total counts from lettuce on ESBL agar significantly increased over time for all four treatments on average from 3 Ã\\x97 104 ± 1 Ã\\x97 104 to 3 Ã\\x97 105 ± 1 Ã\\x97 105 CFU/g (A[w1 versus w6]: P =. 0001,  B[w1 versus w6]: P =. 0385,  C[w1 versus w6]: P =. 0056,  and D[w1 versus w5]: P =. 0014),  while no clear trend was observed for either presumptive KESC or presumptive E.  coli (data not shown).  In soil,  all counts were stable over time with only a few exceptions (data not shown).  Finally,  UV-treatment of river water always significantly reduced bacterial counts on ESBL agar (total: 4 Ã\\x97 101\\x80\\x935 Ã\\x97 102 CFU/100 ml;  presumptive KESC: 2 Ã\\x97 10\\x88\\x921\\x80\\x934 Ã\\x97 100 CFU/100 ml;  presumptive E.  coli: 2 Ã\\x97 10\\x88\\x921\\x80\\x932 Ã\\x97 101 CFU/100 ml) below detection. As for total counts on ESBL agar,  total counts from lettuce on CRE agar consistently increased over time for all four treatments from 2 Ã\\x97 104 ± 4 Ã\\x97 103 to 3 Ã\\x97 105 ± 8 Ã\\x97 104 CFU/g on average (A[w1 versus w6]: P <. 0001,  B[w1 versus w6]: P <. 0001,  C[w1 versus w6]: P =. 0060,  and D[w1 versus w6]: P <. 0001).  A similar trend was observed for presumptive KESC counts though not significant,  while presumptive E.  coli were mainly detected in weeks 1 and 2,  and were mostly below detection thereafter (data not shown).  In soil,  all counts fluctuated with no significant differences (data not shown).  Finally,  UV-treatment of river water reduced bacterial counts on CRE agar (total: 6 Ã\\x97 101\\x80\\x931 Ã\\x97 103 CFU/100 ml;  presumptive KESC: 3 Ã\\x97 100\\x80\\x931 Ã\\x97 102 CFU/100 ml;  and presumptive E.  coli: 6 Ã\\x97 10\\x88\\x921\\x80\\x937 Ã\\x97 100 CFU/100 ml) by up to nearly three log units. As for ESBL and CRE plates,  total counts from lettuce on VRE plates increased with time on average from 2 Ã\\x97 103 ± 2 Ã\\x97 102 to 1 Ã\\x97 105 ± 5 Ã\\x97 104 CFU/g (A[w1 versus w5]: P =. 0039,  B[w1 versus w5]: P <. 0001,  C[w1 versus w5]: P =. 0185,  and D[w1 versus w4]: P < XXX).  In soil,  total VRE counts were mostly stable over time with a few exceptions (data not shown),  while UV-treatment of river water significantly reduced total VRE counts (2 Ã\\x97 100\\x80\\x937 Ã\\x97 101 CFU/100 ml) to below detection (except week 1: reduction to 1 Ã\\x97 101 ± 4 Ã\\x97 100 CFU/100 ml).  Notably,  colonies displaying coloration typical of the target bacteria on VRE agar were never detected. Overall,  significant increases in ARB numbers were observed for lettuce over time within treatments,  while significant differences between treatments of the same time point were rare and with no discernable patterns.  In soil,  ARB numbers were usually stable or fluctuated without correlation to time or treatment.  However,  UV-treatment of river water mostly significantly reduced total as well presumptive bacterial counts on all three antibiotic-containing media. MALDI biotyping reveals rare occurrence of clinically relevant ARBTo further investigate bacterial composition on the chromogenic selective agar plates,  representative colonies displaying typical coloration as well as secondary colonies were identified at genus and species level using MALDI biotyping (Tables S2\\x80\\x93S6). ESBL agar plates were dominated by Pseudomonas spp.  Additionally,  Enterobacter cloacae was detected once on lettuce grown on conventional soil with UV-treated river water irrigation (week 2),  E.  coli once in river water (week 4),  and Serratia fonticola three times in river water (weeks 1\\x80\\x933;  Table S2).  Notably,  E.  coli was also identified in river water (week 4) and in manure on ciprofloxacin-supplemented CM plates (data not shown).  Of the species targeted by CRE agar,  only Klebsiella oxytoca was detected on lettuce grown on manure-amended soil irrigated with river water (week 1;  Table S3).  On the other hand,  no ciprofloxacin-resistant Klebsiella spp.  were identified on SCA plates (data not shown). On VRE agar plates,  Enterococcus spp.  could not be identified at any time,  while Sphingomonas spp.,  Sphingobacterium spp.,  and Pseudomonas spp.  were the most frequent genera (Table S4).  Finally,  on CTM and LEE agar all identified species belonged to the respectively targeted genera Pseudomonas spp.  and Acinetobacter spp.  (Tables S5 and S6).  Notably,  the clinically important species P.  aeruginosa was detected only sporadically and A.  baumannii not at all.  We therefore assumed that lettuce is not an important source for these species and did not investigate their resistance potential further.  Of importance,  other species of the genera Pseudomonas and Acinetobacter are also known as reservoirs of antibiotic resistance,  including for example P.  putida and P.  fluorescens (Sabour et al. ),  or A.  bereziniae and A.  johnsonii (Sheck et al. ) detected by culturing in this study. Microbiome analysisMicrobial communities show temporal shifts in lettuce and a strong impact of UV-treatment in waterTo complement culture-based findings,  bacterial communities of lettuce,  soil,  water,  and manure were characterized by sequencing of 16S rRNA gene fragments amplified from community DNA.  On the leaves of lettuce seedlings before planting,  the prominent phyla were Parcubacteria followed by Proteobacteria,  Planctomycetes,  and Actinobacteria (Fig.  1A,  week 0).  One week after planting,  the latter three phyla were still dominant,  while Parcubacteria were almost undetectable (Fig.  1A,  week 1).  Additionally,  relative abundance of Chloroflexi had increased compared to week 0.  As the plants grew,  Proteobacteria became increasingly dominant until they accounted for more than 90% of the community in most samples by week 5.  Finally,  6 weeks after planting,  the communities most resembled 1-week-old lettuce communities,  with overall a striking decrease in the relative abundance of Proteobacteria but increased Acidobacteria,  Actinobacteria,  Chloroflexi,  and Planctomycetes as compared to week 5 (Fig.  1A,  week 6).  In contrast to this clear shift in communities over the growth period,  no systematic differences were detected between the different treatments within each time point.  Only notable exception was treatment A of week 6,  with more than double the relative abundance of Proteobacteria compared to the other three treatments (Fig.  1A). In manure samples,  Proteobacteria dominated with a relative abundance of more than 90% (Fig.  1B,  panel M).  On the other hand,  soil communities were very stable over time regardless of the applied treatment,  and displayed Planctomycetes as dominant phylum,  followed by Chloroflexi,  Acidobacteria,  Proteobacteria,  and Actinobacteria (Fig.  1B).  Notably,  soil communities overall most resembled lettuce communities of weeks 1 and 6 (Fig.  1A). The microbial communities of the water samples were more diverse,  even within time points (Fig.  1C).  The dominant phyla in both untreated and UV-treated river water included Actinobacteria,  Bacteroidetes,  Parcubacteria,  and Proteobacteria.  In weeks 0,  5,  and 6,  no clear effect of UV-treatment was detected,  while in weeks 1\\x80\\x934,  UV-treated river water contained approximately twice as many Proteobacteria as untreated river water,  accounting for nearly 100% in most cases (Fig.  1C).  Notably,  in the weeks with a clear effect of UV-treatment (weeks 1\\x80\\x934),  Proteobacteria were the dominant phylum in the untreated river water already (that is,  before UV-treatment),  while in the weeks with no clear effect of UV-treatment,  there was not one phylum clearly dominating the microbial community of untreated river water. Clinically relevant bacterial families on lettuce show temporal variations in relative abundanceFor a more detailed analysis of lettuce communities,  relative abundance of the top 50 bacterial families was visualized in a heat map (Fig.  2).  A temporal shift in the relative abundance of families targeted in the culture-based approach (Enterobacteriaceae,  Moraxellaceae,  and Pseudomonadaceae in cluster-3),  characterized by an increase in relative abundance from medium to high from weeks 1 to 2,  followed by a steadily high relative abundance through week 5,  and a final steep decrease in relative abundance in week 6.  Typical soil bacteria such as Tepidisphaeraceae,  Caldilineaceae,  and Intrasporangiaceae (cluster-1) or Gemmatimonadaceae,  Nitrosomonadaceae,  Rhodospirillaceae,  Cytophagaceae,  and Verrucomicrobiaceae (cluster-2) (Huang et al.,  Corrochano-Monsalve et al.,  Hannula et al. ) were abundant from planting until week 2,  after which their relative abundance decreased until week 5,  and was increased again in the last sampling (week 6).  Important to note is that optically,  the lettuces from week 5 (optimal harvest point) markedly differed from the oversized and partially already decaying lettuces from week 6.  Further of note,  treatment A of week 6 markedly differed from the other treatments of week 6 (Fig.  2),  as already observed in the relative abundance bar plots (Fig.  1A).  Finally,  the relative abundance of target bacteria on lettuce was visualized in dot plots and at different taxonomic levels from genus to class (Fig.  S3).  At all levels,  the observed differences between sampling time points were highly significant (one-way ANOVA,  P-values in Fig.  S3).  For Î³-Proteobacteria,  including the target groups Enterobacteriaceae,  Pseudomonas spp.,  and Acinetobacter spp.,  a common trend of increasing relative abundance from weeks 0 to 5 with a drastic decrease in week 6 was observed at all taxonomic levels.  No clear temporal trend was observed for Enterococcus spp.,  except at class level where relative abundance of Bacilli increased during the first 2 weeks but dropped below initial levels in week 6.  The class Bacilli,  however,  also includes many nontarget bacteria.  When comparing treatments instead of time points,  no significant differences were detected (one-way ANOVA,  P \\x89¥. 01;  data not shown). Differential abundance analysis confirms increased abundance of clinically relevant taxa on mature lettuceNext,  the impact of lettuce age,  irrigation water quality,  and soil amendment on the abundance of taxa including clinically relevant bacteria on lettuce was explored.  DA taxa were identified among the 200 most abundant taxa at different taxonomic levels under three conditions: (1) young versus mature lettuce [week 1 (first samples after planting) versus week 5 (optimal harvest point)],  (2) lettuce irrigated with untreated versus UV-treated river water,  and (3) lettuce grown with or without manure was investigated.  Several of the target taxa were significantly more abundant on mature compared to young lettuce of all treatments (Fig.  3A),  with fold changes ranging from 1. 4 to 4. 6 (Table S7).  Of these,  Enterobacteriaceae,  Pseudomonadaceae,  and Moraxellaceae were among the top 10 families with increased abundance on mature compared to young lettuce (Fig.  S4).  On the other hand,  manure application or irrigation water quality resulted in only few DA taxa on lettuce,  never including any of our target taxa of clinical relevance (data not shown).  Of note,  species of the genera Pseudomonas and Acinetobacter other than our target species were not detected among these DA taxa. Diversity in lettuce microbial communities decreasing over time,  but no treatment effectMicrobial species richness,  evenness,  and dominance in lettuce communities were estimated with the Chao1,  Shannon Diversity,  and Inverse Simpson\\x80\\x99s index.  A significant decrease was observed from weeks 0 to 5 (Fig.  S5A).  In the last sampling (week 6),  all three indices increased again approximately to initial levels.  This development was highly significant at all taxonomic levels (one-way ANOVA,  P <. 001) and coincides with the gradual increase and steep decrease of Proteobacteria proportions observed over time (Fig.  1A).  Consistent with our observations when investigating individual taxa,  there was usually no significant difference between treatments (one-way ANOVA,  P \\x89¥. 05;  Fig.  S5B).  In the marginally significant cases,  initial condition Z showed increased diversity compared to treatments A\\x80\\x93D. Structural differences between microbial communities were measured considering taxa abundance and/or phylogeny (Jaccard,  Bray\\x80\\x93Curtis,  Unifrac,  and weighted Unifrac).  Since all measures yielded similar results,  only Bray\\x80\\x93Curtis and Unifrac are presented.  PCoA visually confirmed an effect of time on lettuce community structures: a time shift can be observed as well as clear separation of freshly planted from mature lettuce (week 1 versus week 5;  Fig.  3B),  with lettuce seedlings (week 0) in between the two.  Lettuces of week 6 clustered tightly and were more similar to week 1 lettuces.  When looking at all investigated systems jointly,  it becomes evident that week 1/6 lettuce communities resemble soil communities in their structure (Fig.  S6).  For diversity-measures with homogeneous multivariate dispersions among groups,  beta-diversity was significantly different among all time points with exception of pair week 3/4 (Table S8).  In contrast,  initial condition Z only marginally differed from the four treatments,  becoming nonsignificant after P-value adjustment (Table S9). Resistome analysisCore ARGs detected in all environments mainly belong to class of multidrug ARGsThe antibiotic resistome of lettuce,  soil,  water,  and manure was investigated.  In total,  326 ARG subtypes were identified.  An overview of the number of ARG subtypes [grouped by antibiotic (AB) class] detected in the four environments is shown in Fig.  4(A).  The highest diversity of ARGs was detected in lettuce samples (n = 256),  followed by water (n = 141),  soil (n = 102),  and manure (n = 95).  In terms of shared ARGs,  lettuce shared most ARGs with water and second most with soil (n = 126 and n = 94,  respectively;  Fig.  4B).  All four environments had 30 ARGs in common (core ARGs;  Table 1),  most of which belonged to the class of multidrug ARGs encompassing mainly genes encoding efflux pumps.  Further of interest,  aminoglycoside (aadA,  aph(6)-I),  Î²-lactam (metallo-Î²-lactamase),  glycopeptide (vanR,  vanX),  sulfonamide (sul2),  and tetracycline (tetA) ARGs were among the core ARGs.  The detailed lists of ARGs shared between lettuce and the other three environments are provided in Tables S10\\x80\\x93S12.  Among the ARGs uniquely shared between lettuce and water were mostly MDR and Î²-lactam ARGs (Table S10),  whereas lettuce and soil uniquely shared mainly glycopeptide and tetracycline ARGs (Table S11),  and only four ARGs were unique to lettuce and manure (Table S12).  The distribution of ARGs by class among environments was very homogeneous,  with efflux-related ARGs being the most prominent in all environments,  followed by Î²-lactam related ARGs (Fig.  S7). Sample clustering by ARG profiles reveals impact of manure application on lettuce resistomeSamples were clustered using their ARG profiles with Bray\\x80\\x93Curtis (accounting for ARG presence/absence as well as abundance) or Jaccard (accounting for ARG presence/absence only) as distance measures,  to visualize sample similarities.  Soil samples all clustered very tightly,  irrespective of irrigation regime or amendment (Fig.  5).  In young lettuce (week 1),  a clear separation between treatments A/D and B/C was evident (without and with manure,  respectively),  which was more pronounced when only ARG presence/absence was considered (Fig.  5B).  A similar tendency was observed in mature lettuce (week 5),  though much less pronounced.  UV-treatment of water lead to a clear separation that was more pronounced when considering ARG presence/absence only.  Finally,  manure clustered closest to young lettuce grown in manure-amended soil (Fig.  5B). MDR proportions are increased manifold in mature lettuce and UV-treated river waterThe effect of lettuce growth time,  manure application,  and UV-treatment on the proportions of MDR contigs (number of MDR contigs per total contigs) was investigated (Fig.  6A).  MDR contig proportions in soil were very low irrespective of treatment (<0. 15%),  while manure had a proportion of roughly 2%.  Of all samples,  the highest proportions of MDR contigs were found in mature lettuce grown on manure-amended soil (L5B and L5C) and in UV-treated river water (W1B and W5B).  Compared to untreated water,  UV-treated water had at least four times increased MDR proportions,  and mature lettuce samples had at least 2-fold increased MDR proportions compared to young lettuce samples (Fig.  6A),  the difference being most pronounced when grown with manure. Increased abundance of antibacterial BMRGs in mature versus young lettuceTo investigate whether\\x80\\x94additionally to antibiotic resistance\\x80\\x94biocide and metal resistance contributed to differences between environments and treatments,  BMRGs were annotated (see online Supplementary material for details).  Overall,  in all four environments the most abundant BMRG [counts per one million reads (CPM)] was tupC,  conferring resistance to the metal tungsten (W),  immediately followed by znuC/yebM,  encoding zinc (Zn) resistance.  Third most frequent was wtpC in lettuce,  water,  and manure and modC in soil,  both conferring tungsten- and molybdenum (Mo) resistance.  The top three BMRGs best discriminating young from mature lettuce when comparing average CPM values were fbpC [iron (Fe) and gallium (Ga) resistance],  wtpC,  and adeL [sodium dodecyl sulfate (SDS)/ethidium bromide/safranin O/acridine orange resistance],  all detected at much higher abundance in mature than in young lettuce.  In young lettuce,  treatments A/D (no manure) could best be discerned from treatments B/C (manure) by adeL,  fbpC,  and zraR/hydH (Zn resistance),  the two latter of which were among the top ten BMRGs detected in manure. Lettuce BMRG-profiles reveal high coresistance to antibiotics,  and confirm effect of growth time and manure applicationThe frequency of antibiotic\\x80\\x93biocide/metal coresistance per sample was estimated,  that is,  the proportion of contigs carrying both ARGs and BMRGs in all contigs of a given sample (Fig.  6B).  On average,  mature lettuce had a higher proportion of contigs conferring coresistance to antibiotics\\x80\\x93biocide/metals compared to young lettuce (38. 9 ± 1. 0% versus 31. 1 ± 1. 0%).  Soil had on average the lowest proportion of coresistance (28%).  In UV-treated river water,  average coresistance was higher than in untreated river water (36. 9 ± 0. 7% versus 30. 1 ± 2. 0%). A clustering using NMDS was performed on all samples based on Resfams and/or BacMet annotations,  to compare sample clustering when considering antibiotic or biocide/metal resistance either separately or jointly.  The clustering was very similar whether considering either ARGs or BMRGs,  which can be expected in case of a significant co- and cross-resistance to these compounds (clustering not shown).  Overall,  lettuce samples clustered as already observed in the PCoA analysis,  that is,  according to sampling time while within time clusters treatments A/D or B/C were more similar to each other. Genetic mobility analysis shows high prevalence of Proteobacteria-associated resistance plasmids and dominance of multidrug/efflux ARGs in all samplesSince the transferability of resistance between bacteria is a major concern,  the genetic context of ARGs and BMRGs was investigated by looking for proximal mobility elements.  A mobility incidence (M%) per sample was calculated for ARGs,  BMRGs,  and their combination,  defined as the percentage of resistance gene-encoding contigs flanked with at least one mobility indicator in all resistance contigs (Ju et al. ).  Generally,  BMRG mobility incidence was higher than ARG mobility incidence (Fig.  6B),  especially in lettuce.  On the other hand,  contigs carrying both ARGs and BMRGs were always the less mobile of the three.  In lettuce,  mature plants displayed higher average mobility incidence than young plants,  and within time points,  treatments A and D (no manure) had the lowest mobility incidence.  Soil samples had very low mobility incidences overall (Fig.  6B),  just as was observed for proportions of MDR contigs (Fig.  6A).  Finally,  UV-treatment of water clearly increased mobility incidence of all\\x80\\x94ARGs,  BMRGs,  and their combination\\x80\\x94as compared to untreated river water. The MGEs most typically linked to BMRGs were the tyrosine recombinases xerC and xerD,  the recombination mediator recR,  and the conjugative transfer ATPase cagE.  These four jointly made up more than half (57. 2%) of all detected MGEs.  Interestingly,  the above-mentioned four MGEs were likewise the most frequently linked to ARGs,  again making up more than half (51. 3%) of all detected MGEs. Plasmid tag annotation using PlasFlow revealed proportions of ARG plasmid contigs in all ARG contigs in the different samples (Fig.  7A).  On average,  mature lettuce had higher ARG plasmid proportions than young lettuce (roughly 10%\\x80\\x9312% versus 6%\\x80\\x938%).  The ARG plasmid proportion of manure was comparable to mature lettuce,  while ARG plasmid proportions of soil (whether containing manure or not) were in the range of young lettuce (roughly 6%\\x80\\x938%).  ARG plasmid content of water varied from below 8% to 14%,  while no clear difference between untreated and UV-treated and river water was detectable.  Looking at ARG plasmid phylogeny,  the phylum of Proteobacteria prevailed in all samples (Fig.  7B).  Mature lettuce had higher proportions of Proteobacteria ARG plasmids compared to young lettuce (at least 80% versus around 60%).  On the other hand,  ARG plasmids from Firmicutes were more prominent in young than on mature lettuce.  In manure,  about 35% of ARG plasmids were from Proteobacteria,  followed by about 14% of ARG plasmids from Firmicutes,  while about half the ARG plasmids remained unclassified.  In soil,  ARG plasmids from Proteobacteria consistently prevailed (\\x88¼60%),  followed by much lower proportions of Firmicute and Actinobacteria plasmids.  In water as well,  ARG plasmids from Proteobacteria were most often identified.  While additionally,  ARG plasmids from Firmicutes were clearly detected in untreated river water,  almost none were present in UV-treated river water (about 4% versus 0. 2%).  Further,  ARG plasmids from Bacteroidetes,  Cyanobacteria,  and Spirochaetes were identified in different samples but at very low proportions (<1%) and only sporadically. Finally,  the ARGs localized on plasmid contigs were counted and grouped by AB class (Fig.  7C).  Consistently,  multidrug/efflux ARGs dominated the picture,  followed by glycopeptide and Î²-lactam ARGs.  Minor differences between sample groups include (1) higher proportions of multidrug/efflux and chloramphenicol ARGs in mature versus young lettuce,  (2) lower proportions of Î²-lactam ARGs in mature versus young lettuce,  (3) practical absence of chloramphenicol ARGs in soil and manure,  and (4) lower proportions of multidrug/efflux but slightly higher proportions of MLSB ARGs in untreated versus UV-treated river water. Procrustes analysis reveals high correlation between lettuce microbial community structure and resistomeAt all phylogenetic levels,  the correlation between resistome and microbial community structure was highly significant (shown for phylum and genus in Fig.  S8;  P at least <2e\\x88\\x924),  with the highest significance reached at genus level (Fig.  S8B;  P < 1e\\x88\\x925).  Further,  a clear separation of the three time points (weeks 0,  1,  and 5) was obtained at all phylogenetic levels.  Finally,  a separation of treatments A/D (no manure application) from B/C (manure application) was evident in young lettuce (week 1),  in accordance to previous analyses (Fig.  5). ', 'discussion': 'DiscussionThe boundaries of culture-based techniquesThe suitability of diagnostic plates\\x80\\x94developed for clinical samples\\x80\\x94for use with environmental samples is often discussed,  and our study confirmed this problem.  In the present study,  bacteria culturable on antibiotic-containing selective agar plates from lettuce consistently showed a dependence on plant growth time.  However,  whether the lettuces were grown with or without pig manure,  or irrigated with untreated or UV-treated river water showed no significant effect.  Identification of presumptive ARB colonies from all environments further revealed very low abundance of clinically relevant bacteria,  and the plates were overgrown with typical environmental bacteria such as Pseudomonas spp.  or Sphingomonas spp.  (White et al.,  Crone et al. ).  As very recent work by Schreiber et al.  demonstrated,  adaptation of incubation conditions such as increasing temperature can help to suppress unwanted background microbiota,  and thereby increase sensitivity.  It still remains to be shown,  however,  whether such environmentally optimized culturing can yield enough insight to narrow the gap between culture-based and molecular techniques significantly. Environment-specific microbiome structuresThe main determinant of microbial community composition was environment (lettuce,  soil,  water,  or manure).  Thus,  pig manure was heavily dominated by Proteobacteria,  in accordance with a very recent study by Wang et al.,  who found 85% relative abundance of Proteobacteria in fresh pig manure.  In contrast,  soil was dominated by Planctomycetes,  a phylum typically encountered abundantly in soil (Buckley et al. ).  The overall soil community composition was not significantly affected by either time or treatment,  despite the overwhelming dominance of Proteobacteria in manure as well as UV-treated river water of weeks 1\\x80\\x934.  Confirming these findings,  soil microbial communities have been described as very resilient to change (Griffiths and Philippot). In contrast,  river water was much more variable with major differences between samplings.  While in weeks 0,  5,  and 6 communities were more diverse,  in weeks 1\\x80\\x934 Proteobacteria clearly dominated.  Precipitation data made evident that the more diverse river water samples were collected on days preceded by zero precipitation on at least 4 days prior to sampling and the sampling day itself (data not shown).  In contrast,  weeks 1\\x80\\x934 samplings were all preceded by at least 1 day of light to heavy rainfall (from 2. 4 to 21. 6 mm/h).  While Parcubacteria and Actinobacteria are prevalent in various water systems like lakes,  groundwater,  or freshwater ecosystems (Warnecke et al.,  Proctor et al.,  Tian et al. ),  rainfall has been shown to increase the abundance Proteobacteria,  in particular E.  coli,  especially in the proximity of upstream WWTPs (Shibata et al.,  Tornevi et al. ).  Finally,  in river water with predominant Proteobacteria (i. e.  after rainfall),  UV-disinfection led to a steep increase of their relative abundance,  in accordance with Becerra-Castro et al.  who found that Proteobacteria become predominant by regrowth after water disinfection. In-depth analysis of microbial communities on lettuce consistently showed a significant effect of plant growth time,  while treatment made no significant difference except in a few exceptions,  where seedlings before planting differed from lettuces of the four treatments on the field.  The transfer from the environmental sources investigated (soil,  water,  and manure) to the lettuce was therefore not the decisive factor in shaping the lettuce microbial community or resistome.  Taken together,  abundance of clinically relevant ARB families (Enterobacteriaceae,  Moraxellaceae,  and Pseudomonadaceae,  including the genera Pseudomonas and Acinetobacter) highly increased from very young lettuces (week 1) to mature lettuces (week 5),  but was strongly decreased in oversized and partially already decaying lettuces (week 6).  Overall,  the observed time trend seemed related to the decreasing soil contact as plants grow,  and the strongly reincreased soil contact in week 6,  when the lettuces were already beyond the optimal harvest point.  Soil thereby seems to serve as initial inoculant of young plants,  whose communities thereafter develop independently of the stable communities in the surrounding soil.  In conclusion,  the observed temporal dynamic is relevant with regards to clinically relevant ARB taxa. To compare our findings,  we could not find another work investigating the development of lettuce\\x80\\x99s microbial communities over the growth period of the vegetable,  as one-time samplings of the vegetable grown under different conditions (e. g.  different amendments) are usually performed.  To our knowledge,  the only study including a time series was conducted in 2013 by Holvoet et al.  who sampled lettuce and its production system from different farms,  four times within the production cycle (seedlings,  and 2 weeks before,  1 week before,  and at harvest).  They focused,  however,  on AR E.  coli and did not discuss any changes related to plant growth time.  In contrast to the lack of time series,  many studies have investigated the microbiome of fresh produce to relate it to its production environment and different modes of culture.  They especially detected an effect of various types of amendment or a weak impact of irrigation on the microbiome as well as the resistome of the vegetables (Blau et al.,  Cerqueira et al.,  Fogler et al.,  Zhang et al.,  Summerlin et al.,  Sanz et al.,  Seyoum et al. ). The impact of amendment and UV-disinfection on the lettuce resistomeOf all investigated environments,  manure displayed the lowest diversity of ARGs.  Interestingly,  although the microbial community was heavily dominated by Proteobacteria,  only a third of all identified plasmid contigs were assigned to this phylum,  while about half remained unclassified.  The high proportion of unclassified plasmid contigs remains elusive,  since PlasFlow was developed to identify bacterial plasmid sequences in environmental samples,  and is therefore not biased toward clinical specimens.  While many studies have investigated the effect of soil amendment with manure on the resistome of fresh produce,  manure on its own is rarely analysed or discussed,  while the focus rather lies on the manure-amended soil as an entity.  Wang et al.  quantified ARG abundance in raw pig manure by targeted qPCR of nine ARGs,  showing high relative abundance of sul2 followed by tetM,  as opposed to dominance of sul1 in stored pig manure.  In the present study,  tetM and sul2 were among the top 15 most abundant ARGs in manure,  including mainly other tetracycline-,  MLS-,  and aminoglycoside ARGs such as aadA,  aadE,  tetQ,  tetW,  or ermF,  in good overall accordance with other reports (Wang et al.,  Blau et al.,  Zhang et al. ). Soil was overall very poor in AR determinants,  and comparable to young lettuce only in terms of ARG plasmid proportions.  The majority of the assigned plasmids belonged to Proteobacteria,  followed by Firmicutes,  and Actinobacteria,  a ranking not analogous to the relative abundance of these taxa in the respective microbial communities.  Of importance,  these phyla have been described previously as typical plasmid hosts in soil (Smalla et al. ).  Notably,  soil samples did cluster by sampling time for unknown reasons when considering BMRGs additionally to ARGs,  in contrast to their very tight clustering when considering ARGs only.  Comparable to pristine soils,  multidrug/efflux and aminoglycoside resistance were the most prominent in our soil samples as well,  whereas we additionally detected frequent glycopeptide resistance.  The overall low resistance numbers in soil are unexpected,  since soil is generally acknowledged as a diverse reservoir of antibiotic resistance (SÃ©veno et al. ),  including even resistance toward synthetic antimicrobials (Marshall et al. ),  although significant differences between different types of soil have been reported (Popowska et al. ).  However,  exactly that high complexity of soil might explain this result since soil samples assembled poorly,  probably due to insufficient sequencing depth,  as has been described previously (Wind et al. ).  When designing a study including a variety of environments,  one main dilemma therefore is whether to apply the same protocol to all for best comparability,  or whether to optimize the protocols by environment for maximum information. In water,  the second highest number of ARG subtypes of all environments was identified.  A recent study by Lee et al.  suggested using \\x80\\x98co-occurrence\\x80\\x99 of aadA,  sul1,  and class A Î²-lactamase genes as an indicator of wastewater-related pollution in river water.  In good accordance,  these three ARG subtypes were detected in our study in river water.  Notably,  class A Î²-lactamase genes were detected only in the Resfams annotation,  underlining the importance of combining several databases.  The most frequently detected resistance mechanisms included multidrug and Î²-lactam,  which are frequently detected in pristine waters as well,  while the herein investigated river water additionally contained aminoglycoside ARGs frequently.  Proportion of MDR contigs,  ARG\\x80\\x93BMRG coresistance,  and mobility incidence were always markedly higher in UV-treated compared to untreated river water.  An increase in ARG total relative abundance after UV-disinfection has been described earlier (Hu et al. ).  Mobility incidence (M%) on the other hand has been quantified by Ju et al.  in WWTP resistomes.  The authors found an M% of 8. 6% and 20. 0% for ARGs and MRGs,  respectively.  These are 64-fold and 99-fold higher,  respectively,  than what was found in the present work in untreated river water.  Mobile ARGs and BMRGs seem therefore to be enriched by wastewater treatment processes (as observed in the present work for UV-disinfection of river water),  but diluted again with less mobile resistance determinants when introduced into the environment like a nearby river.  Finally,  Firmicute plasmids were detected practically only in untreated river water,  although this phylum showed very low relative abundance in these samples. Lettuce had the highest ARG diversity of all investigated environments,  independently of the treatment.  In common with pristine plants,  lettuce harbored multidrug/efflux,  Î²-lactam,  and aminoglycoside ARGs most frequently,  while it additionally contained an important number of quinolone ARGs.  Taken together,  the lettuces ready for harvest and consumption compared to young lettuce had increased resistance markers on all examined levels,  with manure slightly exacerbating the effect.  Regarding our central question,  unde venis,  it seems from our findings that soil and manure (when present) were the main source environments for the microbiome and resistome of young plants,  but that the leaves of the growing plants enriched taxa and resistance determinants,  which were below detection in the source environments. Clearly,  in lettuce as in all other environments,  community abundance of a taxon did not correlate with plasmid proportion of that taxon.  We therefore conclude that Proteobacteria were the main contributors of plasmid-borne ARGs in all environments,  irrespective of their community abundance.  A study investigating minimally processed vegetables-producing facilities analysed ARGs in initial and final product,  contact surfaces,  and operator swabs by taxonomy and plasmid-association (Valentino et al. ).  As in the present study,  the proportion of plasmid-associated ARGs was relatively low,  and many of these were associated with Proteobacteria such as Acinetobacter,  Pantoea,  Pseudorhodoferax,  or Rahnella,  while on surfaces and operator swabs often additionally with Firmicutes like Staphylococcus and Bacillus. When comparing the environments for commonalities,  we found in all of them (1) similar distribution of ARGs to the different types,  in accordance to the findings of Wang et al.  finding dominance of efflux-related ARGs,  (2) similar distribution of plasmid-associated ARGs to the different types,  (3) dominance of the same few BMRGs encoding resistance toward W,  Zn,  and Mo,  and (4) higher BMRG mobility compared to ARG mobility,  in accordance to the findings of Ju et al.,  who used quantitative metagenomic and metatranscriptomic approaches to study antibiotic,  biocide,  and metal resistance in different compartments of 12 WWTPs.  Core ARGs detected in all four environments mainly belonged to MDR (efflux).  Further ARGs of interest included aminoglycoside (aadA,  aph(6)-I),  Î²-lactam (metallo-Î²-lactamase),  glycopeptide (vanR,  vanX),  sulfonamide (sul2),  and tetracycline (tetA) ARGs. Interestingly,  lettuce shared most ARGs with water and then with soil.  A study by Shen et al.  investigated the effect of water (with or without pharmaceuticals) applied by overhead versus soil-surface irrigation on the resistome and microbiome of lettuce in controlled greenhouse conditions.  While exposure to pharmaceuticals did not result in consistent patterns of change in soil and lettuce,  overhead irrigation resulted in greater abundance as well as diversity of ARGs and MGEs in lettuce shoots.  Of the 42 ARGs detected in their study,  17 (41%) were also detected in the present work,  among which bacA (bacitracin),  oleC (MLSB),  and mexF (multidrug/efflux),  which were of middle to high relative abundance in both studies.  Another study investigating the use of municipal wastewater effluents also described an irrigation effect on the resistome of romaine lettuce (Summerlin et al. ).  In contrast,  Seyoum et al.  compared irrigation of tomatoes with treated wastewaters to freshwater irrigation and could not detect propagation of the investigated ARGs to the tomatoes.  Finally,  Cerqueira et al.  found a minimal effect of irrigation on ARG abundance by qPCR,  while crop type was more decisive.  Taken together,  findings on the effect of irrigation on the resistome of fresh produce are contradictive,  with some studies detecting an effect while others do not.  It must be kept in mind,  however,  that the type of water investigated and thereby varying loads of contained resistance determinants,  combined with the type of produce under investigation may contribute to this discrepancy. As mentioned earlier,  a higher sequencing depth for complex environments such as soil should be considered,  to comprehend the plethora of soil-borne ARGs in more detail.  Nevertheless,  an indirect effect of soil via amendment could be shown in the present work: Clustering by two methods (PCoA and NMDS) using either ARGs alone or combined with BMRGs revealed an effect of manure on lettuce,  especially on young plants with increased soil contact.  When considering ARG presence/absence only,  manure clustered closely with young lettuce grown in amended soil.  These findings are in good accordance with the work of Sanz et al.,  who concluded that fertilizers rather than soil were the main source of clinically relevant ARGs detected in foods.  Additionally in the present study,  of the five BMRGs most discriminatory for amendment in young lettuce,  three were among the top 10 BMRGs in manure.  An effect of amendment on the resistome of fresh produce has also been described in other studies (Fogler et al.,  Zhang et al.,  Huang et al.,  Wind et al. ).  In the study by Zhang et al.,  the ARG transmission pathways between soil and lettuce were explored 90 days after application of poultry and cattle manure.  They found mainly multidrug/efflux-related ARGs followed by Î²-lactam and aminoglycoside ARGs as in the present study,  and of the 32 ARGs,  which were shared between soil with poultry manure and the lettuce leaf surface,  10 (31%) were detected as well in the present study on lettuce and in soil/manure,  including a variety of tetracycline resistance genes,  bacA,  mexE,  and aadA (aminoglycoside).  They further noted a stronger impact of poultry than cattle manure,  suggesting that the type of manure can be an important factor to consider with respect to ARG abundance in fresh produce grown with such fertilizers.  Taken together,  in the presented study manure was revealed as the main source shaping the resistome of lettuce,  especially that of the young plants. Finally,  we detected a strong correlation between microbial community structures and resistome,  especially in lettuce,  as has been recently described in a very similar system (Sun et al. ).  Therefore,  measures impacting the lettuce microbiome,  e. g.  by decreasing the abundance of Proteobacteria,  are very likely to result in shifts in the lettuce resistome.  The sort and effectiveness of such measures,  however,  must be subject to further studies. ', 'conclusion': 'ConclusionsIn the present work,  the effect of manure and irrigation on the microbiome and resistome of field-grown lettuce was investigated.  By culturing,  only few clinically relevant ARB could be recovered,  including E.  cloacae,  E.  coli,  and S.  fonticola on ESBL agar as well as K.  oxytoca on CRE agar.  Of note,  adding an enrichment step prior to plating can enhance bacterial detection,  whereas direct plating allows quantitative detection.  Microbiome analysis showed a clear shift in microbial communities as a function of lettuce growth time,  but no effect of soil amendment or type of irrigation water.  Resistome analysis confirmed the effect of growth time,  but additionally revealed an impact of manure,  especially in young lettuces with increased soil contact.  An impact of type of irrigation water on the lettuce resistome could not be observed although lettuce shared most ARGs with water.  However,  UV-disinfection did increase the proportion of examined resistance markers such as MDR,  ARG\\x80\\x93BMRG coresistance,  and mobility incidence.  Moreover,  although surface water irrigation did not affect the overall lettuce resistome significantly,  it can carry clinically relevant ARB to the produce,  and should therefore be subject to monitoring.  Overall,  lettuces ready for harvest and consumption had increased resistance markers compared to young plants on all examined levels,  with manure slightly exacerbating the effect.  With regard to our central question,  unde venis,  our study revealed manure as the main source of resistance determinants on lettuce.  Careful handling of organic fertilizers such as manure as well as measures minimizing soil contact of the vegetables is therefore key to minimize antibiotic resistance markers on fresh produce. ', 'references': []}\n"
     ]
    }
   ],
   "source": [
    "# reader = JSONReader()\n",
    "# documents = reader.load_data(input_file=\"mock_research_papers.json\", extra_info={})\n",
    "\n",
    "def process_graph_conversion(batch):\n",
    "    return llm_transformer.convert_to_graph_documents(batch)\n",
    "\n",
    "documents = []\n",
    "for json_data in papers:\n",
    "    print(papers)\n",
    "    print(json_data)\n",
    "# with open('test_document.json', 'r', encoding='utf-8') as f:\n",
    "#     json_data = json.load(f)\n",
    "\n",
    "    json_data_lower = {key.lower(): value for key, value in json_data.items()}\n",
    "    content = (\n",
    "        # json_data_lower.get(\"introduction\", \"\") +\n",
    "        json_data_lower.get(\"methods\", \"\") +\n",
    "        json_data_lower.get(\"results\", \"\") +\n",
    "        json_data_lower.get(\"discussion\", \"\") +\n",
    "        json_data_lower.get(\"conclusion\", \"\")\n",
    "    )       \n",
    "\n",
    "    metadata = {\n",
    "        \"title\": json_data_lower.get(\"title\", \"Untitled\"),\n",
    "        \"authors\": json_data_lower.get(\"authors\", \"Unknown\"),\n",
    "        \"publication_date\": json_data_lower.get(\"publication_date\", \"Unknown\"),\n",
    "        \"journal_name\": json_data_lower.get(\"journal_name\", \"Unknown\")\n",
    "    }\n",
    "\n",
    "    document = Document(page_content=content, metadata=metadata)\n",
    "    documents.append(document)\n",
    "\n",
    "# text_splitter = TokenTextSplitter(chunk_size=4000, chunk_overlap=200)\n",
    "# texts = text_splitter.split_text(content)\n",
    "\n",
    "# for chunk in texts:\n",
    "#     document = Document(page_content=chunk, metadata=metadata)\n",
    "#     documents.append(document)\n",
    "\n",
    "\n",
    "llm=ChatOpenAI(temperature=0, model_name=\"gpt-4o-mini-2024-07-18\") \n",
    "llm_transformer = LLMGraphTransformer(llm=llm)\n",
    "\n",
    "\n",
    "batch_size = 10  # Adjust the batch size based on API limits and performance testing\n",
    "batched_documents = [documents[i:i+batch_size] for i in range(0, len(documents), batch_size)]\n",
    "\n",
    "with ThreadPoolExecutor(max_workers=13) as executor:  # Adjust max_workers based on system resources\n",
    "    future_to_batch = {executor.submit(process_graph_conversion, batch): batch for batch in batched_documents}\n",
    "\n",
    "    for future in as_completed(future_to_batch):\n",
    "        graph_documents = future.result()\n",
    "        graph.add_graph_documents(\n",
    "            graph_documents,\n",
    "            baseEntityLabel=True,\n",
    "            include_source=True\n",
    "        )\n",
    "\n",
    "# graph_documents = llm_transformer.convert_to_graph_documents(documents)\n",
    "# graph.add_graph_documents(\n",
    "#     graph_documents,\n",
    "#     baseEntityLabel=True,\n",
    "#     include_source=True\n",
    "# )\n",
    "\n",
    "client.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "972b95ff539e4c73a9b70d0caed4c19f",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "GraphWidget(layout=Layout(height='800px', width='100%'))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# directly show the graph resulting from the given Cypher query\n",
    "default_cypher =\"MATCH (s)-[r:!MENTIONS]->(t) RETURN s,r,t LIMIT 50\"\n",
    "\n",
    "def fetchGraphData(cypher: str = default_cypher):\n",
    "    # create a neo4j session to run queries\n",
    "    driver = GraphDatabase.driver(\n",
    "        uri=os.environ[\"NEO4J_URI\"],\n",
    "        auth=(os.environ[\"NEO4J_USERNAME\"], os.environ[\"NEO4J_PASSWORD\"])\n",
    "    )\n",
    "    session = driver.session()\n",
    "\n",
    "    # Run the query and fetch results\n",
    "    result = session.run(cypher)\n",
    "\n",
    "    # Convert results to a format suitable for JSON\n",
    "    graph_data = []\n",
    "    for record in result:\n",
    "        source = record['s']\n",
    "        relationship = record['r']\n",
    "        target = record['t']\n",
    "        \n",
    "        graph_data.append({\n",
    "            'source': dict(source),  # Convert Node to dict\n",
    "            'relationship': {\n",
    "                'type': relationship.type,  # Extract relationship type\n",
    "                'properties': dict(relationship)  # Convert Relationship properties to dict\n",
    "            },\n",
    "            'target': dict(target)  # Convert Node to dict\n",
    "        })\n",
    "\n",
    "    session.close()\n",
    "    return graph_data\n",
    "\n",
    "\n",
    "def saveGraphDataToJSON(graph_data, filename='graph_data.json'):\n",
    "    # Save the graph data to a JSON file\n",
    "    with open(filename, 'w') as json_file:\n",
    "        json.dump(graph_data, json_file, indent=4)\n",
    "\n",
    "\n",
    "def showGraph(cypher: str = default_cypher):\n",
    "    # create a neo4j session to run queries\n",
    "    driver = GraphDatabase.driver(\n",
    "        uri = os.environ[\"NEO4J_URI\"],\n",
    "        auth = (os.environ[\"NEO4J_USERNAME\"],\n",
    "                os.environ[\"NEO4J_PASSWORD\"]))\n",
    "    session = driver.session()\n",
    "    widget = GraphWidget(graph = session.run(cypher).graph())\n",
    "    widget.node_label_mapping = 'id'\n",
    "    #display(widget)\n",
    "    return widget\n",
    "\n",
    "showGraph()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "graph_data = fetchGraphData()\n",
    "saveGraphDataToJSON(graph_data) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "vector_index = Neo4jVector.from_existing_graph(\n",
    "    OpenAIEmbeddings(),\n",
    "    search_type=\"hybrid\",\n",
    "    node_label=\"Document\",\n",
    "    text_node_properties=[\"text\"],\n",
    "    embedding_node_property=\"embedding\"\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'llm' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[5], line 30\u001b[0m\n\u001b[0;32m     16\u001b[0m \u001b[38;5;66;03m# Edit your prompt to focus on extracting more information from papers\u001b[39;00m\n\u001b[0;32m     17\u001b[0m prompt \u001b[38;5;241m=\u001b[39m ChatPromptTemplate\u001b[38;5;241m.\u001b[39mfrom_messages(\n\u001b[0;32m     18\u001b[0m     [\n\u001b[0;32m     19\u001b[0m         (\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m     27\u001b[0m     ]\n\u001b[0;32m     28\u001b[0m )\n\u001b[1;32m---> 30\u001b[0m entity_chain \u001b[38;5;241m=\u001b[39m prompt \u001b[38;5;241m|\u001b[39m \u001b[43mllm\u001b[49m\u001b[38;5;241m.\u001b[39mwith_structured_output(Entities)\n",
      "\u001b[1;31mNameError\u001b[0m: name 'llm' is not defined"
     ]
    }
   ],
   "source": [
    "class Entities(BaseModel):\n",
    "    \"\"\"Extracting information about research papers and related entities.\"\"\"\n",
    "    names: List[str] = Field(\n",
    "        ...,\n",
    "        description=\"All the people, group, entities, methods and organization that appear in the text\",\n",
    "    )\n",
    "    tools: List[str] = Field(\n",
    "        ...,\n",
    "        description=\"Tools, techniques, or software mentioned in the text\",\n",
    "    )\n",
    "    topics: List[str] = Field(\n",
    "        ...,\n",
    "        description=\"Relevant topics or keywords from the text\",\n",
    "    )\n",
    "\n",
    "# Edit your prompt to focus on extracting more information from papers\n",
    "prompt = ChatPromptTemplate.from_messages(\n",
    "    [\n",
    "        (\n",
    "            \"system\",\n",
    "            \"You are extracting people, groups, entities, results, methods, topics and tools from research papers for a knowledge graph.\",\n",
    "        ),\n",
    "        (\n",
    "            \"human\",\n",
    "            \"Use the given format to extract information from the following \"\n",
    "            \"input: {question}\",\n",
    "        ),\n",
    "    ]\n",
    ")\n",
    "\n",
    "entity_chain = prompt | llm.with_structured_output(Entities)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Entities(names=['gut microbiome', 'tools'])"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "entity_chain.invoke({\"question\": \"What are the different tools for gut microbiome?\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "def generate_full_text_query(input: str) -> str:\n",
    "    \"\"\"\n",
    "    Generate a full-text search query for a given input string.\n",
    "\n",
    "    This function constructs a query string suitable for a full-text search.\n",
    "    It processes the input string by splitting it into words and appending a\n",
    "    similarity threshold (~2 changed characters) to each word, then combines\n",
    "    them using the AND operator. Useful for mapping entities from user questions\n",
    "    to database values, and allows for some misspelings.\n",
    "    \"\"\"\n",
    "    full_text_query = \"\"\n",
    "    words = [el for el in remove_lucene_chars(input).split() if el]\n",
    "    for word in words[:-1]:\n",
    "        full_text_query += f\" {word}~2 AND\"\n",
    "    full_text_query += f\" {words[-1]}~2\"\n",
    "    return full_text_query.strip()\n",
    "\n",
    "# Fulltext index query\n",
    "def structured_retriever(question: str) -> str:\n",
    "    \"\"\"\n",
    "    Collects the neighborhood of entities mentioned\n",
    "    in the question\n",
    "    \"\"\"\n",
    "    result = \"\"\n",
    "    entities = entity_chain.invoke({\"question\": question})\n",
    "    for entity in entities.names:\n",
    "        response = graph.query(\n",
    "            \"\"\"CALL db.index.fulltext.queryNodes('entity', $query, {limit: 2})\n",
    "                YIELD node, score\n",
    "                CALL (node) {\n",
    "                WITH node\n",
    "                MATCH (node)-[r:MENTIONS]->(neighbor)\n",
    "                RETURN node.id + ' - ' + type(r) + ' -> ' + neighbor.id AS output\n",
    "                UNION ALL\n",
    "                WITH node\n",
    "                MATCH (node)<-[r:MENTIONS]-(neighbor)\n",
    "                RETURN neighbor.id + ' - ' + type(r) + ' -> ' + node.id AS output\n",
    "                }\n",
    "                RETURN output LIMIT 50\n",
    "            \"\"\",\n",
    "            {\"query\": generate_full_text_query(entity)},\n",
    "        )\n",
    "        result += \"\\n\".join([el['output'] for el in response])\n",
    "    return result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "print(structured_retriever(\"What is gut microbiome?\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "def retriever(question: str):\n",
    "    print(f\"Search query: {question}\")\n",
    "    structured_data = structured_retriever(question)\n",
    "    unstructured_data = [el.page_content for el in vector_index.similarity_search(question)]\n",
    "    final_data = f\"\"\"Structured data:\n",
    "{structured_data}\n",
    "Unstructured data:\n",
    "{\"#Document \". join(unstructured_data)}\n",
    "    \"\"\"\n",
    "    return final_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Condense a chat history and follow-up question into a standalone question\n",
    "_template = \"\"\"Given the following conversation and a follow up question, rephrase the follow up question to be a standalone question,\n",
    "in its original language.\n",
    "Chat History:\n",
    "{chat_history}\n",
    "Follow Up Input: {question}\n",
    "Standalone question:\"\"\"  # noqa: E501\n",
    "CONDENSE_QUESTION_PROMPT = PromptTemplate.from_template(_template)\n",
    "\n",
    "def _format_chat_history(chat_history: List[Tuple[str, str]]) -> List:\n",
    "    buffer = []\n",
    "    for human, ai in chat_history:\n",
    "        buffer.append(HumanMessage(content=human))\n",
    "        buffer.append(AIMessage(content=ai))\n",
    "    return buffer\n",
    "\n",
    "_search_query = RunnableBranch(\n",
    "    # If input includes chat_history, we condense it with the follow-up question\n",
    "    (\n",
    "        RunnableLambda(lambda x: bool(x.get(\"chat_history\"))).with_config(\n",
    "            run_name=\"HasChatHistoryCheck\"\n",
    "        ),  # Condense follow-up question and chat into a standalone_question\n",
    "        RunnablePassthrough.assign(\n",
    "            chat_history=lambda x: _format_chat_history(x[\"chat_history\"])\n",
    "        )\n",
    "        | CONDENSE_QUESTION_PROMPT\n",
    "        | ChatOpenAI(temperature=0)\n",
    "        | StrOutputParser(),\n",
    "    ),\n",
    "    # Else, we have no chat history, so just pass through the question\n",
    "    RunnableLambda(lambda x : x[\"question\"]),\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "template = \"\"\"Answer the question based only on the following context:\n",
    "{context}\n",
    "\n",
    "Question: {question}\n",
    "Use natural language and be concise.\n",
    "Answer:\"\"\"\n",
    "prompt = ChatPromptTemplate.from_template(template)\n",
    "\n",
    "chain = (\n",
    "    RunnableParallel(\n",
    "        {\n",
    "            \"context\": _search_query | retriever,\n",
    "            \"question\": RunnablePassthrough(),\n",
    "        }\n",
    "    )\n",
    "    | prompt\n",
    "    | llm\n",
    "    | StrOutputParser()\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Search query: What is gut microbiome?\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Received notification from DBMS server: {severity: WARNING} {code: Neo.ClientNotification.Statement.FeatureDeprecationWarning} {category: DEPRECATION} {title: This feature is deprecated and will be removed in future versions.} {description: CALL subquery without a variable scope clause is now deprecated. Use CALL () { ... }} {position: line: 1, column: 1, offset: 0} for query: \"CALL { CALL db.index.vector.queryNodes($index, $k, $embedding) YIELD node, score WITH collect({node:node, score:score}) AS nodes, max(score) AS max UNWIND nodes AS n RETURN n.node AS node, (n.score / max) AS score UNION CALL db.index.fulltext.queryNodes($keyword_index, $query, {limit: $k}) YIELD node, score WITH collect({node:node, score:score}) AS nodes, max(score) AS max UNWIND nodes AS n RETURN n.node AS node, (n.score / max) AS score } WITH node, max(score) AS score ORDER BY score DESC LIMIT $k RETURN reduce(str='', k IN ['text'] | str + '\\\\n' + k + ': ' + coalesce(node[k], '')) AS text, node {.*, `embedding`: Null, id: Null, `text`: Null} AS metadata, score\"\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'The gut microbiome refers to the complex community of microorganisms, including bacteria, fungi, and viruses, that inhabit the digestive tract. It plays a crucial role in digestion, immune function, and has been linked to various health conditions, including mental health issues like depression.'"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chain.invoke({\"question\": \"What is gut microbiome?\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.4 (main, Jul 25 2023, 17:36:13) [Clang 14.0.3 (clang-1403.0.22.14.1)]"
  },
  "vscode": {
   "interpreter": {
    "hash": "b0fa6594d8f4cbf19f97940f81e996739fb7646882a419484c72d19e05852a7e"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
